compose: TestComposeCompare failed

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3199215&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3199215&tab=artifacts#/) on master @ [5fd1dda1cde0a91643d39209c6458dcefcbfc180](https://github.com/cockroachdb/cockroach/commits/5fd1dda1cde0a91643d39209c6458dcefcbfc180):


```
        test_1        |         compare_test.go:188: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			tab_319517.col1_4 AS col_496993,
        test_1        |             			tab_319518.col3_13 AS col_496994,
        test_1        |             			NULL AS col_496995,
        test_1        |             			tab_319521.col2_0 AS col_496996,
        test_1        |             			tab_319518.tableoid AS col_496997,
        test_1        |             			ARRAY[14924:::INT8,6093:::INT8,3097:::INT8,(-30703):::INT8] AS col_496998
        test_1        |             		FROM
        test_1        |             			postgres.public.table1@[0] AS tab_319517,
        test_1        |             			postgres.public.table3@table3_col3_4_col3_2_col3_13_col3_8_idx AS tab_319518
        test_1        |             			JOIN postgres.public.table4@table4_col4_6_col4_2_key AS tab_319519
        test_1        |             				JOIN postgres.public.table2@table2_col2_3_col2_2_key AS tab_319520
        test_1        |             					JOIN postgres.public.table2@[0] AS tab_319521 ON (tab_319520.col2_3) = (tab_319521.tableoid)
        test_1        |             					JOIN postgres.public.table3@[0] AS tab_319522 ON (tab_319520.col2_2) = (tab_319522.tableoid) ON
        test_1        |             						(tab_319519.tableoid) = (tab_319520.col2_2)
        test_1        |             						AND (tab_319519.col4_4) = (tab_319520.col2_4)
        test_1        |             						AND (tab_319519.col4_1) = (tab_319522.col3_3)
        test_1        |             				JOIN postgres.public.table2 AS tab_319523
        test_1        |             					JOIN postgres.public.table1 AS tab_319524 ON (tab_319523.col2_3) = (tab_319524.col1_3) ON
        test_1        |             						(tab_319522.col3_12) = (tab_319524.col1_3)
        test_1        |             						AND (tab_319521.col2_3) = (tab_319524.col1_3)
        test_1        |             						AND (tab_319520.col2_3) = (tab_319523.col2_3)
        test_1        |             						AND (tab_319521.col2_2) = (tab_319524.col1_3) ON (tab_319518.col3_13) = (tab_319522.col3_6)
        test_1        |             		ORDER BY
        test_1        |             			tab_319517.crdb_internal_mvcc_timestamp
        test_1        |             	)
        test_1        |             		AS tab_319525
        test_1        |             ORDER BY
        test_1        |             	col_496993 NULLS FIRST,
        test_1        |             	col_496994 NULLS FIRST,
        test_1        |             	col_496995 NULLS FIRST,
        test_1        |             	col_496996 NULLS FIRST,
        test_1        |             	col_496997 NULLS FIRST,
        test_1        |             	col_496998 NULLS FIRST
        test_1        |         compare_test.go:189: cockroach2: ping: context deadline exceeded
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-8721

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3209832&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3209832&tab=artifacts#/) on master @ [f0e2aa6abbbbf3318ea20e7dbcbe40819a809b83](https://github.com/cockroachdb/cockroach/commits/f0e2aa6abbbbf3318ea20e7dbcbe40819a809b83):


```
        test_1        |             			tab_198823.col1_1 AS col_350666,
        test_1        |             			tab_198819.col1_14 AS col_350667
        test_1        |             		FROM
        test_1        |             			postgres.public.table1@[0] AS tab_198815,
        test_1        |             			postgres.public.table1@[0] AS tab_198816
        test_1        |             			JOIN postgres.public.table1@[0] AS tab_198817
        test_1        |             				JOIN postgres.public.table1@[0] AS tab_198818
        test_1        |             					JOIN postgres.public.table1@[0] AS tab_198819
        test_1        |             						JOIN postgres.public.table1@[0] AS tab_198820
        test_1        |             							JOIN postgres.public.table1@[0] AS tab_198821 ON
        test_1        |             									(tab_198820.col1_11) = (tab_198821.col1_11) ON
        test_1        |             								(tab_198819.col1_2) = (tab_198820.col1_2)
        test_1        |             						JOIN postgres.public.table1@[0] AS tab_198822 ON (tab_198820.col1_0) = (tab_198822.col1_0) ON
        test_1        |             							(tab_198818.col1_8) = (tab_198821.col1_8)
        test_1        |             							AND (tab_198818.col1_6) = (tab_198822.col1_15)
        test_1        |             							AND (tab_198818.col1_14) = (tab_198821.col1_15)
        test_1        |             							AND (tab_198818.col1_6) = (tab_198820.col1_0)
        test_1        |             							AND (tab_198818.col1_0) = (tab_198820.col1_0)
        test_1        |             							AND (tab_198818.col1_5) = (tab_198822.col1_9) ON
        test_1        |             						(tab_198817.col1_6) = (tab_198818.col1_6) ON
        test_1        |             					(tab_198816.col1_6) = (tab_198820.col1_0) AND (tab_198816.col1_0) = (tab_198817.col1_14),
        test_1        |             			postgres.public.table1@[0] AS tab_198823
        test_1        |             	)
        test_1        |             		AS tab_198824
        test_1        |             ORDER BY
        test_1        |             	col_350664 NULLS FIRST, col_350665 NULLS FIRST, col_350666 NULLS FIRST, col_350667 NULLS FIRST
        test_1        |         compare_test.go:189: cockroach1: ping: context deadline exceeded
        test_1        |     --- FAIL: TestCompare/postgres (1.52s)
        test_1        |         compare_test.go:125: starting test: postgres
        test_1        |         compare_test.go:132: initializing connection: cockroach1
        test_1        |         compare_test.go:132: initializing connection: postgres
        test_1        |         compare_test.go:149: CREATE TABLE table1 (col1_0 REGCLASS NOT NULL, col1_1 INT4 NOT NULL, col1_2 DECIMAL NOT NULL, col1_3 VARCHAR, col1_4 REGPROCEDURE NOT NULL, col1_5 JSONB NOT NULL, col1_6 TIMESTAMP NOT NULL, col1_7 "char" NOT NULL, PRIMARY KEY (col1_2, col1_7, col1_0, col1_6, col1_1, col1_4), col1_8 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_6 IS NULL THEN e'\x1b\x03e\x10'::TEXT ELSE e'\x1c'::TEXT END) STORED, col1_9 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_6 IS NULL THEN NULL ELSE ''::TEXT END) STORED, col1_10 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_6 IS NULL THEN NULL ELSE e'\x00'::TEXT END) STORED, col1_11 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_0 IS NULL THEN e'R6Bvg\x19'::TEXT ELSE NULL END) STORED, col1_12 TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_7)) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_3, col1_8 DESC, col1_1, col1_10 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_1 DESC, col1_9 DESC, col1_3, col1_10 DESC, col1_2 ASC, col1_11 ASC, col1_8 DESC);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('x', 'xcogl', 'gsrzhc', 'gmw', 'eaoem');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('eb', 'zjgs', 'ttab', 'el', 'nld', 'ishpey');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('i', 'tng', 'hcv', 'edxkp');
        test_1        |             
        test_1        |         compare_test.go:150: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3254516&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3254516&tab=artifacts#/) on master @ [1c46e1cd4e5be986bf9d13799bb7e13ddc896ed2](https://github.com/cockroachdb/cockroach/commits/1c46e1cd4e5be986bf9d13799bb7e13ddc896ed2):


```
        postgres_1    | 2021-08-03 05:21:02.694 UTC [106] STATEMENT:  CREATE TABLE table1 (col1_0 TEXT NOT NULL, col1_1 BYTEA NOT NULL, col1_2 TIMETZ NOT NULL, PRIMARY KEY (col1_0, col1_2, col1_1), col1_3 TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_0)) STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_3, col1_1 ASC, col1_0);
        postgres_1    | 	CREATE INDEX ON table1 (col1_0 ASC, col1_3, col1_1 ASC, col1_2 ASC);
        postgres_1    | 	CREATE INDEX ON table1 (col1_0 ASC);
        postgres_1    | 	CREATE INDEX ON table1 (col1_0 DESC, col1_3 ASC, col1_1 DESC, col1_2 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_1, col1_3 DESC);
        postgres_1    | 	CREATE INDEX ON table1 (col1_2, col1_3 DESC);
        postgres_1    | 	CREATE TABLE table2 (col2_0 FLOAT8 NOT NULL, col2_1 BIT(5) NULL, col2_2 "char", col2_3 FLOAT4 NULL, col2_4 TEXT NULL, col2_5 GEOMETRY NULL, col2_6 INET NULL, col2_7 FLOAT8 NOT NULL GENERATED ALWAYS AS (col2_0 + 1.7976931348623157e+308::FLOAT8) STORED, col2_8 FLOAT4 NULL GENERATED ALWAYS AS (col2_3 + col2_0) STORED, col2_9 FLOAT4 NULL GENERATED ALWAYS AS (col2_3 + col2_0) STORED, col2_10 FLOAT8 NULL GENERATED ALWAYS AS (col2_0 + col2_3) STORED, col2_11 TEXT NULL GENERATED ALWAYS AS (lower(col2_4)) STORED);
        postgres_1    | 	CREATE TABLE table3 (col3_0 TEXT NOT NULL, col3_1 NAME NULL, col3_2 REGTYPE, col3_3 GEOGRAPHY, col3_4 GEOGRAPHY NULL, col3_5 OID NOT NULL, col3_6 INTERVAL NOT NULL, col3_7 FLOAT4, col3_8 GEOMETRY, col3_9 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_5 IS NULL THEN e'\x00'::TEXT ELSE e'\U00002603'::TEXT END) STORED, col3_10 TEXT NULL GENERATED ALWAYS AS (lower(col3_1)) STORED, col3_11 TEXT NULL GENERATED ALWAYS AS (lower(col3_1)) STORED);
        postgres_1    | 	CREATE INDEX ON table3 (col3_9 DESC, col3_2 DESC, col3_6) INCLUDE (col3_1, col3_3, col3_4, col3_5, col3_7, col3_8);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('ppeall', 'qz', 'ryqly');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('e', 'jep');
        postgres_1    | 	CREATE TYPE rand_typ_2 AS ENUM ('k');
        postgres_1    | 	CREATE TYPE rand_typ_3 AS ENUM ('lwrsnh', 'al', 'hrpw', 'qkvp');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (607.10s)
        test_1        |     compare_test.go:116: Checking connection to: postgres
        test_1        |     compare_test.go:116: Checking connection to: cockroach1
        test_1        |     compare_test.go:116: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (2.67s)
        test_1        |         compare_test.go:125: starting test: postgres
        test_1        |         compare_test.go:132: initializing connection: cockroach1
        test_1        |         compare_test.go:132: initializing connection: postgres
        test_1        |         compare_test.go:149: CREATE TABLE table1 (col1_0 TEXT NOT NULL, col1_1 BYTEA NOT NULL, col1_2 TIMETZ NOT NULL, PRIMARY KEY (col1_0, col1_2, col1_1), col1_3 TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_0)) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_3, col1_1 ASC, col1_0);
        test_1        |             CREATE INDEX ON table1 (col1_0 ASC, col1_3, col1_1 ASC, col1_2 ASC);
        test_1        |             CREATE INDEX ON table1 (col1_0 ASC);
        test_1        |             CREATE INDEX ON table1 (col1_0 DESC, col1_3 ASC, col1_1 DESC, col1_2 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_1, col1_3 DESC);
        test_1        |             CREATE INDEX ON table1 (col1_2, col1_3 DESC);
        test_1        |             CREATE TABLE table2 (col2_0 FLOAT8 NOT NULL, col2_1 BIT(5) NULL, col2_2 "char", col2_3 FLOAT4 NULL, col2_4 TEXT NULL, col2_5 GEOMETRY NULL, col2_6 INET NULL, col2_7 FLOAT8 NOT NULL GENERATED ALWAYS AS (col2_0 + 1.7976931348623157e+308::FLOAT8) STORED, col2_8 FLOAT4 NULL GENERATED ALWAYS AS (col2_3 + col2_0) STORED, col2_9 FLOAT4 NULL GENERATED ALWAYS AS (col2_3 + col2_0) STORED, col2_10 FLOAT8 NULL GENERATED ALWAYS AS (col2_0 + col2_3) STORED, col2_11 TEXT NULL GENERATED ALWAYS AS (lower(col2_4)) STORED);
        test_1        |             CREATE TABLE table3 (col3_0 TEXT NOT NULL, col3_1 NAME NULL, col3_2 REGTYPE, col3_3 GEOGRAPHY, col3_4 GEOGRAPHY NULL, col3_5 OID NOT NULL, col3_6 INTERVAL NOT NULL, col3_7 FLOAT4, col3_8 GEOMETRY, col3_9 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_5 IS NULL THEN e'\x00'::TEXT ELSE e'\U00002603'::TEXT END) STORED, col3_10 TEXT NULL GENERATED ALWAYS AS (lower(col3_1)) STORED, col3_11 TEXT NULL GENERATED ALWAYS AS (lower(col3_1)) STORED);
        test_1        |             CREATE INDEX ON table3 (col3_9 DESC, col3_2 DESC, col3_6) INCLUDE (col3_1, col3_3, col3_4, col3_5, col3_7, col3_8);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('ppeall', 'qz', 'ryqly');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('e', 'jep');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('k');
        test_1        |             CREATE TYPE rand_typ_3 AS ENUM ('lwrsnh', 'al', 'hrpw', 'qkvp');
        test_1        |             
        test_1        |         compare_test.go:150: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3273418&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3273418&tab=artifacts#/) on master @ [cab185ff71f0924953d987fe6ffd14efdd32a3a0](https://github.com/cockroachdb/cockroach/commits/cab185ff71f0924953d987fe6ffd14efdd32a3a0):


```
        test_1        | --- FAIL: TestCompare (3.12s)
        test_1        |     compare_test.go:116: Checking connection to: postgres
        test_1        |     compare_test.go:116: Checking connection to: cockroach1
        test_1        |     compare_test.go:116: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.00s)
        test_1        |         compare_test.go:125: starting test: postgres
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 24 [running]:
        test_1        | testing.tRunner.func1.2(0x1f97240, 0xc000bc7740)
        test_1        | 	/usr/local/go/src/testing/testing.go:1143 +0x332
        test_1        | testing.tRunner.func1(0xc0004c6a80)
        test_1        | 	/usr/local/go/src/testing/testing.go:1146 +0x4b6
        test_1        | panic(0x1f97240, 0xc000bc7740)
        test_1        | 	/usr/local/go/src/runtime/panic.go:965 +0x1b9
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x3fdff20, 0x3138b58, 0xc000bf7c20, 0x3138b58, 0xc000bf7c20, 0x6)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:368 +0x5d3
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0xc000bf60f0, 0x3fdff20, 0x3fdff20, 0xd50b044e21570b1, 0xc000100000, 0xd00000000000000)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:301 +0x146
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc000bdf200)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:220 +0x9d
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc000bf60f0, 0xc000bb1e00, 0x4, 0x4, 0x0, 0x0, 0x0, 0x0)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:251 +0xbe6
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0x2689f08, 0xc000bf60f0, 0xc000bb1e00, 0x4, 0x4, 0x0, 0xc00020a780, 0xe0, 0x0)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:97 +0x4e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0xc000bf60f0, 0x2129ca8, 0x5, 0x4, 0xc000c07a40, 0x3, 0x3, 0xe0, 0x4436d7, 0x8)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:94 +0x224
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0xc000bf60f0, 0x4, 0x3, 0x0)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:91 +0x19e
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc000bf60f0, 0x4394f8, 0x30)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:77 +0x4f
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0xc000bf60f0, 0xb8e46160daed7f78, 0x11)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:43 +0x44
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc0004c6a80)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:127 +0x113
        test_1        | testing.tRunner(0xc0004c6a80, 0xc000bf6090)
        test_1        | 	/usr/local/go/src/testing/testing.go:1193 +0xef
        test_1        | created by testing.(*T).Run
        test_1        | 	/usr/local/go/src/testing/testing.go:1238 +0x2b3
        compare_test_1 exited with code 2
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 2
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3286049&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3286049&tab=artifacts#/) on master @ [90809c048d05f923a67ce9b89597b2779fc73e32](https://github.com/cockroachdb/cockroach/commits/90809c048d05f923a67ce9b89597b2779fc73e32):


```
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_536576 NULLS FIRST,
        postgres_1    | 		col_536577 NULLS FIRST,
        postgres_1    | 		col_536578 NULLS FIRST,
        postgres_1    | 		col_536579 NULLS FIRST,
        postgres_1    | 		col_536580 NULLS FIRST,
        postgres_1    | 		col_536581 NULLS FIRST
        test_1        | --- FAIL: TestCompare (945.18s)
        test_1        |     compare_test.go:116: Checking connection to: postgres
        test_1        |     compare_test.go:116: Checking connection to: cockroach1
        test_1        |     compare_test.go:116: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (341.02s)
        test_1        |         compare_test.go:125: starting test: postgres
        test_1        |         compare_test.go:132: initializing connection: cockroach1
        test_1        |         compare_test.go:132: initializing connection: postgres
        test_1        |         compare_test.go:179: compare cockroach1 to postgres:
        test_1        |             unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	string("0.0349207694917477F"),
        test_1        |             + 	string("0.03492076949174774F"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT (tand(2.0:::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8 || 'F':::STRING::STRING::STRING)::STRING AS col_536817)
        test_1        |             		AS tab_357603
        test_1        |             ORDER BY
        test_1        |             	col_536817 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT (tand(2.0::FLOAT8::FLOAT8::FLOAT8)::FLOAT8::FLOAT8 || 'F'::TEXT::TEXT::TEXT)::TEXT AS col_536817)
        test_1        |             		AS tab_357603
        test_1        |             ORDER BY
        test_1        |             	col_536817 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3290923&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3290923&tab=artifacts#/) on master @ [e723a85739a2e1b514f5de3221a2177a3a37d0e6](https://github.com/cockroachdb/cockroach/commits/e723a85739a2e1b514f5de3221a2177a3a37d0e6):


```
        test_1        |     compare_test.go:116: Checking connection to: cockroach1
        test_1        |     compare_test.go:116: Checking connection to: cockroach2
        test_1        |     compare_test.go:116: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (83.69s)
        test_1        |         compare_test.go:125: starting test: postgres
        test_1        |         compare_test.go:132: initializing connection: cockroach1
        test_1        |         compare_test.go:132: initializing connection: postgres
        test_1        |         compare_test.go:179: compare postgres to cockroach1:
        test_1        |             unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	string("A2000-01-01 01:00:04+00"),
        test_1        |             + 	string("A2000-01-01 01:00:04+00:00"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			('A':::STRING::STRING::STRING || ('2000-01-01 01:00:00+00:00':::TIMESTAMPTZ::TIMESTAMPTZ::TIMESTAMPTZ + '00:00:04':::INTERVAL::INTERVAL::INTERVAL)::TIMESTAMPTZ::TIMESTAMPTZ)::STRING
        test_1        |             				AS col_121814
        test_1        |             	)
        test_1        |             		AS tab_76423
        test_1        |             ORDER BY
        test_1        |             	col_121814 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			('A'::TEXT::TEXT::TEXT || ('2000-01-01 01:00:00+00:00'::TIMESTAMPTZ::TIMESTAMPTZ::TIMESTAMPTZ + '00:00:04'::INTERVAL::INTERVAL::INTERVAL)::TIMESTAMPTZ::TIMESTAMPTZ)::TEXT
        test_1        |             				AS col_121814
        test_1        |             	)
        test_1        |             		AS tab_76423
        test_1        |             ORDER BY
        test_1        |             	col_121814 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3303321&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3303321&tab=artifacts#/) on master @ [ee3efd6b1e24a3e1676778f5028fa0a35266f683](https://github.com/cockroachdb/cockroach/commits/ee3efd6b1e24a3e1676778f5028fa0a35266f683):


```
        postgres_1    | 	CREATE UNIQUE INDEX ON table3 (col3_4, col3_1 ASC, col3_3 DESC, col3_0 DESC);
        postgres_1    | 	CREATE INDEX ON table3 (col3_1 ASC, col3_2 DESC, col3_3 ASC, col3_4 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table3 (col3_0 DESC, col3_4 ASC, col3_2);
        postgres_1    | 	CREATE TABLE table4 (col4_0 BIT(31) NOT NULL, col4_1 BYTEA, col4_2 BOX2D NOT NULL, PRIMARY KEY (col4_0), col4_3 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col4_2 AS TEXT))) STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (col4_3 ASC, col4_1);
        postgres_1    | 	CREATE INDEX ON table4 (col4_1, col4_0 ASC);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('jgiyp', 'dcc');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('somtp');
        postgres_1    | 	CREATE TYPE rand_typ_2 AS ENUM ('mk');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (607.70s)
        test_1        |     compare_test.go:116: Checking connection to: cockroach1
        test_1        |     compare_test.go:116: Checking connection to: cockroach2
        test_1        |     compare_test.go:116: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (4.75s)
        test_1        |         compare_test.go:125: starting test: postgres
        test_1        |         compare_test.go:132: initializing connection: cockroach1
        test_1        |         compare_test.go:132: initializing connection: postgres
        test_1        |         compare_test.go:149: CREATE TABLE table1 (col1_0 FLOAT8 NOT NULL, col1_1 REGPROCEDURE, col1_2 "char" NOT NULL, col1_3 OID NULL, col1_4 "char", col1_5 TEXT, col1_6 GEOMETRY, col1_7 BIT(17), col1_8 TEXT GENERATED ALWAYS AS (CASE WHEN col1_1 IS NULL THEN e'\x00'::TEXT ELSE '6T'::TEXT END) STORED, col1_9 TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_2)) STORED, col1_10 TEXT GENERATED ALWAYS AS (lower(col1_5)) STORED);
        test_1        |             CREATE INDEX ON table1 (col1_10 ASC) INCLUDE (col1_0, col1_2, col1_4, col1_5, col1_7, col1_9);
        test_1        |             CREATE TABLE table2 (col2_0 INET[] NOT NULL, col2_1 "char" NOT NULL, col2_2 TIMESTAMPTZ NOT NULL, col2_3 REGPROC, col2_4 TIMETZ, col2_5 OID NOT NULL, col2_6 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_0 IS NULL THEN 't'::TEXT ELSE '[*r^*H'::TEXT END) STORED, col2_7 TEXT GENERATED ALWAYS AS (CASE WHEN col2_3 IS NULL THEN e'\U00002603'::TEXT ELSE '"'::TEXT END) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_4 ASC, col2_5 DESC);
        test_1        |             CREATE INDEX ON table2 (col2_5 ASC, col2_7) INCLUDE (col2_0, col2_1, col2_3, col2_4);
        test_1        |             CREATE TABLE table3 (col3_0 TIMESTAMPTZ NOT NULL, col3_1 INT2 NOT NULL, col3_2 BYTEA NOT NULL, col3_3 NAME NOT NULL, PRIMARY KEY (col3_3, col3_1, col3_0, col3_2), col3_4 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_2 IS NULL THEN ''::TEXT ELSE e'l\x1a\x15\x10S\x16\x1a'::TEXT END) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_3, col3_4 DESC, col3_0 DESC);
        test_1        |             CREATE INDEX ON table3 (col3_2, col3_1 ASC, col3_4, col3_0 DESC);
        test_1        |             CREATE INDEX ON table3 (col3_0 DESC, col3_3 DESC) INCLUDE (col3_4);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_2 ASC, col3_3);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_4, col3_1 ASC, col3_3 DESC, col3_0 DESC);
        test_1        |             CREATE INDEX ON table3 (col3_1 ASC, col3_2 DESC, col3_3 ASC, col3_4 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_0 DESC, col3_4 ASC, col3_2);
        test_1        |             CREATE TABLE table4 (col4_0 BIT(31) NOT NULL, col4_1 BYTEA, col4_2 BOX2D NOT NULL, PRIMARY KEY (col4_0), col4_3 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col4_2 AS TEXT))) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_3 ASC, col4_1);
        test_1        |             CREATE INDEX ON table4 (col4_1, col4_0 ASC);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('jgiyp', 'dcc');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('somtp');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('mk');
        test_1        |             
        test_1        |         compare_test.go:150: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 1
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3311214&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3311214&tab=artifacts#/) on master @ [dd82053908203cf6d77c36c06a8280831bb93d57](https://github.com/cockroachdb/cockroach/commits/dd82053908203cf6d77c36c06a8280831bb93d57):


```
        postgres_1    | 
        postgres_1    | 2021-08-17 05:20:07.784 UTC [1] LOG:  starting PostgreSQL 13.4 (Debian 13.4-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
        postgres_1    | 2021-08-17 05:20:07.785 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2021-08-17 05:20:07.785 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2021-08-17 05:20:07.788 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2021-08-17 05:20:07.792 UTC [95] LOG:  database system was shut down at 2021-08-17 05:20:07 UTC
        postgres_1    | 2021-08-17 05:20:07.796 UTC [1] LOG:  database system is ready to accept connections
        test_1        | --- FAIL: TestCompare (603.70s)
        test_1        |     compare_test.go:116: Checking connection to: postgres
        test_1        |     compare_test.go:116: Checking connection to: cockroach1
        test_1        |     compare_test.go:116: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.01s)
        test_1        |         compare_test.go:125: starting test: postgres
        test_1        | panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        test_1        | 	panic: runtime error: invalid memory address or nil pointer dereference
        test_1        | [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19cd610]
        test_1        | 
        test_1        | goroutine 1240316 [running]:
        test_1        | testing.tRunner.func1.2(0x1eeb5a0, 0x4043df0)
        test_1        | 	/usr/local/go/src/testing/testing.go:1143 +0x332
        test_1        | testing.tRunner.func1(0xc000c9cc00)
        test_1        | 	/usr/local/go/src/testing/testing.go:1146 +0x4b6
        test_1        | panic(0x1eeb5a0, 0x4043df0)
        test_1        | 	/usr/local/go/src/runtime/panic.go:965 +0x1b9
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/types.(*T).Family(...)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/types/types.go:1130
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.postgresCreateTableMutator(0xc000792a50, 0xc0001d1680, 0x14, 0x14, 0x10, 0xc0005e2ad0, 0xc000341200, 0xc000300000)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:768 +0x9d0
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0x26b4f48, 0xc000792a50, 0xc0001d1680, 0x14, 0x14, 0x0, 0x1, 0x0, 0x1f03720)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:97 +0x4e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.Apply(0xc000792a50, 0xc0001d1680, 0x14, 0x14, 0xc0005e2ad0, 0x1, 0x1, 0x1, 0x0, 0x0, ...)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:108 +0xad
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.ApplyString(0xc000792a50, 0xc00036c000, 0x7a79, 0xc000763450, 0x1, 0x1, 0x308f6e0, 0xc000b964c0, 0xc0000cfbe0)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:159 +0x2e5
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc000c9cc00)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:128 +0x147
        test_1        | testing.tRunner(0xc000c9cc00, 0xc0007929c0)
        test_1        | 	/usr/local/go/src/testing/testing.go:1193 +0xef
        test_1        | created by testing.(*T).Run
        test_1        | 	/usr/local/go/src/testing/testing.go:1238 +0x2b3
        compare_test_1 exited with code 2
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:49: exit status 2
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestComposeCompare PKG=./pkg/compose TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3831155&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3831155&tab=artifacts#/) on master @ [2b8b0f64c63a376784926667f5fbffa4c4fcc60b](https://github.com/cockroachdb/cockroach/commits/2b8b0f64c63a376784926667f5fbffa4c4fcc60b):


```
        test_1        |             ORDER BY
        test_1        |             	col_864 NULLS FIRST;
        test_1        |             
        test_1        |     --- FAIL: TestCompare/mutators (464.25s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: cockroach2
        test_1        |         compare_test.go:193: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(tab_280862.crdb_internal_mvcc_timestamp::DECIMAL * tab_280858.crdb_internal_mvcc_timestamp::DECIMAL)::DECIMAL
        test_1        |             				AS col_424670,
        test_1        |             			tab_280861.col2_15 AS col_424671
        test_1        |             		FROM
        test_1        |             			postgres.public.table1@[0] AS tab_280856,
        test_1        |             			postgres.public.table2@[0] AS tab_280857
        test_1        |             			JOIN postgres.public.table3@[0] AS tab_280858
        test_1        |             				JOIN postgres.public.table1@[0] AS tab_280859
        test_1        |             					JOIN postgres.public.table1@[0] AS tab_280860 ON (tab_280859.col1_6) = (tab_280860.col1_1) ON
        test_1        |             						(tab_280858.tableoid) = (tab_280859.col1_3) AND (tab_280858.col3_0) = (tab_280859.col1_2)
        test_1        |             				JOIN postgres.public.table2@[0] AS tab_280861 ON
        test_1        |             						(tab_280860.col1_0) = (tab_280861.col2_8)
        test_1        |             						AND (tab_280860.col1_0) = (tab_280861.col2_4)
        test_1        |             						AND (tab_280859.tableoid) = (tab_280861.col2_10)
        test_1        |             				JOIN postgres.public.table3@[0] AS tab_280862 ON
        test_1        |             						(tab_280861.col2_10) = (tab_280862.tableoid)
        test_1        |             						AND (tab_280860.crdb_internal_mvcc_timestamp) = (tab_280862.crdb_internal_mvcc_timestamp)
        test_1        |             				JOIN postgres.public.table3@[0] AS tab_280863 ON
        test_1        |             						(tab_280860.col1_2) = (tab_280863.col3_0)
        test_1        |             						AND (tab_280859.tableoid) = (tab_280863.tableoid)
        test_1        |             						AND (tab_280859.col1_3) = (tab_280863.col3_0) ON
        test_1        |             					(tab_280857.col2_4) = (tab_280858.col3_0) AND (tab_280857.tableoid) = (tab_280861.col2_3)
        test_1        |             	)
        test_1        |             		AS tab_280864
        test_1        |             ORDER BY
        test_1        |             	col_424670 NULLS FIRST, col_424671 NULLS FIRST
        test_1        |         compare_test.go:194: cockroach1: ping: conn closed
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3847823&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3847823&tab=artifacts#/) on master @ [8b696d17241b2eb2d333e23750cf6fdc433897d8](https://github.com/cockroachdb/cockroach/commits/8b696d17241b2eb2d333e23750cf6fdc433897d8):


```
        postgres_1    | 2021-12-05 06:14:51.458 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2021-12-05 06:14:51.463 UTC [69] LOG:  database system was shut down at 2021-12-05 06:14:51 UTC
        postgres_1    | 2021-12-05 06:14:51.468 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2021-12-05 06:14:54.717 UTC [79] ERROR:  functions in index expression must be marked IMMUTABLE
        postgres_1    | 2021-12-05 06:14:54.717 UTC [79] STATEMENT:  CREATE TABLE table1 (col1_0 INTERVAL NOT NULL, col1_1 INTERVAL NOT NULL, col1_2 BOOL NOT NULL, col1_3 REGCLASS NOT NULL, col1_4 VARCHAR NOT NULL, col1_5 TEXT NOT NULL, col1_6 INET, col1_7 TIME NOT NULL, col1_8 BIT(35) NULL, col1_9 GEOMETRY NOT NULL, col1_10 BOX2D NOT NULL, col1_11 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, col1_12 TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_4)) STORED, col1_13 TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_5)) STORED, col1_14 TEXT GENERATED ALWAYS AS (lower(CAST(col1_6 AS TEXT))) STORED, col1_15 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_10 AS TEXT))) STORED);
        postgres_1    | 	CREATE INDEX ON table1 (lower(CAST(col1_9 AS TEXT)) DESC, col1_8, col1_7 ASC, col1_4, lower(CAST(col1_6 AS TEXT)) ASC, col1_3 ASC, col1_5 ASC, col1_12 ASC, col1_14 DESC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_0, col1_8 ASC, col1_15, col1_1 DESC, col1_4 ASC, col1_2 DESC, col1_7, col1_5 ASC, lower(CAST(col1_7 AS TEXT)), col1_13, col1_12 DESC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table1 (col1_3, col1_6 ASC, col1_15 ASC, col1_2, lower(CAST(col1_2 AS TEXT)) DESC, col1_0, col1_4 DESC, col1_11 DESC, col1_1, col1_12, col1_8 ASC, col1_14 DESC, col1_13 ASC) INCLUDE (col1_10);
        postgres_1    | 	CREATE TABLE table2 (col2_0 REGNAMESPACE NOT NULL, col2_1 BIT(42) NOT NULL, col2_2 BIT(30) NOT NULL, col2_3 TIMETZ NOT NULL, col2_4 BIT(20) NOT NULL, col2_5 INT8 NOT NULL, col2_6 TIMETZ NOT NULL, PRIMARY KEY (col2_4, col2_6, col2_5, col2_3, col2_1, col2_0, col2_2), col2_7 INT8 NOT NULL GENERATED ALWAYS AS (col2_5 + (-4387640342690645200)::INT8) STORED, col2_8 INT8 NOT NULL GENERATED ALWAYS AS (col2_5 + 2291379472140351265::INT8) STORED, col2_9 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_3 AS TEXT))) STORED, col2_10 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_3 AS TEXT))) STORED, col2_11 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_0 IS NULL THEN 'X'::TEXT ELSE ''::TEXT END) STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_0, col2_8 ASC, col2_3 ASC, col2_11, col2_6 DESC, col2_9 DESC, (col2_5 + 9033786886123119531::INT8) ASC) INCLUDE (col2_10);
        postgres_1    | 	CREATE INDEX ON table2 (col2_11 ASC, col2_4 ASC, col2_7, col2_1 ASC, col2_3 ASC, col2_9 DESC, col2_2 DESC, col2_6, col2_5 ASC, col2_8 ASC, col2_10 ASC, col2_0 DESC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_10 DESC, col2_5 ASC, col2_9, lower(CAST(col2_0 AS TEXT)) DESC, col2_4 DESC, col2_0 DESC);
        postgres_1    | 	CREATE INDEX ON table2 (col2_4, col2_6 DESC, col2_7 ASC, col2_3 ASC, col2_11 DESC, col2_10);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('qkv', 'wsmd', 'ra', 'ighus', 'xrvye');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('camrm', 'isysr', 'hikjcj', 'gn', 'kjbv', 'u');
        postgres_1    | 	CREATE TYPE rand_typ_2 AS ENUM ('e', 'uvose', 'v', 'u', 'urbme', 'bfsa');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.90s)
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (2.70s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 INTERVAL NOT NULL, col1_1 INTERVAL NOT NULL, col1_2 BOOL NOT NULL, col1_3 REGCLASS NOT NULL, col1_4 VARCHAR NOT NULL, col1_5 TEXT NOT NULL, col1_6 INET, col1_7 TIME NOT NULL, col1_8 BIT(35) NULL, col1_9 GEOMETRY NOT NULL, col1_10 BOX2D NOT NULL, col1_11 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, col1_12 TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_4)) STORED, col1_13 TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_5)) STORED, col1_14 TEXT GENERATED ALWAYS AS (lower(CAST(col1_6 AS TEXT))) STORED, col1_15 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_10 AS TEXT))) STORED);
        test_1        |             CREATE INDEX ON table1 (lower(CAST(col1_9 AS TEXT)) DESC, col1_8, col1_7 ASC, col1_4, lower(CAST(col1_6 AS TEXT)) ASC, col1_3 ASC, col1_5 ASC, col1_12 ASC, col1_14 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_0, col1_8 ASC, col1_15, col1_1 DESC, col1_4 ASC, col1_2 DESC, col1_7, col1_5 ASC, lower(CAST(col1_7 AS TEXT)), col1_13, col1_12 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_3, col1_6 ASC, col1_15 ASC, col1_2, lower(CAST(col1_2 AS TEXT)) DESC, col1_0, col1_4 DESC, col1_11 DESC, col1_1, col1_12, col1_8 ASC, col1_14 DESC, col1_13 ASC) INCLUDE (col1_10);
        test_1        |             CREATE TABLE table2 (col2_0 REGNAMESPACE NOT NULL, col2_1 BIT(42) NOT NULL, col2_2 BIT(30) NOT NULL, col2_3 TIMETZ NOT NULL, col2_4 BIT(20) NOT NULL, col2_5 INT8 NOT NULL, col2_6 TIMETZ NOT NULL, PRIMARY KEY (col2_4, col2_6, col2_5, col2_3, col2_1, col2_0, col2_2), col2_7 INT8 NOT NULL GENERATED ALWAYS AS (col2_5 + (-4387640342690645200)::INT8) STORED, col2_8 INT8 NOT NULL GENERATED ALWAYS AS (col2_5 + 2291379472140351265::INT8) STORED, col2_9 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_3 AS TEXT))) STORED, col2_10 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_3 AS TEXT))) STORED, col2_11 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_0 IS NULL THEN 'X'::TEXT ELSE ''::TEXT END) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_0, col2_8 ASC, col2_3 ASC, col2_11, col2_6 DESC, col2_9 DESC, (col2_5 + 9033786886123119531::INT8) ASC) INCLUDE (col2_10);
        test_1        |             CREATE INDEX ON table2 (col2_11 ASC, col2_4 ASC, col2_7, col2_1 ASC, col2_3 ASC, col2_9 DESC, col2_2 DESC, col2_6, col2_5 ASC, col2_8 ASC, col2_10 ASC, col2_0 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_10 DESC, col2_5 ASC, col2_9, lower(CAST(col2_0 AS TEXT)) DESC, col2_4 DESC, col2_0 DESC);
        test_1        |             CREATE INDEX ON table2 (col2_4, col2_6 DESC, col2_7 ASC, col2_3 ASC, col2_11 DESC, col2_10);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('qkv', 'wsmd', 'ra', 'ighus', 'xrvye');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('camrm', 'isysr', 'hikjcj', 'gn', 'kjbv', 'u');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('e', 'uvose', 'v', 'u', 'urbme', 'bfsa');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3851947&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3851947&tab=artifacts#/) on master @ [835ceca5d25d4a62233ddde4f493dbcf68302f1e](https://github.com/cockroachdb/cockroach/commits/835ceca5d25d4a62233ddde4f493dbcf68302f1e):


```
        postgres_1    | 	CREATE TABLE table5 (col5_0 VARCHAR, col5_1 REGTYPE NULL, col5_2 INET NULL, col5_3 BIT NULL, col5_4 REGTYPE NULL, col5_5 REGROLE NULL, col5_6 BOOL NOT NULL, col5_7 OID NOT NULL, col5_8 DATE NOT NULL, col5_9 BOOL NOT NULL, col5_10 FLOAT8 NULL, col5_11 TIMESTAMP NOT NULL, col5_12 REGTYPE, col5_13 TIME NOT NULL, PRIMARY KEY (col5_8, col5_11), col5_14 FLOAT8 NULL GENERATED ALWAYS AS (col5_10 + 0.9055480054642125::FLOAT8) STORED, col5_15 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col5_3 AS TEXT))) STORED, col5_16 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col5_11 IS NULL THEN e'\x7fh:Z\x02\x16'::TEXT ELSE e'X@=wFV^\x17'::TEXT END) STORED, col5_17 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col5_13 AS TEXT))) STORED, col5_18 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col5_1 IS NULL THEN e'G\x1f\bQt\x0e9m'::TEXT ELSE e'\x18D\x0f\x14+V'::TEXT END) STORED);
        postgres_1    | 	CREATE INDEX ON table5 (col5_6 DESC) INCLUDE (col5_0, col5_2, col5_4, col5_5, col5_10, col5_12);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('b', 'tcta', 'cmgcfr', 'hul', 'zqjdgn', 'yq');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('njf', 'lq');
        postgres_1    | 	CREATE TYPE rand_typ_2 AS ENUM ('bofh', 'olv', 'lq', 's');
        postgres_1    | 	CREATE TYPE rand_typ_3 AS ENUM ('sjlaiy', 'gdrud', 'l');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (607.75s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (4.68s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 BIT(5) NULL, col1_1 FLOAT4[], col1_2 REGROLE NULL, col1_3 INT2, col1_4 INT8 NULL, col1_5 OID NOT NULL, col1_6 BYTEA NULL, col1_7 REGNAMESPACE NULL, col1_8 BOOL NULL, col1_9 TIME NULL, col1_10 INT2, col1_11 INET, col1_12 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col1_0 AS TEXT))) STORED, col1_13 INT2 GENERATED ALWAYS AS ((col1_3 + col1_10) + col1_4) STORED, col1_14 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col1_8 AS TEXT))) STORED, col1_15 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col1_2 IS NULL THEN NULL ELSE e' \x13y/q\x0e'::TEXT END) STORED, col1_16 INT8 NULL GENERATED ALWAYS AS (col1_4 + col1_10) STORED);
        test_1        |             CREATE INDEX ON table1 (lower(CAST(col1_2 AS TEXT)), col1_13 ASC, col1_4 ASC, col1_0 ASC, col1_12, col1_11 ASC, col1_10, col1_2 ASC) INCLUDE (col1_1, col1_3, col1_5, col1_6, col1_7, col1_8, col1_9, col1_16);
        test_1        |             CREATE INDEX ON table1 (col1_2, (col1_3 + 24493::INT8) ASC, col1_5, col1_16, col1_8 DESC, col1_4 DESC, col1_11 ASC, col1_9 ASC, col1_3 DESC, col1_13, col1_0);
        test_1        |             CREATE TABLE table2 (col2_0 REGCLASS NULL, col2_1 REGCLASS NOT NULL, col2_2 GEOGRAPHY NOT NULL, col2_3 INTERVAL, col2_4 CHAR, col2_5 REGTYPE NULL, col2_6 REGCLASS NOT NULL, col2_7 NAME);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_4, col2_0 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_5 ASC, col2_0 DESC, col2_4 DESC, col2_6 DESC, col2_7 ASC) INCLUDE (col2_3);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_4, col2_3, col2_0 DESC);
        test_1        |             CREATE TABLE table3 (col3_0 REGCLASS NULL, col3_1 JSONB NULL, col3_2 INT4 NOT NULL, col3_3 FLOAT8, col3_4 INT2, col3_5 FLOAT4, col3_6 REGCLASS, col3_7 INT4 GENERATED ALWAYS AS (col3_2 + col3_4) STORED, col3_8 FLOAT8 GENERATED ALWAYS AS (col3_3 + 0.21622949857702678::FLOAT8) STORED, col3_9 INT2 GENERATED ALWAYS AS (col3_4 + col3_2) STORED, col3_10 FLOAT4 GENERATED ALWAYS AS (col3_5 + col3_3) STORED, col3_11 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col3_0 IS NULL THEN '8+'::TEXT ELSE ''::TEXT END) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_5 ASC, col3_4 ASC, col3_7, col3_3 DESC) INCLUDE (col3_0, col3_1, col3_6, col3_11);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_3, col3_6 DESC) INCLUDE (col3_4, col3_5, col3_11);
        test_1        |             CREATE TABLE table4 (col4_0 BIT NULL, col4_1 BIT(29) NOT NULL, PRIMARY KEY (col4_1), col4_2 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col4_1 AS TEXT))) STORED);
        test_1        |             CREATE INDEX ON table4 (col4_1) INCLUDE (col4_0);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_1 DESC, col4_2, col4_0);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_1 DESC, col4_0, col4_2);
        test_1        |             CREATE INDEX ON table4 (col4_0 DESC);
        test_1        |             CREATE INDEX ON table4 (col4_2 ASC) INCLUDE (col4_0);
        test_1        |             CREATE TABLE table5 (col5_0 VARCHAR, col5_1 REGTYPE NULL, col5_2 INET NULL, col5_3 BIT NULL, col5_4 REGTYPE NULL, col5_5 REGROLE NULL, col5_6 BOOL NOT NULL, col5_7 OID NOT NULL, col5_8 DATE NOT NULL, col5_9 BOOL NOT NULL, col5_10 FLOAT8 NULL, col5_11 TIMESTAMP NOT NULL, col5_12 REGTYPE, col5_13 TIME NOT NULL, PRIMARY KEY (col5_8, col5_11), col5_14 FLOAT8 NULL GENERATED ALWAYS AS (col5_10 + 0.9055480054642125::FLOAT8) STORED, col5_15 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col5_3 AS TEXT))) STORED, col5_16 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col5_11 IS NULL THEN e'\x7fh:Z\x02\x16'::TEXT ELSE e'X@=wFV^\x17'::TEXT END) STORED, col5_17 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col5_13 AS TEXT))) STORED, col5_18 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col5_1 IS NULL THEN e'G\x1f\bQt\x0e9m'::TEXT ELSE e'\x18D\x0f\x14+V'::TEXT END) STORED);
        test_1        |             CREATE INDEX ON table5 (col5_6 DESC) INCLUDE (col5_0, col5_2, col5_4, col5_5, col5_10, col5_12);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('b', 'tcta', 'cmgcfr', 'hul', 'zqjdgn', 'yq');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('njf', 'lq');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('bofh', 'olv', 'lq', 's');
        test_1        |             CREATE TYPE rand_typ_3 AS ENUM ('sjlaiy', 'gdrud', 'l');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3882331&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3882331&tab=artifacts#/) on master @ [884088d64d0859d72e42ee4fde1cf65192f1cb99](https://github.com/cockroachdb/cockroach/commits/884088d64d0859d72e42ee4fde1cf65192f1cb99):


```
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (7.40s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 BOOL NOT NULL, col1_1 DATE NOT NULL, col1_2 INT2 NOT NULL, col1_3 TEXT NOT NULL, col1_4 BOX2D[], col1_5 BIT(4) NOT NULL, PRIMARY KEY (col1_5, col1_1, col1_0, col1_2), col1_6 TEXT GENERATED ALWAYS AS (CASE WHEN col1_4 IS NULL THEN e'-196{O\x10\x1c\x19'::TEXT ELSE 'St'::TEXT END) STORED);
        test_1        |             CREATE INDEX ON table1 (col1_6, lower(CAST(col1_1 AS TEXT)), col1_2, (col1_2 + (-9485)::INT8), col1_1, col1_5 ASC) INCLUDE (col1_4);
        test_1        |             CREATE INDEX ON table1 (col1_3 ASC);
        test_1        |             CREATE INDEX ON table1 (col1_3 ASC) INCLUDE (col1_6);
        test_1        |             CREATE INDEX ON table1 (col1_6 ASC, col1_2 DESC) INCLUDE (col1_3, col1_4);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_5);
        test_1        |             CREATE TABLE table2 (col2_0 BIT(46) NOT NULL, col2_1 REGROLE NOT NULL, col2_2 BIT(49) NOT NULL, col2_3 REGROLE NOT NULL, col2_4 REGPROCEDURE NOT NULL, PRIMARY KEY (col2_1, col2_0, col2_2, col2_4, col2_3));
        test_1        |             CREATE TABLE table3 (col3_0 INET NOT NULL, col3_1 TIMETZ NOT NULL, col3_2 BIT(35) NOT NULL, col3_3 JSONB NOT NULL, col3_4 BIT(20) NOT NULL, col3_5 DATE NOT NULL, col3_6 BOOL, PRIMARY KEY (col3_2, col3_0, col3_4, col3_1), col3_7 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_3 AS TEXT))) STORED, col3_8 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_4 AS TEXT))) STORED, col3_9 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_1 AS TEXT))) STORED);
        test_1        |             CREATE INDEX ON table3 (col3_4, col3_2) INCLUDE (col3_3, col3_5, col3_8);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_6);
        test_1        |             CREATE TABLE table4 (col4_0 FLOAT8 NOT NULL, col4_1 REGROLE NOT NULL, col4_2 TEXT NOT NULL, PRIMARY KEY (col4_2, col4_0));
        test_1        |             CREATE INDEX ON table4 (col4_1 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_1 DESC, col4_2 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_1 DESC, col4_2 ASC);
        test_1        |             CREATE INDEX ON table4 (col4_0) INCLUDE (col4_1);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_0 ASC, col4_2 DESC, col4_1 ASC);
        test_1        |             CREATE INDEX ON table4 (col4_2, col4_1 ASC, col4_0);
        test_1        |             CREATE INDEX ON table4 (col4_1 DESC, col4_2, col4_0 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_2 ASC, col4_0 ASC);
        test_1        |             CREATE INDEX ON table4 (col4_1, col4_2 DESC, col4_0);
        test_1        |             CREATE TABLE table5 (col5_0 TEXT NULL, col5_1 INTERVAL NOT NULL, col5_2 BIT(40) NOT NULL, col5_3 VARCHAR NOT NULL, col5_4 INT4 NULL, col5_5 BOOL NOT NULL, col5_6 INT4 NOT NULL, PRIMARY KEY (col5_1, col5_6, col5_3), col5_7 INT4 NOT NULL GENERATED ALWAYS AS (col5_6 + 1187101097::INT8) STORED, col5_8 TEXT NULL GENERATED ALWAYS AS (lower(col5_0)) STORED, col5_9 INT4 NULL GENERATED ALWAYS AS (col5_4 + col5_6) STORED, col5_10 INT4 NULL GENERATED ALWAYS AS (col5_6 + col5_4) STORED, col5_11 INT4 NULL GENERATED ALWAYS AS (col5_4 + col5_6) STORED, col5_12 INT4 NULL GENERATED ALWAYS AS (col5_6 + col5_4) STORED);
        test_1        |             CREATE INDEX ON table5 (col5_7 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table5 ((col5_6 + col5_4), col5_4 ASC, col5_2 ASC, col5_0 DESC, col5_8 DESC, col5_7);
        test_1        |             CREATE UNIQUE INDEX ON table5 ((col5_4 + col5_6) DESC, col5_11 ASC, col5_10, col5_4, col5_8, col5_12 ASC, col5_9 DESC, lower(CAST(col5_1 AS TEXT)), col5_1 DESC, col5_2 DESC, col5_3 DESC, col5_5, col5_7);
        test_1        |             CREATE INDEX ON table5 (col5_5 ASC, col5_11 DESC, col5_1) INCLUDE (col5_7, col5_9);
        test_1        |             CREATE INDEX ON table5 ((col5_4 + (-852429796)::INT8), col5_8, col5_4 ASC) INCLUDE (col5_0, col5_2, col5_9, col5_12);
        test_1        |             ALTER TABLE table1 ADD FOREIGN KEY (col1_0) REFERENCES table3 (col3_6);
        test_1        |             ALTER TABLE table5 ADD FOREIGN KEY (col5_2) REFERENCES table1 (col1_5) ON UPDATE CASCADE;
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('mnughf', 'oex', 'icscid', 'motn', 'lmmo');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('l', 'uemg', 'jrcqw', 'rhy', 'elbrz');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3889445&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3889445&tab=artifacts#/) on master @ [406e0fe1e44cabc8a3384703c642469567c6915e](https://github.com/cockroachdb/cockroach/commits/406e0fe1e44cabc8a3384703c642469567c6915e):


```
        test_1        | --- FAIL: TestCompare (2.13s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.00s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 50 [running]:
        test_1        | testing.tRunner.func1.2({0x1bc5220, 0xc000bce220})
        test_1        | 	/usr/local/go/src/testing/testing.go:1209 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	/usr/local/go/src/testing/testing.go:1212 +0x218
        test_1        | panic({0x1bc5220, 0xc000bce220})
        test_1        | 	/usr/local/go/src/runtime/panic.go:1038 +0x215
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x3dd2420, {0x2dd5e40, 0xc000bcd7a0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:368 +0x487
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x2cb0780, 0x3dd2420)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:301 +0x14c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc000baf080)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:220 +0x92
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc00099f050, {0xc00099f080, 0x3, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:251 +0xa4d
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0x90, 0x1d2a941, {0xc00099f080, 0x8e, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:97 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x203000, {0x1d2a941, 0x5}, 0x3, {0xc000a55a30, 0x3, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:85 +0x157
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x2cc5460, 0xc00001c000)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:89 +0x187
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0x790aee4f2090a22d)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:76 +0x33
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x1d2fd9f)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:46 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc0000cb040)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:128 +0xe5
        test_1        | testing.tRunner(0xc0000cb040, 0xc00099efc0)
        test_1        | 	/usr/local/go/src/testing/testing.go:1259 +0x102
        test_1        | created by testing.(*T).Run
        test_1        | 	/usr/local/go/src/testing/testing.go:1306 +0x35a
        compare_test_1 exited with code 2
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 2
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3894081&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3894081&tab=artifacts#/) on master @ [22df0a6658a927e7b65dafbdfc9d790500791993](https://github.com/cockroachdb/cockroach/commits/22df0a6658a927e7b65dafbdfc9d790500791993):


```
        postgres_1    | 	CREATE INDEX ON table2 (col2_3 DESC, col2_4 DESC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_10 ASC, col2_1 ASC, col2_12 DESC, lower(CAST(col2_4 AS TEXT))) INCLUDE (col2_0, col2_2, col2_4, col2_6);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_9 ASC, col2_8 DESC, col2_10 DESC, lower(CAST(col2_4 AS TEXT)) DESC, col2_11, col2_2 ASC, col2_3, col2_6 ASC, col2_5 ASC) INCLUDE (col2_0, col2_7, col2_12);
        postgres_1    | 	CREATE TABLE table3 (col3_0 TIMESTAMPTZ[], col3_1 BIT(7), col3_2 BOOL NULL, col3_3 INET NULL, col3_4 "char" NOT NULL, col3_5 INT8 NULL, col3_6 REGPROC, col3_7 REGTYPE NOT NULL, col3_8 BYTEA, col3_9 BIT(34), col3_10 NAME NULL, col3_11 "char" NOT NULL, col3_12 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col3_3 AS TEXT))) STORED, col3_13 TEXT NOT NULL GENERATED ALWAYS AS (lower(col3_11)) STORED, col3_14 TEXT NOT NULL GENERATED ALWAYS AS (lower(col3_4)) STORED, col3_15 TEXT NULL GENERATED ALWAYS AS (lower(col3_10)) STORED);
        postgres_1    | 	CREATE TABLE table4 (col4_0 INT2 NOT NULL, col4_1 NAME, col4_2 NAME, col4_3 BOX2D, col4_4 TIMESTAMP NOT NULL, col4_5 BIT(13) NULL, col4_6 GEOMETRY, col4_7 DATE NOT NULL, col4_8 GEOGRAPHY, PRIMARY KEY (col4_7, col4_4, col4_0), col4_9 TEXT GENERATED ALWAYS AS (lower(CAST(col4_8 AS TEXT))) STORED, col4_10 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col4_4 IS NULL THEN e'v&l\x05'::TEXT ELSE e'\x11M^B\tQ'::TEXT END) STORED, col4_11 INT2 NOT NULL GENERATED ALWAYS AS (col4_0 + (-13147)::INT8) STORED, col4_12 TEXT GENERATED ALWAYS AS (lower(CAST(col4_3 AS TEXT))) STORED);
        postgres_1    | 	CREATE TABLE table5 (col5_0 INTERVAL NULL, col5_1 "char" NOT NULL, col5_2 TIMETZ NULL, col5_3 NAME, col5_4 GEOMETRY, col5_5 TEXT, col5_6 DATE NOT NULL, col5_7 BOX2D NOT NULL, col5_8 REGCLASS, col5_9 REGPROC NULL, col5_10 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col5_0 AS TEXT))) STORED, col5_11 TEXT GENERATED ALWAYS AS (lower(col5_5)) STORED, col5_12 TEXT GENERATED ALWAYS AS (lower(CAST(col5_4 AS TEXT))) STORED, col5_13 TEXT GENERATED ALWAYS AS (lower(CAST(col5_4 AS TEXT))) STORED, col5_14 TEXT GENERATED ALWAYS AS (CASE WHEN col5_8 IS NULL THEN ''::TEXT ELSE e'\x0b\x7f,X\x0f\fJ'::TEXT END) STORED, col5_15 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col5_9 IS NULL THEN NULL ELSE ''::TEXT END) STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON table5 (lower(CAST(col5_7 AS TEXT)) ASC, col5_13 ASC, col5_2, col5_6, col5_3 ASC, col5_11, col5_10) INCLUDE (col5_0, col5_12, col5_15);
        postgres_1    | 	CREATE INDEX ON table5 (col5_1 DESC, col5_12 ASC, lower(CAST(col5_2 AS TEXT)) ASC, col5_9, col5_10, lower(CAST(col5_8 AS TEXT)), col5_11 ASC, col5_0 ASC, col5_15, col5_6 ASC, col5_5, col5_13 DESC, lower(CAST(col5_6 AS TEXT)) ASC, col5_14 ASC) INCLUDE (col5_3, col5_8);
        postgres_1    | 	CREATE UNIQUE INDEX ON table5 (col5_12 DESC, col5_15, col5_5 ASC, col5_9 DESC, col5_2 ASC, col5_0 DESC);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('a', 'sbuzgk', 'ecvzl', 'betot');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('wg', 'tllis', 'drbi', 'vg', 'wrr', 'sv');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.69s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (3.48s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 TIMESTAMPTZ NOT NULL, col1_1 "char" NOT NULL, col1_2 INT4 NOT NULL, col1_3 INET NULL, col1_4 JSONB NOT NULL, col1_5 TIMESTAMPTZ NOT NULL, col1_6 REGTYPE, col1_7 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col1_3 AS TEXT))) STORED, col1_8 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_5 IS NULL THEN 'Bu'::TEXT ELSE e'\x11]|\x1eTn\x01\x16\x01'::TEXT END) STORED, col1_9 TEXT NOT NULL GENERATED ALWAYS AS (lower(col1_1)) STORED);
        test_1        |             CREATE INDEX ON table1 (col1_2, (CASE WHEN col1_5 IS NULL THEN e'\x0fTt\x1aOO'::TEXT ELSE e'\x0eM8\x1e\x1c'::TEXT END), col1_9 ASC, lower(CAST(col1_6 AS TEXT)) ASC, col1_0, col1_8) INCLUDE (col1_1, col1_3, col1_5, col1_6);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_1 DESC) INCLUDE (col1_0, col1_6);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_3 ASC, col1_0 DESC, col1_6, col1_9, col1_2 DESC, col1_8, col1_5 ASC) INCLUDE (col1_4);
        test_1        |             CREATE TABLE table2 (col2_0 FLOAT8 NOT NULL, col2_1 REGCLASS NOT NULL, col2_2 INT2, col2_3 TIMESTAMPTZ NOT NULL, col2_4 BIT(38) NOT NULL, col2_5 REGROLE, col2_6 INET NOT NULL, col2_7 INT4 NOT NULL, PRIMARY KEY (col2_3), col2_8 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_6 AS TEXT))) STORED, col2_9 INT2 GENERATED ALWAYS AS (col2_2 + col2_7) STORED, col2_10 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_6 AS TEXT))) STORED, col2_11 INT4 GENERATED ALWAYS AS (col2_7 + col2_2) STORED, col2_12 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_1 IS NULL THEN '~-_'::TEXT ELSE e'd!\x02h'::TEXT END) STORED);
        test_1        |             CREATE INDEX ON table2 (col2_8, col2_7, col2_1 DESC, col2_5, col2_11 ASC);
        test_1        |             CREATE INDEX ON table2 (col2_3 DESC, col2_4 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_10 ASC, col2_1 ASC, col2_12 DESC, lower(CAST(col2_4 AS TEXT))) INCLUDE (col2_0, col2_2, col2_4, col2_6);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_9 ASC, col2_8 DESC, col2_10 DESC, lower(CAST(col2_4 AS TEXT)) DESC, col2_11, col2_2 ASC, col2_3, col2_6 ASC, col2_5 ASC) INCLUDE (col2_0, col2_7, col2_12);
        test_1        |             CREATE TABLE table3 (col3_0 TIMESTAMPTZ[], col3_1 BIT(7), col3_2 BOOL NULL, col3_3 INET NULL, col3_4 "char" NOT NULL, col3_5 INT8 NULL, col3_6 REGPROC, col3_7 REGTYPE NOT NULL, col3_8 BYTEA, col3_9 BIT(34), col3_10 NAME NULL, col3_11 "char" NOT NULL, col3_12 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col3_3 AS TEXT))) STORED, col3_13 TEXT NOT NULL GENERATED ALWAYS AS (lower(col3_11)) STORED, col3_14 TEXT NOT NULL GENERATED ALWAYS AS (lower(col3_4)) STORED, col3_15 TEXT NULL GENERATED ALWAYS AS (lower(col3_10)) STORED);
        test_1        |             CREATE TABLE table4 (col4_0 INT2 NOT NULL, col4_1 NAME, col4_2 NAME, col4_3 BOX2D, col4_4 TIMESTAMP NOT NULL, col4_5 BIT(13) NULL, col4_6 GEOMETRY, col4_7 DATE NOT NULL, col4_8 GEOGRAPHY, PRIMARY KEY (col4_7, col4_4, col4_0), col4_9 TEXT GENERATED ALWAYS AS (lower(CAST(col4_8 AS TEXT))) STORED, col4_10 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col4_4 IS NULL THEN e'v&l\x05'::TEXT ELSE e'\x11M^B\tQ'::TEXT END) STORED, col4_11 INT2 NOT NULL GENERATED ALWAYS AS (col4_0 + (-13147)::INT8) STORED, col4_12 TEXT GENERATED ALWAYS AS (lower(CAST(col4_3 AS TEXT))) STORED);
        test_1        |             CREATE TABLE table5 (col5_0 INTERVAL NULL, col5_1 "char" NOT NULL, col5_2 TIMETZ NULL, col5_3 NAME, col5_4 GEOMETRY, col5_5 TEXT, col5_6 DATE NOT NULL, col5_7 BOX2D NOT NULL, col5_8 REGCLASS, col5_9 REGPROC NULL, col5_10 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col5_0 AS TEXT))) STORED, col5_11 TEXT GENERATED ALWAYS AS (lower(col5_5)) STORED, col5_12 TEXT GENERATED ALWAYS AS (lower(CAST(col5_4 AS TEXT))) STORED, col5_13 TEXT GENERATED ALWAYS AS (lower(CAST(col5_4 AS TEXT))) STORED, col5_14 TEXT GENERATED ALWAYS AS (CASE WHEN col5_8 IS NULL THEN ''::TEXT ELSE e'\x0b\x7f,X\x0f\fJ'::TEXT END) STORED, col5_15 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col5_9 IS NULL THEN NULL ELSE ''::TEXT END) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table5 (lower(CAST(col5_7 AS TEXT)) ASC, col5_13 ASC, col5_2, col5_6, col5_3 ASC, col5_11, col5_10) INCLUDE (col5_0, col5_12, col5_15);
        test_1        |             CREATE INDEX ON table5 (col5_1 DESC, col5_12 ASC, lower(CAST(col5_2 AS TEXT)) ASC, col5_9, col5_10, lower(CAST(col5_8 AS TEXT)), col5_11 ASC, col5_0 ASC, col5_15, col5_6 ASC, col5_5, col5_13 DESC, lower(CAST(col5_6 AS TEXT)) ASC, col5_14 ASC) INCLUDE (col5_3, col5_8);
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_12 DESC, col5_15, col5_5 ASC, col5_9 DESC, col5_2 ASC, col5_0 DESC);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('a', 'sbuzgk', 'ecvzl', 'betot');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('wg', 'tllis', 'drbi', 'vg', 'wrr', 'sv');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3904070&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3904070&tab=artifacts#/) on master @ [5cd5be83218475ef0cc3c62a856b8d9d22cf142b](https://github.com/cockroachdb/cockroach/commits/5cd5be83218475ef0cc3c62a856b8d9d22cf142b):


```
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (col4_10 ASC, col4_7 ASC, col4_5 ASC, col4_11, col4_12 DESC, (CASE WHEN col4_6 IS NULL THEN e'\x00'::TEXT ELSE '<'::TEXT END) ASC) INCLUDE (col4_0, col4_2, col4_6);
        postgres_1    | 	CREATE TABLE table5 (col5_0 TEXT NOT NULL, col5_1 FLOAT4 NOT NULL, col5_2 BOX2D NOT NULL, col5_3 GEOMETRY NOT NULL, col5_4 FLOAT8 NOT NULL, col5_5 DECIMAL NOT NULL, col5_6 INT8 NULL, col5_7 INT4, col5_8 "char" NOT NULL, PRIMARY KEY (col5_0, col5_4), col5_9 TEXT NOT NULL GENERATED ALWAYS AS (lower(col5_8)) STORED, col5_10 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col5_3 AS TEXT))) STORED, col5_11 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col5_2 AS TEXT))) STORED, col5_12 FLOAT4 NOT NULL GENERATED ALWAYS AS (col5_1 + col5_4) STORED);
        postgres_1    | 	CREATE INDEX ON table5 (col5_10 DESC, col5_9 ASC, col5_4 DESC, col5_8 DESC, (col5_6 + 32767::INT8) DESC, col5_5 DESC, col5_6 DESC, col5_0 DESC) INCLUDE (col5_3);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('m', 'q');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('b', 'ca', 'osiiun', 'ghkku', 'fgdb', 'rcucj');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (608.13s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (5.41s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 JSONB NULL, col1_1 FLOAT4, col1_2 REGCLASS NULL, col1_3 OID NOT NULL, col1_4 REGPROC, col1_5 TIMESTAMPTZ, col1_6 "char" NOT NULL, col1_7 NAME NULL, col1_8 "char", col1_9 JSONB, col1_10 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col1_0 AS TEXT))) STORED, col1_11 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_3 IS NULL THEN e'\x01"X\x038z0\x05'::TEXT ELSE e'TNF\x03k\x04'::TEXT END) STORED);
        test_1        |             CREATE INDEX ON table1 (col1_2 ASC, col1_4, lower(col1_8) ASC, col1_10 ASC, col1_6) INCLUDE (col1_1, col1_5, col1_7, col1_8);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_4 DESC, col1_10 ASC, col1_8 DESC, col1_11 DESC);
        test_1        |             CREATE INDEX ON table1 (col1_7, col1_5 ASC, lower(CAST(col1_0 AS TEXT)), col1_3 DESC, col1_6 DESC, lower(col1_7) ASC, col1_8, lower(col1_6), lower(col1_6) DESC, col1_2 DESC) INCLUDE (col1_0, col1_4);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_4, col1_10, col1_11, col1_8);
        test_1        |             CREATE TABLE table2 (col2_0 REGTYPE, col2_1 BYTEA NULL, col2_2 VARCHAR NOT NULL, col2_3 FLOAT8[] NULL, col2_4 TIMESTAMP NULL, col2_5 NAME, col2_6 BIT(19), col2_7 TEXT, col2_8 TIMETZ, col2_9 JSONB NULL, col2_10 BIT(2), col2_11 REGNAMESPACE NULL, col2_12 TIMETZ NOT NULL, col2_13 CHAR NOT NULL, col2_14 REGPROCEDURE NULL, col2_15 BYTEA NOT NULL, col2_16 BIT(22) NOT NULL, col2_17 TIMESTAMPTZ NOT NULL);
        test_1        |             CREATE INDEX ON table2 (lower(CAST(col2_11 AS TEXT)), col2_0, col2_4 DESC, (CASE WHEN col2_1 IS NULL THEN e'j\x0f\x17dP{'::TEXT ELSE ''::TEXT END) ASC, lower(col2_7) DESC, col2_5, col2_8 ASC);
        test_1        |             CREATE TABLE table3 (col3_0 BYTEA NOT NULL, col3_1 REGNAMESPACE NOT NULL, col3_2 BIT NOT NULL, col3_3 INET NOT NULL, col3_4 REGPROC NOT NULL, col3_5 BOOL NOT NULL, col3_6 VARCHAR NOT NULL, col3_7 INT8 NOT NULL, PRIMARY KEY (col3_4, col3_7, col3_3, col3_0, col3_2, col3_6, col3_1, col3_5), col3_8 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_1 IS NULL THEN e'\x16v\x0eyTO4?\x16'::TEXT ELSE e'&#N\x13*6_\x14i'::TEXT END) STORED, col3_9 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_0 IS NULL THEN e'l\x01L'::TEXT ELSE e'4n<\x19(cK'::TEXT END) STORED, col3_10 INT8 NOT NULL GENERATED ALWAYS AS (col3_7 + (-2147483648)::INT8) STORED);
        test_1        |             CREATE INDEX ON table3 (col3_10 DESC, lower(CAST(col3_1 AS TEXT)), col3_0 DESC, col3_8 DESC, col3_1 ASC, col3_2 ASC, col3_5, col3_6 DESC, col3_9 ASC, col3_3 ASC, col3_4 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table3 (lower(CAST(col3_5 AS TEXT)) DESC, col3_10, lower(col3_6) ASC, col3_9 ASC, col3_8 DESC, col3_5, col3_1 ASC, col3_6 DESC);
        test_1        |             CREATE INDEX ON table3 (col3_3 DESC, col3_6 ASC, col3_5 DESC, col3_9 DESC, col3_8 ASC, col3_2, lower(CAST(col3_4 AS TEXT)) ASC, col3_4 DESC, col3_1 ASC, col3_10 ASC);
        test_1        |             CREATE INDEX ON table3 (col3_7 ASC, col3_2, col3_8 DESC, col3_5, (col3_7 + 6866396345441906187::INT8) ASC, col3_3 ASC, col3_4 ASC, col3_0 ASC, lower(col3_6) DESC, col3_10);
        test_1        |             CREATE INDEX ON table3 (col3_9 DESC, col3_0 DESC);
        test_1        |             CREATE INDEX ON table3 (col3_0 ASC, col3_3, col3_6 DESC, col3_9 DESC, col3_10 ASC, col3_1, col3_8 DESC);
        test_1        |             CREATE INDEX ON table3 (col3_9 ASC, lower(CAST(col3_5 AS TEXT)) DESC, col3_2 ASC, col3_8 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_6, col3_1 DESC, col3_3 DESC, col3_10, lower(CAST(col3_5 AS TEXT)) DESC, col3_9, col3_7);
        test_1        |             CREATE TABLE table4 (col4_0 GEOGRAPHY, col4_1 INT2 NULL, col4_2 VARCHAR NULL, col4_3 BIT(27), col4_4 DATE[], col4_5 OID, col4_6 REGCLASS, col4_7 REGPROC NULL, col4_8 TEXT GENERATED ALWAYS AS (lower(CAST(col4_3 AS TEXT))) STORED, col4_9 TEXT GENERATED ALWAYS AS (lower(CAST(col4_3 AS TEXT))) STORED, col4_10 INT2 NULL GENERATED ALWAYS AS (col4_1 + (-24655)::INT8) STORED, col4_11 TEXT GENERATED ALWAYS AS (CASE WHEN col4_6 IS NULL THEN e',I\x18JE\x02K-r'::TEXT ELSE NULL END) STORED, col4_12 TEXT GENERATED ALWAYS AS (CASE WHEN col4_6 IS NULL THEN e'\x1d\\\x17\x0b\x13\x1filt'::TEXT ELSE 't>!^~'::TEXT END) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_7, (CASE WHEN col4_5 IS NULL THEN e'%n^xf%\x04\x1c\x02'::TEXT ELSE e'y;\x11Y\x04_'::TEXT END) ASC, col4_9 DESC, col4_3 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_10 ASC, col4_7 ASC, col4_5 ASC, col4_11, col4_12 DESC, (CASE WHEN col4_6 IS NULL THEN e'\x00'::TEXT ELSE '<'::TEXT END) ASC) INCLUDE (col4_0, col4_2, col4_6);
        test_1        |             CREATE TABLE table5 (col5_0 TEXT NOT NULL, col5_1 FLOAT4 NOT NULL, col5_2 BOX2D NOT NULL, col5_3 GEOMETRY NOT NULL, col5_4 FLOAT8 NOT NULL, col5_5 DECIMAL NOT NULL, col5_6 INT8 NULL, col5_7 INT4, col5_8 "char" NOT NULL, PRIMARY KEY (col5_0, col5_4), col5_9 TEXT NOT NULL GENERATED ALWAYS AS (lower(col5_8)) STORED, col5_10 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col5_3 AS TEXT))) STORED, col5_11 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col5_2 AS TEXT))) STORED, col5_12 FLOAT4 NOT NULL GENERATED ALWAYS AS (col5_1 + col5_4) STORED);
        test_1        |             CREATE INDEX ON table5 (col5_10 DESC, col5_9 ASC, col5_4 DESC, col5_8 DESC, (col5_6 + 32767::INT8) DESC, col5_5 DESC, col5_6 DESC, col5_0 DESC) INCLUDE (col5_3);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('m', 'q');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('b', 'ca', 'osiiun', 'ghkku', 'fgdb', 'rcucj');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3914505&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3914505&tab=artifacts#/) on master @ [acf40e97f64a97338cbebd20d7fbe176c4e37c34](https://github.com/cockroachdb/cockroach/commits/acf40e97f64a97338cbebd20d7fbe176c4e37c34):


```
        postgres_1    | 			ORDER BY
        postgres_1    | 				tab_199901.col4_3 DESC
        postgres_1    | 		)
        postgres_1    | 			AS tab_199904
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_360071 NULLS FIRST,
        postgres_1    | 		col_360072 NULLS FIRST,
        postgres_1    | 		col_360073 NULLS FIRST,
        postgres_1    | 		col_360074 NULLS FIRST,
        postgres_1    | 		col_360075 NULLS FIRST
        test_1        | --- FAIL: TestCompare (883.04s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (280.08s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	nil,
        test_1        |             + 	string("BOX(-180 45,-135 90)"),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_box2dfromgeohash('B'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::BOX2D AS col_360366) AS tab_200070
        test_1        |             ORDER BY
        test_1        |             	CAST(col_360366 AS TEXT) NULLS FIRST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_box2dfromgeohash('B':::STRING::STRING::STRING, NULL::INT4::INT4)::BOX2D AS col_360366) AS tab_200070
        test_1        |             ORDER BY
        test_1        |             	CAST(col_360366 AS STRING) NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3924287&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3924287&tab=artifacts#/) on master @ [dd7fd4e5a371ce4437fce383b9674cba2e212fe6](https://github.com/cockroachdb/cockroach/commits/dd7fd4e5a371ce4437fce383b9674cba2e212fe6):


```
        test_1        |             		AS tab_93764
        test_1        |             ORDER BY
        test_1        |             	col_172518 NULLS FIRST, col_172519 NULLS FIRST;
        test_1        |             
        test_1        |     --- FAIL: TestCompare/mutators (0.00s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 191933 [running]:
        test_1        | testing.tRunner.func1.2({0x1bcb160, 0xc000157900})
        test_1        | 	/usr/local/go/src/testing/testing.go:1209 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	/usr/local/go/src/testing/testing.go:1212 +0x218
        test_1        | panic({0x1bcb160, 0xc000157900})
        test_1        | 	/usr/local/go/src/runtime/panic.go:1038 +0x215
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x3de64a0, {0x2de25e0, 0xc0009aa030})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:368 +0x467
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x2cbcce0, 0x3de64a0)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:301 +0x14c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc00019d800)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:220 +0x92
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc000c84060, {0xc000f6ab90, 0x5, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:251 +0xa4d
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0x90, 0x1d315c3, {0xc000f6ab90, 0x8e, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:97 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x203000, {0x1d315c3, 0x5}, 0x5, {0xc0005f7a30, 0x3, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:85 +0x157
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x2cd19c0, 0xc000d06a00)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:89 +0x187
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0x6808fe14ff338558)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:76 +0x33
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x1d3681b)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:46 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc000f46820)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:128 +0xe5
        test_1        | testing.tRunner(0xc000f46820, 0xc000c84000)
        test_1        | 	/usr/local/go/src/testing/testing.go:1259 +0x102
        test_1        | created by testing.(*T).Run
        test_1        | 	/usr/local/go/src/testing/testing.go:1306 +0x35a
        compare_test_1 exited with code 2
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 2
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3935438&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3935438&tab=artifacts#/) on master @ [243d90e97789ea0206356b215c0e79aedb4f2e14](https://github.com/cockroachdb/cockroach/commits/243d90e97789ea0206356b215c0e79aedb4f2e14):


```
        test_1        |     --- FAIL: TestCompare/postgres (8.44s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 DATE NOT NULL, col1_1 BOOL NULL, col1_2 TIMESTAMP NOT NULL, col1_3 REGTYPE, col1_4 OID NOT NULL, col1_5 "char" NOT NULL, col1_6 TIMESTAMPTZ NULL, col1_7 BIT NOT NULL, col1_8 INET NOT NULL, col1_9 TIMESTAMP, PRIMARY KEY (col1_5, col1_0, col1_7, col1_4, col1_2), col1_10 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_0 IS NULL THEN ''::TEXT ELSE ''::TEXT END) STORED);
        test_1        |             CREATE INDEX ON table1 (col1_6 DESC, col1_1 ASC, col1_8 ASC);
        test_1        |             CREATE INDEX ON table1 (col1_4 ASC, col1_7, col1_5, col1_1 DESC, col1_2 DESC, col1_6 DESC, col1_9 DESC, col1_3 ASC, col1_8 DESC, col1_10);
        test_1        |             CREATE INDEX ON table1 (col1_10);
        test_1        |             CREATE INDEX ON table1 (col1_0, col1_7 DESC, col1_3 DESC, col1_6 DESC, col1_10 DESC, col1_5);
        test_1        |             CREATE INDEX ON table1 (col1_3);
        test_1        |             CREATE UNIQUE INDEX ON table1 ((CASE WHEN col1_4 IS NULL THEN 'wK|a'::TEXT ELSE NULL END), col1_8 DESC);
        test_1        |             CREATE INDEX ON table1 (col1_7, col1_1 ASC, col1_10 DESC);
        test_1        |             CREATE INDEX ON table1 (col1_1 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_3);
        test_1        |             CREATE TABLE table2 (col2_0 TIMESTAMPTZ NULL, col2_1 GEOMETRY, col2_2 BIT(23) NOT NULL, col2_3 DECIMAL, col2_4 GEOGRAPHY, col2_5 GEOMETRY NULL, col2_6 TEXT GENERATED ALWAYS AS (lower(CAST(col2_1 AS TEXT))) STORED);
        test_1        |             CREATE INDEX ON table2 (lower(CAST(col2_5 AS TEXT)) ASC) INCLUDE (col2_2, col2_4, col2_5, col2_6);
        test_1        |             CREATE TABLE table3 (col3_0 UUID NULL, col3_1 JSONB, col3_2 CHAR NOT NULL, col3_3 TEXT NOT NULL GENERATED ALWAYS AS (lower(col3_2)) STORED, col3_4 TEXT GENERATED ALWAYS AS (lower(CAST(col3_1 AS TEXT))) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_3, col3_4 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_2, col3_3 DESC) INCLUDE (col3_1);
        test_1        |             CREATE TABLE table4 (col4_0 REGPROCEDURE NULL, col4_1 INT2, col4_2 CHAR, col4_3 REGNAMESPACE NOT NULL, col4_4 INTERVAL, col4_5 CHAR NOT NULL, col4_6 GEOGRAPHY, PRIMARY KEY (col4_3), col4_7 TEXT GENERATED ALWAYS AS (lower(CAST(col4_6 AS TEXT))) STORED);
        test_1        |             CREATE INDEX ON table4 (col4_1 DESC);
        test_1        |             CREATE INDEX ON table4 (col4_1 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_2 DESC, col4_1 ASC);
        test_1        |             CREATE INDEX ON table4 (col4_0, lower(CAST(col4_6 AS TEXT)) ASC, col4_5 ASC, col4_3 ASC) INCLUDE (col4_1, col4_2, col4_6, col4_7);
        test_1        |             CREATE INDEX ON table4 (col4_0 DESC, col4_4 ASC, col4_1) INCLUDE (col4_2, col4_6, col4_7);
        test_1        |             CREATE TABLE table5 (col5_0 "char" NOT NULL, col5_1 BIT(44) NOT NULL, col5_2 INT4 NULL, col5_3 FLOAT4 NOT NULL, col5_4 BOOL NOT NULL, col5_5 CHAR NOT NULL, col5_6 REGPROC, col5_7 DATE NOT NULL, col5_8 TIMESTAMP NULL, col5_9 REGNAMESPACE NULL, col5_10 REGNAMESPACE NULL, col5_11 OID NOT NULL, col5_12 REGPROCEDURE, PRIMARY KEY (col5_3, col5_0), col5_13 TEXT GENERATED ALWAYS AS (CASE WHEN col5_12 IS NULL THEN e'\x18!\x11&\f'::TEXT ELSE 'z'::TEXT END) STORED);
        test_1        |             CREATE INDEX ON table5 (col5_6, col5_1 ASC, col5_5 ASC, col5_2 ASC, col5_9 ASC, col5_0, col5_4) INCLUDE (col5_7, col5_11);
        test_1        |             CREATE INDEX ON table5 (col5_1 DESC, lower(col5_0), col5_3, lower(CAST(col5_10 AS TEXT)) DESC, col5_0 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_13 ASC, col5_10, col5_12 DESC, col5_5, col5_1, col5_9, lower(CAST(col5_1 AS TEXT)) ASC, col5_6, col5_3 DESC, col5_4);
        test_1        |             CREATE INDEX ON table5 (col5_9, lower(CAST(col5_12 AS TEXT)) DESC, col5_0 DESC, col5_5, col5_11);
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_12, col5_8 ASC, col5_3 DESC, col5_4 DESC, col5_6, col5_1 DESC, col5_5 DESC, (CASE WHEN col5_9 IS NULL THEN 'BkyOJ'::TEXT ELSE 'F'::TEXT END) DESC, col5_9 ASC);
        test_1        |             CREATE INDEX ON table5 (col5_12, col5_7 DESC, col5_9 DESC, col5_6 ASC, col5_8 ASC, col5_0 ASC, col5_5 ASC, col5_11 ASC) INCLUDE (col5_2, col5_4);
        test_1        |             CREATE INDEX ON table5 (lower(CAST(col5_11 AS TEXT)), col5_10 ASC, col5_3) INCLUDE (col5_2, col5_5, col5_8, col5_12);
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_5, col5_8, col5_13 DESC, col5_3, col5_4 ASC, col5_11 ASC, col5_10 DESC, col5_2, col5_0, col5_9, lower(col5_0) ASC, col5_7, lower(CAST(col5_4 AS TEXT)) DESC);
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_5 ASC, col5_7 DESC, col5_13 DESC, col5_2 DESC, col5_8 DESC, col5_9 DESC, col5_12 DESC);
        test_1        |             ALTER TABLE table4 ADD FOREIGN KEY (col4_0) REFERENCES table1 (col1_3) ON UPDATE RESTRICT;
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('oaikum');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3944364&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3944364&tab=artifacts#/) on master @ [6f4d61b2855ee057f7776d5f5fece72b753176be](https://github.com/cockroachdb/cockroach/commits/6f4d61b2855ee057f7776d5f5fece72b753176be):


```
        test_1        |             						AND (tab_263048.col1_10) = (tab_263050.col3_2)
        test_1        |             						AND (tab_263048.col1_1) = (tab_263051.col2_8)
        test_1        |             						AND (tab_263049.col2_7) = (tab_263050.col3_3)
        test_1        |             				JOIN postgres.public.table1@table1_col1_8_col1_3_col1_10_idx AS tab_263052 ON
        test_1        |             						(tab_263049.col2_7) = (tab_263052.col1_1)
        test_1        |             						AND (tab_263048.col1_10) = (tab_263052.col1_1)
        test_1        |             						AND (tab_263051.col2_10) = (tab_263052.col1_8) ON
        test_1        |             					(tab_263047.col1_4) = (tab_263048.col1_4)
        test_1        |             					AND (tab_263047.col1_14) = (tab_263048.col1_8)
        test_1        |             					AND (tab_263047.col1_0) = (tab_263048.col1_11)
        test_1        |             					AND (tab_263047.tableoid) = (tab_263051.tableoid)
        test_1        |             					AND (tab_263046.col1_6) = (tab_263052.col1_9)
        test_1        |             					AND (tab_263047.col1_1) = (tab_263048.col1_10)
        test_1        |             					AND (tab_263047.col1_0) = (tab_263052.col1_12)
        test_1        |             					AND (tab_263046.col1_5) = (tab_263048.col1_12)
        test_1        |             					AND (tab_263047.col1_12) = (tab_263051.col2_3)
        test_1        |             		WHERE
        test_1        |             			true
        test_1        |             		ORDER BY
        test_1        |             			tab_263052.col1_11 ASC, tab_263051.col2_3 DESC
        test_1        |             	)
        test_1        |             		AS tab_263053
        test_1        |             ORDER BY
        test_1        |             	col_444488 NULLS FIRST, col_444489 NULLS FIRST, col_444490 NULLS FIRST, col_444491 NULLS FIRST
        test_1        |         compare_test.go:194: cockroach1: ping: conn closed
        test_1        |     --- FAIL: TestCompare/postgres (209.07s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 JSONB NULL, col1_1 INET NOT NULL, col1_2 BOOL NULL, col1_3 FLOAT4 NOT NULL, col1_4 TEXT NOT NULL, col1_5 REGTYPE NULL, col1_6 REGCLASS NULL, col1_7 INT8 NULL, col1_8 DECIMAL, col1_9 INT8 NULL, col1_10 BIT(24) NOT NULL, col1_11 DATE, col1_12 INT2 NULL, col1_13 INET NOT NULL, col1_14 FLOAT8 NULL, col1_15 BYTEA NULL, col1_16 FLOAT8 NULL GENERATED ALWAYS AS (col1_14 + col1_3) STORED, col1_17 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, col1_18 DECIMAL GENERATED ALWAYS AS (col1_8 + (-8.368515051553329394E+31)::DECIMAL) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table1 ((col1_14 + (-1.0)::FLOAT8) ASC, col1_11 ASC, col1_13 DESC, col1_18 DESC, col1_8 ASC, col1_7 DESC, col1_17, col1_3, col1_6, col1_4 ASC, col1_1 ASC) INCLUDE (col1_0, col1_9, col1_14);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_6, col1_4 ASC, col1_17 DESC, col1_9 ASC, col1_15, col1_12) INCLUDE (col1_0, col1_3, col1_7, col1_10, col1_11, col1_18);
        test_1        |             CREATE INDEX ON table1 (col1_3 ASC, col1_10, col1_15, col1_16, col1_17, (CASE WHEN col1_15 IS NULL THEN '~|FAc#4'::TEXT ELSE 'j'::TEXT END) ASC, col1_18, col1_8 DESC, col1_9 ASC, ((col1_7 + col1_12) + col1_9) DESC, col1_5 ASC, ((col1_12 + col1_9) + col1_7) ASC, col1_11, col1_14) INCLUDE (col1_1, col1_7);
        test_1        |             CREATE INDEX ON table1 (lower(CAST(col1_11 AS TEXT)) ASC, col1_15 DESC, col1_17 ASC, col1_14 DESC, col1_13 ASC, col1_11 ASC, lower(col1_4), col1_18, col1_7 DESC, col1_4 ASC, col1_12, col1_6, col1_8 DESC, col1_16 ASC) INCLUDE (col1_2, col1_5, col1_10);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('jlz');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('dznos');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('m', 'lalkvq', 'bz', 'hx', 'edi', 'aonztk');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3949695&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3949695&tab=artifacts#/) on master @ [4cbe795c7f8c36c007c3ca1250367cb181f5ca0c](https://github.com/cockroachdb/cockroach/commits/4cbe795c7f8c36c007c3ca1250367cb181f5ca0c):


```
        postgres_1    | 2021-12-19 06:15:10.639 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2021-12-19 06:15:10.644 UTC [71] LOG:  database system was shut down at 2021-12-19 06:15:10 UTC
        postgres_1    | 2021-12-19 06:15:10.649 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2021-12-19 06:15:14.308 UTC [81] ERROR:  functions in index expression must be marked IMMUTABLE
        postgres_1    | 2021-12-19 06:15:14.308 UTC [81] STATEMENT:  CREATE TABLE table1 (col1_0 TEXT, col1_1 CHAR, col1_2 CHAR, col1_3 NAME NULL);
        postgres_1    | 	CREATE TABLE table2 (col2_0 REGPROC NOT NULL, col2_1 BOX2D NOT NULL, col2_2 "char" NOT NULL, col2_3 BIT(49) NOT NULL, col2_4 REGTYPE NOT NULL, col2_5 DECIMAL NULL, col2_6 BIT(40) NOT NULL, col2_7 INT2 NOT NULL, col2_8 OID NOT NULL, col2_9 TIMESTAMPTZ NOT NULL, col2_10 UUID NOT NULL, col2_11 TIMESTAMP NOT NULL, PRIMARY KEY (col2_11, col2_4, col2_0, col2_10, col2_9, col2_3, col2_6, col2_7, col2_2, col2_8), col2_12 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_1 AS TEXT))) STORED, col2_13 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_4 IS NULL THEN e'dC\x1d'::TEXT ELSE e'\x1d'::TEXT END) STORED, col2_14 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_8 IS NULL THEN e'3_\t8;*'::TEXT ELSE e'7,H\bbSu'::TEXT END) STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_7 DESC, (col2_7 + (-27531)::INT8), col2_6 ASC) INCLUDE (col2_12);
        postgres_1    | 	CREATE INDEX ON table2 (col2_5 ASC, col2_12 ASC, col2_4 ASC, col2_11 DESC, col2_7 DESC, col2_8 ASC, col2_0 DESC, lower(CAST(col2_0 AS TEXT)) DESC);
        postgres_1    | 	CREATE INDEX ON table2 (col2_8 ASC, col2_2 DESC, col2_4, col2_9, lower(CAST(col2_1 AS TEXT)) DESC, col2_3 ASC, lower(CAST(col2_0 AS TEXT)));
        postgres_1    | 	CREATE INDEX ON table2 (col2_14, col2_3 DESC, col2_4 DESC, col2_10, col2_8 ASC, col2_2, col2_6 DESC, col2_12 ASC, col2_9 ASC, col2_5, col2_7 DESC, col2_13 ASC, col2_0 ASC, col2_11 DESC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 ((CASE WHEN col2_4 IS NULL THEN 'WOq:"t'::TEXT ELSE NULL END), col2_9 ASC, col2_8 DESC, (CASE WHEN col2_11 IS NULL THEN e'd_\x1f\x1ba3\'d'::TEXT ELSE 'c51D<'::TEXT END), col2_12 DESC, col2_2 DESC, col2_6 ASC, col2_14 ASC, col2_5 ASC, col2_7, col2_4);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_8 ASC) INCLUDE (col2_5, col2_14);
        postgres_1    | 	CREATE INDEX ON table2 (col2_10 ASC, col2_14 ASC, col2_3 ASC, col2_6, col2_9 ASC, col2_2 ASC, col2_5 ASC, col2_13 DESC, lower(CAST(col2_8 AS TEXT)) ASC, col2_12, col2_7 DESC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_2, col2_12);
        postgres_1    | 	ALTER TABLE table1 ADD FOREIGN KEY (col1_1, col1_0) REFERENCES table2 (col2_2, col2_12);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('xapvb');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.82s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (3.20s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 TEXT, col1_1 CHAR, col1_2 CHAR, col1_3 NAME NULL);
        test_1        |             CREATE TABLE table2 (col2_0 REGPROC NOT NULL, col2_1 BOX2D NOT NULL, col2_2 "char" NOT NULL, col2_3 BIT(49) NOT NULL, col2_4 REGTYPE NOT NULL, col2_5 DECIMAL NULL, col2_6 BIT(40) NOT NULL, col2_7 INT2 NOT NULL, col2_8 OID NOT NULL, col2_9 TIMESTAMPTZ NOT NULL, col2_10 UUID NOT NULL, col2_11 TIMESTAMP NOT NULL, PRIMARY KEY (col2_11, col2_4, col2_0, col2_10, col2_9, col2_3, col2_6, col2_7, col2_2, col2_8), col2_12 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_1 AS TEXT))) STORED, col2_13 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_4 IS NULL THEN e'dC\x1d'::TEXT ELSE e'\x1d'::TEXT END) STORED, col2_14 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_8 IS NULL THEN e'3_\t8;*'::TEXT ELSE e'7,H\bbSu'::TEXT END) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_7 DESC, (col2_7 + (-27531)::INT8), col2_6 ASC) INCLUDE (col2_12);
        test_1        |             CREATE INDEX ON table2 (col2_5 ASC, col2_12 ASC, col2_4 ASC, col2_11 DESC, col2_7 DESC, col2_8 ASC, col2_0 DESC, lower(CAST(col2_0 AS TEXT)) DESC);
        test_1        |             CREATE INDEX ON table2 (col2_8 ASC, col2_2 DESC, col2_4, col2_9, lower(CAST(col2_1 AS TEXT)) DESC, col2_3 ASC, lower(CAST(col2_0 AS TEXT)));
        test_1        |             CREATE INDEX ON table2 (col2_14, col2_3 DESC, col2_4 DESC, col2_10, col2_8 ASC, col2_2, col2_6 DESC, col2_12 ASC, col2_9 ASC, col2_5, col2_7 DESC, col2_13 ASC, col2_0 ASC, col2_11 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table2 ((CASE WHEN col2_4 IS NULL THEN 'WOq:"t'::TEXT ELSE NULL END), col2_9 ASC, col2_8 DESC, (CASE WHEN col2_11 IS NULL THEN e'd_\x1f\x1ba3\'d'::TEXT ELSE 'c51D<'::TEXT END), col2_12 DESC, col2_2 DESC, col2_6 ASC, col2_14 ASC, col2_5 ASC, col2_7, col2_4);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_8 ASC) INCLUDE (col2_5, col2_14);
        test_1        |             CREATE INDEX ON table2 (col2_10 ASC, col2_14 ASC, col2_3 ASC, col2_6, col2_9 ASC, col2_2 ASC, col2_5 ASC, col2_13 DESC, lower(CAST(col2_8 AS TEXT)) ASC, col2_12, col2_7 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_2, col2_12);
        test_1        |             ALTER TABLE table1 ADD FOREIGN KEY (col1_1, col1_0) REFERENCES table2 (col2_2, col2_12);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('xapvb');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3954618&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3954618&tab=artifacts#/) on master @ [7bd8974eca566ca7625b6621decc5ae52d4fd38e](https://github.com/cockroachdb/cockroach/commits/7bd8974eca566ca7625b6621decc5ae52d4fd38e):


```
        postgres_1    | 2021-12-20 06:14:36.249 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2021-12-20 06:14:39.169 UTC [79] ERROR:  functions in index expression must be marked IMMUTABLE
        postgres_1    | 2021-12-20 06:14:39.169 UTC [79] STATEMENT:  CREATE TABLE table1 (col1_0 INET NULL, col1_1 GEOGRAPHY NULL, col1_2 "char", col1_3 TIMESTAMP NOT NULL, col1_4 TIMETZ NOT NULL, col1_5 REGPROCEDURE NULL, col1_6 FLOAT4 NULL, col1_7 INTERVAL NOT NULL, col1_8 BOOL NOT NULL, col1_9 DATE, col1_10 REGROLE, col1_11 REGCLASS NULL, col1_12 FLOAT4 NULL GENERATED ALWAYS AS (col1_6 + 1.4906501770019531::FLOAT8) STORED, col1_13 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, col1_14 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col1_11 IS NULL THEN e'\rU\x1e'::TEXT ELSE 'R'::TEXT END) STORED, col1_15 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_3 IS NULL THEN e'9\x10'::TEXT ELSE e'sh\x1a'::TEXT END) STORED, col1_16 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col1_11 IS NULL THEN ''::TEXT ELSE NULL END) STORED);
        postgres_1    | 	CREATE TABLE table2 (col2_0 FLOAT4[] NULL, col2_1 REGTYPE, col2_2 JSONB NOT NULL, col2_3 BYTEA, col2_4 REGCLASS NOT NULL, col2_5 "char" NOT NULL, col2_6 TIMESTAMPTZ, col2_7 BOOL, col2_8 "char" NOT NULL, col2_9 GEOMETRY NULL, col2_10 FLOAT4 NOT NULL, col2_11 VARCHAR NOT NULL, col2_12 CHAR, col2_13 REGPROCEDURE NULL, col2_14 INET);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_14 DESC);
        postgres_1    | 	CREATE TABLE table3 (col3_0 JSONB, col3_1 INTERVAL, col3_2 TIMESTAMP NOT NULL, col3_3 TIMETZ NOT NULL, col3_4 INT2 NOT NULL, col3_5 "char" NULL, col3_6 DECIMAL NOT NULL, col3_7 REGPROC NOT NULL, col3_8 TIME NOT NULL, col3_9 VARCHAR NULL, col3_10 GEOMETRY NULL, PRIMARY KEY (col3_2, col3_4, col3_8, col3_6, col3_3), col3_11 TEXT GENERATED ALWAYS AS (lower(CAST(col3_0 AS TEXT))) STORED, col3_12 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_3 AS TEXT))) STORED, col3_13 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_2 IS NULL THEN e'^:?K\n\x18\x079N'::TEXT ELSE ''::TEXT END) STORED, col3_14 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col3_10 AS TEXT))) STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON table3 (col3_13 ASC);
        postgres_1    | 	CREATE TABLE table4 (col4_0 REGPROC NOT NULL, col4_1 DATE NOT NULL, col4_2 NAME NOT NULL, col4_3 FLOAT4 NOT NULL, col4_4 CHAR NOT NULL, col4_5 TIMESTAMP NOT NULL, col4_6 GEOGRAPHY NULL, col4_7 INET NOT NULL, PRIMARY KEY (col4_3, col4_1), col4_8 TEXT NOT NULL GENERATED ALWAYS AS (lower(col4_2)) STORED, col4_9 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col4_6 AS TEXT))) STORED, col4_10 TEXT NOT NULL GENERATED ALWAYS AS (lower(col4_2)) STORED);
        postgres_1    | 	CREATE INDEX ON table4 (col4_7, lower(CAST(col4_0 AS TEXT)), col4_10 DESC, col4_8 DESC, col4_5, col4_3, (col4_3 + 0.8494518995285034::FLOAT8) DESC, col4_0, col4_4, col4_1 ASC) INCLUDE (col4_2, col4_9);
        postgres_1    | 	CREATE INDEX ON table4 (col4_7, col4_5);
        postgres_1    | 	CREATE INDEX ON table4 (lower(CAST(col4_5 AS TEXT)) ASC, col4_10 ASC, lower(CAST(col4_5 AS TEXT)), col4_5 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (lower(CAST(col4_7 AS TEXT)) DESC);
        postgres_1    | 	CREATE TABLE table5 (col5_0 FLOAT4 NULL, col5_1 REGPROCEDURE NOT NULL, col5_2 REGCLASS, col5_3 REGCLASS, col5_4 INT2 NULL, col5_5 INET NULL, col5_6 TIMESTAMP NOT NULL, col5_7 JSONB, col5_8 BOX2D[] NOT NULL);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('usiepg', 'dkdojf', 'k', 'yqx');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('mygnj', 'u', 'labj', 'uaq', 'urfe', 'salmar');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.52s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (2.37s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 INET NULL, col1_1 GEOGRAPHY NULL, col1_2 "char", col1_3 TIMESTAMP NOT NULL, col1_4 TIMETZ NOT NULL, col1_5 REGPROCEDURE NULL, col1_6 FLOAT4 NULL, col1_7 INTERVAL NOT NULL, col1_8 BOOL NOT NULL, col1_9 DATE, col1_10 REGROLE, col1_11 REGCLASS NULL, col1_12 FLOAT4 NULL GENERATED ALWAYS AS (col1_6 + 1.4906501770019531::FLOAT8) STORED, col1_13 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, col1_14 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col1_11 IS NULL THEN e'\rU\x1e'::TEXT ELSE 'R'::TEXT END) STORED, col1_15 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_3 IS NULL THEN e'9\x10'::TEXT ELSE e'sh\x1a'::TEXT END) STORED, col1_16 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col1_11 IS NULL THEN ''::TEXT ELSE NULL END) STORED);
        test_1        |             CREATE TABLE table2 (col2_0 FLOAT4[] NULL, col2_1 REGTYPE, col2_2 JSONB NOT NULL, col2_3 BYTEA, col2_4 REGCLASS NOT NULL, col2_5 "char" NOT NULL, col2_6 TIMESTAMPTZ, col2_7 BOOL, col2_8 "char" NOT NULL, col2_9 GEOMETRY NULL, col2_10 FLOAT4 NOT NULL, col2_11 VARCHAR NOT NULL, col2_12 CHAR, col2_13 REGPROCEDURE NULL, col2_14 INET);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_14 DESC);
        test_1        |             CREATE TABLE table3 (col3_0 JSONB, col3_1 INTERVAL, col3_2 TIMESTAMP NOT NULL, col3_3 TIMETZ NOT NULL, col3_4 INT2 NOT NULL, col3_5 "char" NULL, col3_6 DECIMAL NOT NULL, col3_7 REGPROC NOT NULL, col3_8 TIME NOT NULL, col3_9 VARCHAR NULL, col3_10 GEOMETRY NULL, PRIMARY KEY (col3_2, col3_4, col3_8, col3_6, col3_3), col3_11 TEXT GENERATED ALWAYS AS (lower(CAST(col3_0 AS TEXT))) STORED, col3_12 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_3 AS TEXT))) STORED, col3_13 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_2 IS NULL THEN e'^:?K\n\x18\x079N'::TEXT ELSE ''::TEXT END) STORED, col3_14 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col3_10 AS TEXT))) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_13 ASC);
        test_1        |             CREATE TABLE table4 (col4_0 REGPROC NOT NULL, col4_1 DATE NOT NULL, col4_2 NAME NOT NULL, col4_3 FLOAT4 NOT NULL, col4_4 CHAR NOT NULL, col4_5 TIMESTAMP NOT NULL, col4_6 GEOGRAPHY NULL, col4_7 INET NOT NULL, PRIMARY KEY (col4_3, col4_1), col4_8 TEXT NOT NULL GENERATED ALWAYS AS (lower(col4_2)) STORED, col4_9 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col4_6 AS TEXT))) STORED, col4_10 TEXT NOT NULL GENERATED ALWAYS AS (lower(col4_2)) STORED);
        test_1        |             CREATE INDEX ON table4 (col4_7, lower(CAST(col4_0 AS TEXT)), col4_10 DESC, col4_8 DESC, col4_5, col4_3, (col4_3 + 0.8494518995285034::FLOAT8) DESC, col4_0, col4_4, col4_1 ASC) INCLUDE (col4_2, col4_9);
        test_1        |             CREATE INDEX ON table4 (col4_7, col4_5);
        test_1        |             CREATE INDEX ON table4 (lower(CAST(col4_5 AS TEXT)) ASC, col4_10 ASC, lower(CAST(col4_5 AS TEXT)), col4_5 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (lower(CAST(col4_7 AS TEXT)) DESC);
        test_1        |             CREATE TABLE table5 (col5_0 FLOAT4 NULL, col5_1 REGPROCEDURE NOT NULL, col5_2 REGCLASS, col5_3 REGCLASS, col5_4 INT2 NULL, col5_5 INET NULL, col5_6 TIMESTAMP NOT NULL, col5_7 JSONB, col5_8 BOX2D[] NOT NULL);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('usiepg', 'dkdojf', 'k', 'yqx');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('mygnj', 'u', 'labj', 'uaq', 'urfe', 'salmar');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3964617&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3964617&tab=artifacts#/) on master @ [2bf850847554519dc139289615e1a061b6a251de](https://github.com/cockroachdb/cockroach/commits/2bf850847554519dc139289615e1a061b6a251de):


```
        postgres_1    | 				postgres.public.table1 AS tab_38404
        postgres_1    | 			GROUP BY
        postgres_1    | 				tab_38404.col1_1, tab_38404.col1_0, tab_38404.tableoid
        postgres_1    | 			HAVING
        postgres_1    | 				bool_and(false::BOOL::BOOL)::BOOL
        postgres_1    | 			ORDER BY
        postgres_1    | 				tab_38404.col1_0 DESC
        postgres_1    | 		)
        postgres_1    | 			AS tab_38405
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_59162 NULLS FIRST, col_59163 NULLS FIRST, col_59164 NULLS FIRST
        test_1        | --- FAIL: TestCompare (659.16s)
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (55.52s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	string("{NULL,NULL}"),
        test_1        |             + 	string("{NULL:NULL}"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			((NULL::GEOGRAPHY[]::GEOGRAPHY[] || NULL::GEOGRAPHY::GEOGRAPHY)::GEOGRAPHY[]::GEOGRAPHY[] || NULL::GEOGRAPHY::GEOGRAPHY)::GEOGRAPHY[]
        test_1        |             				AS col_59573
        test_1        |             	)
        test_1        |             		AS tab_38673
        test_1        |             ORDER BY
        test_1        |             	col_59573 NULLS FIRST;
        test_1        |             
        test_1        |             postgres: [same as previous]
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3975116&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3975116&tab=artifacts#/) on master @ [cadf1bf18001d00fc03f32f22d4e18b920e583ba](https://github.com/cockroachdb/cockroach/commits/cadf1bf18001d00fc03f32f22d4e18b920e583ba):


```
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2021-12-22 06:15:18.309 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2021-12-22 06:15:18.309 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2021-12-22 06:15:18.309 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2021-12-22 06:15:18.313 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2021-12-22 06:15:18.318 UTC [68] LOG:  database system was shut down at 2021-12-22 06:15:18 UTC
        postgres_1    | 2021-12-22 06:15:18.324 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2021-12-22 06:25:21.597 UTC [98] ERROR:  functions in index expression must be marked IMMUTABLE
        postgres_1    | 2021-12-22 06:25:21.597 UTC [98] STATEMENT:  CREATE TABLE table1 (col1_0 BOX2D, col1_1 JSONB NULL, col1_2 BIT(27) NOT NULL, col1_3 VARCHAR NOT NULL, col1_4 REGPROCEDURE NULL, col1_5 REGTYPE NOT NULL, col1_6 JSONB, col1_7 REGTYPE NULL, PRIMARY KEY (col1_2, col1_3));
        postgres_1    | 	CREATE INDEX ON table1 (lower(CAST(col1_5 AS TEXT)) ASC, col1_4 ASC, col1_2 ASC, col1_5 DESC, col1_3 ASC);
        postgres_1    | 	CREATE INDEX ON table1 (col1_4 ASC, lower(CAST(col1_7 AS TEXT)) DESC, lower(CAST(col1_7 AS TEXT)) ASC, col1_2 DESC, col1_3 DESC, col1_7, col1_5) INCLUDE (col1_0, col1_1);
        postgres_1    | 	CREATE TABLE table2 (col2_0 VARCHAR NULL, col2_1 BYTEA NOT NULL, col2_2 BIT(38) NOT NULL, col2_3 REGROLE NOT NULL, PRIMARY KEY (col2_1), col2_4 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_1 IS NULL THEN 'u?|":0'::TEXT ELSE e'%vb\x1c\\f\x11'::TEXT END) STORED, col2_5 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_2 AS TEXT))) STORED, col2_6 TEXT NULL GENERATED ALWAYS AS (lower(col2_0)) STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_2);
        postgres_1    | 	ALTER TABLE table1 ADD FOREIGN KEY (col1_2) REFERENCES table2 (col2_2);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('j');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('e', 's', 'k', 'vmr');
        postgres_1    | 	CREATE TYPE rand_typ_2 AS ENUM ('jpwy');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.07s)
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (1.91s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 BOX2D, col1_1 JSONB NULL, col1_2 BIT(27) NOT NULL, col1_3 VARCHAR NOT NULL, col1_4 REGPROCEDURE NULL, col1_5 REGTYPE NOT NULL, col1_6 JSONB, col1_7 REGTYPE NULL, PRIMARY KEY (col1_2, col1_3));
        test_1        |             CREATE INDEX ON table1 (lower(CAST(col1_5 AS TEXT)) ASC, col1_4 ASC, col1_2 ASC, col1_5 DESC, col1_3 ASC);
        test_1        |             CREATE INDEX ON table1 (col1_4 ASC, lower(CAST(col1_7 AS TEXT)) DESC, lower(CAST(col1_7 AS TEXT)) ASC, col1_2 DESC, col1_3 DESC, col1_7, col1_5) INCLUDE (col1_0, col1_1);
        test_1        |             CREATE TABLE table2 (col2_0 VARCHAR NULL, col2_1 BYTEA NOT NULL, col2_2 BIT(38) NOT NULL, col2_3 REGROLE NOT NULL, PRIMARY KEY (col2_1), col2_4 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_1 IS NULL THEN 'u?|":0'::TEXT ELSE e'%vb\x1c\\f\x11'::TEXT END) STORED, col2_5 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_2 AS TEXT))) STORED, col2_6 TEXT NULL GENERATED ALWAYS AS (lower(col2_0)) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_2);
        test_1        |             ALTER TABLE table1 ADD FOREIGN KEY (col1_2) REFERENCES table2 (col2_2);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('j');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('e', 's', 'k', 'vmr');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('jpwy');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3985329&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3985329&tab=artifacts#/) on master @ [8d4953ef35bf310c72108c2790ba21971e9e7291](https://github.com/cockroachdb/cockroach/commits/8d4953ef35bf310c72108c2790ba21971e9e7291):


```
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (347.39s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	nil,
        test_1        |             + 	string("BOX(0 -90,45 -45)"),
        test_1        |               	int32(8),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_box2dfromgeohash('H':::STRING::STRING::STRING, NULL::INT4::INT4)::BOX2D AS col_236498,
        test_1        |             			8:::INT8::INT4 AS col_236499
        test_1        |             	)
        test_1        |             		AS tab_157668
        test_1        |             ORDER BY
        test_1        |             	CAST(col_236498 AS STRING) NULLS FIRST, col_236499 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_box2dfromgeohash('H'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::BOX2D AS col_236498,
        test_1        |             			8::INT8::INT4 AS col_236499
        test_1        |             	)
        test_1        |             		AS tab_157668
        test_1        |             ORDER BY
        test_1        |             	CAST(col_236498 AS TEXT) NULLS FIRST, col_236499 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3993764&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3993764&tab=artifacts#/) on master @ [8c1528fff5c132b971428bc7f18bf0ea5eb16799](https://github.com/cockroachdb/cockroach/commits/8c1528fff5c132b971428bc7f18bf0ea5eb16799):


```
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2021-12-24 06:15:42.173 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2021-12-24 06:15:42.173 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2021-12-24 06:15:42.173 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2021-12-24 06:15:42.177 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2021-12-24 06:15:42.181 UTC [70] LOG:  database system was shut down at 2021-12-24 06:15:42 UTC
        postgres_1    | 2021-12-24 06:15:42.186 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2021-12-24 06:15:43.475 UTC [80] ERROR:  syntax error at or near "ASC" at character 209
        postgres_1    | 2021-12-24 06:15:43.475 UTC [80] STATEMENT:  CREATE TABLE table1 (col1_0 TIMESTAMPTZ NULL, col1_1 REGNAMESPACE NULL, col1_2 BYTEA NOT NULL, col1_3 TIME NOT NULL, col1_4 DECIMAL NOT NULL, col1_5 DECIMAL NULL, col1_6 BIT(21) NOT NULL, PRIMARY KEY (col1_4 ASC, col1_6 ASC), UNIQUE ((col1_4 + col1_5) ASC));
        postgres_1    | 	CREATE TABLE table2 (col2_0 INET, col2_1 BOOL NULL, col2_2 REGPROCEDURE NOT NULL, col2_3 CHAR, col2_4 REGPROCEDURE, col2_5 REGPROCEDURE, col2_6 REGTYPE NULL, col2_7 NAME, col2_8 REGTYPE NULL, col2_9 TEXT GENERATED ALWAYS AS (lower(CAST(col2_0 AS TEXT))) STORED, UNIQUE (col2_7 DESC, col2_2 DESC, col2_0 ASC), UNIQUE (col2_5, col2_2 ASC, lower(CAST(col2_5 AS TEXT)) ASC, lower(CAST(col2_4 AS TEXT)) DESC) WHERE (((table2.col2_9 > 'X'::TEXT) OR (NOT table2.col2_1)) AND (table2.col2_7 < ''::TEXT)) OR (table2.col2_3 < ''::TEXT));
        postgres_1    | 	CREATE TABLE table3 (col3_0 INT8[] NOT NULL, col3_1 FLOAT4 NOT NULL, col3_2 INT8 NULL, col3_3 FLOAT4 NULL, col3_4 FLOAT4 NULL GENERATED ALWAYS AS (col3_3 + col3_1) STORED, INDEX (col3_3, col3_1 DESC) WHERE (table3.col3_1 < 0.0::FLOAT8) AND (table3.col3_3 >= '-Inf'::FLOAT8), INDEX (col3_4 ASC) WHERE ((table3.col3_4 <= 0.0::FLOAT8) OR (table3.col3_2 >= 127::INT8)) OR (table3.col3_1 = 1.0::FLOAT8), UNIQUE ((col3_1 + col3_3) ASC, col3_2, (col3_1 + col3_3)) WHERE ((table3.col3_3 < 'NaN'::FLOAT8) OR (table3.col3_2 < (-2147483648)::INT8)) OR (table3.col3_1 <= 1.401298464324817e-45::FLOAT8), UNIQUE (col3_4 ASC, col3_3), UNIQUE (col3_1 ASC, col3_4 DESC, (col3_3 + col3_1) DESC), UNIQUE ((col3_2 + 7669678327495171027::INT8) DESC, col3_1 DESC, col3_4 ASC, col3_2 ASC, (col3_2 + (-6123987459877231477)::INT8)), INDEX (col3_1) INCLUDE (col3_2));
        postgres_1    | 	CREATE TABLE table4 (col4_0 DATE NULL, col4_1 CHAR NOT NULL, col4_2 TEXT NOT NULL GENERATED ALWAYS AS (lower(col4_1)) STORED, PRIMARY KEY (col4_2 DESC, col4_1 ASC), INDEX (col4_0, col4_1, col4_2 DESC) WHERE (table4.col4_2 < e'\''::TEXT) AND (table4.col4_1 >= e'\U00002603'::TEXT), INDEX (col4_0, col4_2, col4_1 DESC), UNIQUE (col4_1), UNIQUE (col4_2 DESC) WHERE (table4.col4_2 >= ''::TEXT) OR (table4.col4_1 > 'X'::TEXT), UNIQUE (col4_1 ASC, col4_0, col4_2 ASC));
        postgres_1    | 	CREATE TABLE table5 (col5_0 TIMESTAMP NOT NULL, col5_1 TIMESTAMP NOT NULL, col5_2 INT2[] NULL, col5_3 FLOAT8[], col5_4 TIMETZ, col5_5 BOOL NULL, col5_6 REGPROC NOT NULL, col5_7 FLOAT4 NULL, col5_8 TEXT GENERATED ALWAYS AS (lower(CAST(col5_4 AS TEXT))) STORED);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('ayd', 'obgvy');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('jfda');
        postgres_1    | 	CREATE TYPE rand_typ_2 AS ENUM ('r', 'qb');
        postgres_1    | 	CREATE TYPE rand_typ_3 AS ENUM ('cp', 'laneha', 'tzabjf');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (604.59s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.89s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 TIMESTAMPTZ NULL, col1_1 REGNAMESPACE NULL, col1_2 BYTEA NOT NULL, col1_3 TIME NOT NULL, col1_4 DECIMAL NOT NULL, col1_5 DECIMAL NULL, col1_6 BIT(21) NOT NULL, PRIMARY KEY (col1_4 ASC, col1_6 ASC), UNIQUE ((col1_4 + col1_5) ASC));
        test_1        |             CREATE TABLE table2 (col2_0 INET, col2_1 BOOL NULL, col2_2 REGPROCEDURE NOT NULL, col2_3 CHAR, col2_4 REGPROCEDURE, col2_5 REGPROCEDURE, col2_6 REGTYPE NULL, col2_7 NAME, col2_8 REGTYPE NULL, col2_9 TEXT GENERATED ALWAYS AS (lower(CAST(col2_0 AS TEXT))) STORED, UNIQUE (col2_7 DESC, col2_2 DESC, col2_0 ASC), UNIQUE (col2_5, col2_2 ASC, lower(CAST(col2_5 AS TEXT)) ASC, lower(CAST(col2_4 AS TEXT)) DESC) WHERE (((table2.col2_9 > 'X'::TEXT) OR (NOT table2.col2_1)) AND (table2.col2_7 < ''::TEXT)) OR (table2.col2_3 < ''::TEXT));
        test_1        |             CREATE TABLE table3 (col3_0 INT8[] NOT NULL, col3_1 FLOAT4 NOT NULL, col3_2 INT8 NULL, col3_3 FLOAT4 NULL, col3_4 FLOAT4 NULL GENERATED ALWAYS AS (col3_3 + col3_1) STORED, INDEX (col3_3, col3_1 DESC) WHERE (table3.col3_1 < 0.0::FLOAT8) AND (table3.col3_3 >= '-Inf'::FLOAT8), INDEX (col3_4 ASC) WHERE ((table3.col3_4 <= 0.0::FLOAT8) OR (table3.col3_2 >= 127::INT8)) OR (table3.col3_1 = 1.0::FLOAT8), UNIQUE ((col3_1 + col3_3) ASC, col3_2, (col3_1 + col3_3)) WHERE ((table3.col3_3 < 'NaN'::FLOAT8) OR (table3.col3_2 < (-2147483648)::INT8)) OR (table3.col3_1 <= 1.401298464324817e-45::FLOAT8), UNIQUE (col3_4 ASC, col3_3), UNIQUE (col3_1 ASC, col3_4 DESC, (col3_3 + col3_1) DESC), UNIQUE ((col3_2 + 7669678327495171027::INT8) DESC, col3_1 DESC, col3_4 ASC, col3_2 ASC, (col3_2 + (-6123987459877231477)::INT8)), INDEX (col3_1) INCLUDE (col3_2));
        test_1        |             CREATE TABLE table4 (col4_0 DATE NULL, col4_1 CHAR NOT NULL, col4_2 TEXT NOT NULL GENERATED ALWAYS AS (lower(col4_1)) STORED, PRIMARY KEY (col4_2 DESC, col4_1 ASC), INDEX (col4_0, col4_1, col4_2 DESC) WHERE (table4.col4_2 < e'\''::TEXT) AND (table4.col4_1 >= e'\U00002603'::TEXT), INDEX (col4_0, col4_2, col4_1 DESC), UNIQUE (col4_1), UNIQUE (col4_2 DESC) WHERE (table4.col4_2 >= ''::TEXT) OR (table4.col4_1 > 'X'::TEXT), UNIQUE (col4_1 ASC, col4_0, col4_2 ASC));
        test_1        |             CREATE TABLE table5 (col5_0 TIMESTAMP NOT NULL, col5_1 TIMESTAMP NOT NULL, col5_2 INT2[] NULL, col5_3 FLOAT8[], col5_4 TIMETZ, col5_5 BOOL NULL, col5_6 REGPROC NOT NULL, col5_7 FLOAT4 NULL, col5_8 TEXT GENERATED ALWAYS AS (lower(CAST(col5_4 AS TEXT))) STORED);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('ayd', 'obgvy');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('jfda');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('r', 'qb');
        test_1        |             CREATE TYPE rand_typ_3 AS ENUM ('cp', 'laneha', 'tzabjf');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: syntax error at or near "ASC" (SQLSTATE 42601)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3999332&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3999332&tab=artifacts#/) on master @ [f3710ae65e327d0795123029db679f8c9f3e610e](https://github.com/cockroachdb/cockroach/commits/f3710ae65e327d0795123029db679f8c9f3e610e):


```
        test_1        |             						JOIN postgres.public.table4@[0] AS tab_18027
        test_1        |             							JOIN postgres.public.table1@table1_col1_6_expr_col1_3_expr1_col1_7_col1_10_col1_0_idx
        test_1        |             									AS tab_18028 ON
        test_1        |             									(tab_18027.col4_3) = (tab_18028.col1_8)
        test_1        |             									AND (tab_18027.col4_8) = (tab_18028.col1_9) ON
        test_1        |             								(tab_18023.col1_6) = (tab_18028.col1_8)
        test_1        |             								AND (tab_18023.crdb_internal_mvcc_timestamp) = (tab_18027.col4_14) ON
        test_1        |             							(tab_18021.col4_2) = (tab_18027.col4_2)
        test_1        |             							AND (tab_18022.col4_2) = (tab_18027.col4_10)
        test_1        |             							AND (tab_18021.col4_2) = (tab_18027.col4_13) ON
        test_1        |             						(tab_18020.col3_11) = (tab_18022.col4_2) AND (tab_18020.col3_12) = (tab_18027.col4_16) ON
        test_1        |             					(tab_18019.col4_17) = (tab_18020.crdb_internal_mvcc_timestamp)
        test_1        |             					AND (tab_18019.col4_10) = (tab_18020.col3_4)
        test_1        |             	)
        test_1        |             		AS tab_18029
        test_1        |             ORDER BY
        test_1        |             	col_31111 NULLS FIRST, col_31112 NULLS FIRST
        test_1        |         compare_test.go:194: cockroach1: ping: conn closed
        test_1        |     --- FAIL: TestCompare/postgres (40.08s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:193: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			6:::INT8::INT8 AS col_6063,
        test_1        |             			9:::INT8::INT8 AS col_6064,
        test_1        |             			tab_3096.col2_11 AS col_6065,
        test_1        |             			tab_3096.col2_11 AS col_6066
        test_1        |             		FROM
        test_1        |             			postgres.public.table2 AS tab_3095
        test_1        |             			JOIN postgres.public.table2 AS tab_3096 ON (tab_3095.col2_4) = (tab_3096.col2_4)
        test_1        |             		ORDER BY
        test_1        |             			tab_3096.col2_11 ASC
        test_1        |             	)
        test_1        |             		AS tab_3097
        test_1        |             ORDER BY
        test_1        |             	col_6063 NULLS FIRST, col_6064 NULLS FIRST, col_6065 NULLS FIRST, col_6066 NULLS FIRST
        test_1        |         compare_test.go:194: cockroach1: ping: conn closed
        test_1        | FAIL
        compare_cockroach1_1 exited with code 2
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4010619&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4010619&tab=artifacts#/) on master @ [5ac733bb4927020bc1c52da24b2591742fde8e1f](https://github.com/cockroachdb/cockroach/commits/5ac733bb4927020bc1c52da24b2591742fde8e1f):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | 2021-12-30 06:16:13.449 UTC [51] LOG:  received fast shutdown request
        postgres_1    | waiting for server to shut down....2021-12-30 06:16:13.451 UTC [51] LOG:  aborting any active transactions
        postgres_1    | 2021-12-30 06:16:13.452 UTC [51] LOG:  background worker "logical replication launcher" (PID 58) exited with exit code 1
        postgres_1    | 2021-12-30 06:16:13.454 UTC [53] LOG:  shutting down
        postgres_1    | 2021-12-30 06:16:13.549 UTC [51] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2021-12-30 06:16:13.671 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2021-12-30 06:16:13.671 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2021-12-30 06:16:13.671 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2021-12-30 06:16:13.675 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2021-12-30 06:16:13.680 UTC [70] LOG:  database system was shut down at 2021-12-30 06:16:13 UTC
        postgres_1    | 2021-12-30 06:16:13.685 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2021-12-30 06:26:16.466 UTC [100] ERROR:  syntax error at or near "DESC" at character 453
        postgres_1    | 2021-12-30 06:26:16.466 UTC [100] STATEMENT:  CREATE TABLE table1 (col1_0 TIMESTAMP NULL, col1_1 GEOMETRY, col1_2 BOOL NULL, col1_3 INT2 NULL, col1_4 REGPROC NULL, col1_5 VARCHAR, col1_6 REGPROC NOT NULL, col1_7 REGNAMESPACE NOT NULL, col1_8 TIME NULL, col1_9 CHAR NOT NULL, col1_10 REGPROC NOT NULL, col1_11 FLOAT8, col1_12 INT2 NULL GENERATED ALWAYS AS (col1_3 + (-4693)::INT8) STORED, col1_13 FLOAT8 GENERATED ALWAYS AS (col1_11 + 0.5595726132591788::FLOAT8) STORED, PRIMARY KEY (col1_9, col1_6 DESC), UNIQUE (col1_4));
        postgres_1    | 	CREATE TABLE table2 (col2_0 TIMESTAMP, col2_1 BYTEA NOT NULL, col2_2 GEOMETRY, col2_3 BYTEA NOT NULL, col2_4 NAME, col2_5 REGPROC, col2_6 REGROLE NULL, col2_7 FLOAT8 NULL, col2_8 OID, col2_9 TIMESTAMPTZ, col2_10 TIMESTAMPTZ NOT NULL, col2_11 UUID NOT NULL, col2_12 DECIMAL NULL, col2_13 TIMESTAMPTZ NOT NULL, col2_14 JSONB NULL, col2_15 TIMETZ[], col2_16 CHAR NOT NULL, col2_17 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col2_6 AS TEXT))) STORED, col2_18 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_11 AS TEXT))) STORED);
        postgres_1    | 	ALTER TABLE table2 ADD FOREIGN KEY (col2_5) REFERENCES table1 (col1_4) ON UPDATE CASCADE;
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('uwhhxl', 'whvstb', 'puerg');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (605.35s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.28s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 TIMESTAMP NULL, col1_1 GEOMETRY, col1_2 BOOL NULL, col1_3 INT2 NULL, col1_4 REGPROC NULL, col1_5 VARCHAR, col1_6 REGPROC NOT NULL, col1_7 REGNAMESPACE NOT NULL, col1_8 TIME NULL, col1_9 CHAR NOT NULL, col1_10 REGPROC NOT NULL, col1_11 FLOAT8, col1_12 INT2 NULL GENERATED ALWAYS AS (col1_3 + (-4693)::INT8) STORED, col1_13 FLOAT8 GENERATED ALWAYS AS (col1_11 + 0.5595726132591788::FLOAT8) STORED, PRIMARY KEY (col1_9, col1_6 DESC), UNIQUE (col1_4));
        test_1        |             CREATE TABLE table2 (col2_0 TIMESTAMP, col2_1 BYTEA NOT NULL, col2_2 GEOMETRY, col2_3 BYTEA NOT NULL, col2_4 NAME, col2_5 REGPROC, col2_6 REGROLE NULL, col2_7 FLOAT8 NULL, col2_8 OID, col2_9 TIMESTAMPTZ, col2_10 TIMESTAMPTZ NOT NULL, col2_11 UUID NOT NULL, col2_12 DECIMAL NULL, col2_13 TIMESTAMPTZ NOT NULL, col2_14 JSONB NULL, col2_15 TIMETZ[], col2_16 CHAR NOT NULL, col2_17 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col2_6 AS TEXT))) STORED, col2_18 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col2_11 AS TEXT))) STORED);
        test_1        |             ALTER TABLE table2 ADD FOREIGN KEY (col2_5) REFERENCES table1 (col1_4) ON UPDATE CASCADE;
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('uwhhxl', 'whvstb', 'puerg');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: syntax error at or near "DESC" (SQLSTATE 42601)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4017237&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4017237&tab=artifacts#/) on master @ [3471dc50db8406f3faa638940492dbe0cab53435](https://github.com/cockroachdb/cockroach/commits/3471dc50db8406f3faa638940492dbe0cab53435):


```
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (158.67s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	nil,
        test_1        |             + 	string("BOX(-90 45,-45 90)"),
        test_1        |               	nil,
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_box2dfromgeohash('F':::STRING::STRING::STRING, NULL::INT4::INT4)::BOX2D AS col_204456,
        test_1        |             			NULL::BOX2D AS col_204457
        test_1        |             	)
        test_1        |             		AS tab_121473
        test_1        |             ORDER BY
        test_1        |             	CAST(col_204456 AS STRING) NULLS FIRST, CAST(col_204457 AS STRING) NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_box2dfromgeohash('F'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::BOX2D AS col_204456,
        test_1        |             			NULL::BOX2D AS col_204457
        test_1        |             	)
        test_1        |             		AS tab_121473
        test_1        |             ORDER BY
        test_1        |             	CAST(col_204456 AS TEXT) NULLS FIRST, CAST(col_204457 AS TEXT) NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4022206&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4022206&tab=artifacts#/) on master @ [fa9480a8b661b31a3c5e4c3428595b3e7c320700](https://github.com/cockroachdb/cockroach/commits/fa9480a8b661b31a3c5e4c3428595b3e7c320700):


```
        postgres_1    | 2022-01-01 06:26:08.706 UTC [78] HINT:  Use an explicit ordering operator or modify the query.
        postgres_1    | 2022-01-01 06:26:08.706 UTC [78] STATEMENT:  SELECT
        postgres_1    | 		*
        postgres_1    | 	FROM
        postgres_1    | 		(SELECT NULL::BOX2D AS col_789694, NULL::VOID AS col_789695) AS tab_421707
        postgres_1    | 	ORDER BY
        postgres_1    | 		CAST(col_789694 AS TEXT) NULLS FIRST, col_789695 NULLS FIRST
        test_1        | --- FAIL: TestCompare (1046.02s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/mutators (441.46s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: cockroach2
        test_1        |         compare_test.go:193: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			tab_201642.col3_0 AS col_333488, (-1.0):::FLOAT8 AS col_333489
        test_1        |             		FROM
        test_1        |             			postgres.public.table4@table4_pkey AS tab_201641
        test_1        |             			JOIN postgres.public.table3@table3_pkey AS tab_201642 ON
        test_1        |             					(tab_201641.col4_9) = (tab_201642.tableoid) AND (tab_201641.col4_13) = (tab_201642.col3_5),
        test_1        |             			postgres.public.table3@[0] AS tab_201643
        test_1        |             			JOIN postgres.public.table4 AS tab_201644
        test_1        |             				JOIN postgres.public.table4 AS tab_201645 ON
        test_1        |             						(tab_201644.col4_3) = (tab_201645.col4_3)
        test_1        |             						AND (tab_201644.col4_17) = (tab_201645.col4_17)
        test_1        |             						AND (tab_201644.col4_6) = (tab_201645.col4_6)
        test_1        |             						AND (tab_201644.col4_15) = (tab_201645.col4_15)
        test_1        |             						AND (tab_201644.col4_16) = (tab_201645.col4_16)
        test_1        |             						AND (tab_201644.col4_5) = (tab_201645.col4_5) ON (tab_201643.col3_5) = (tab_201644.col4_11)
        test_1        |             		ORDER BY
        test_1        |             			tab_201642.col3_0, tab_201645.col4_11 ASC, tab_201641.col4_7 ASC, tab_201642.col3_4 DESC
        test_1        |             	)
        test_1        |             		AS tab_201646
        test_1        |             ORDER BY
        test_1        |             	col_333488 NULLS FIRST, col_333489 NULLS FIRST
        test_1        |         compare_test.go:194: cockroach1: ping: conn closed
        test_1        | FAIL
        compare_cockroach1_1 exited with code 2
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4026853&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4026853&tab=artifacts#/) on master @ [65c6377f4f486d537ef7dfc59f8f9c78f1a6018a](https://github.com/cockroachdb/cockroach/commits/65c6377f4f486d537ef7dfc59f8f9c78f1a6018a):


```
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('qnwy', 'kcacy');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('jchuk', 'n', 'm', 'mmfs', 'ffp', 'iqi');
        postgres_1    | 	CREATE TYPE rand_typ_2 AS ENUM ('zbup', 'celro', 'pwdf', 'h', 'prtt');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (609.28s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (5.37s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 INT8 NOT NULL, col1_1 REGNAMESPACE NOT NULL, col1_2 INT4 NULL, col1_3 TIMETZ NULL, col1_4 "char" NULL, col1_5 REGNAMESPACE, col1_6 UUID NOT NULL, col1_7 INT8 NOT NULL GENERATED ALWAYS AS (col1_0 + col1_2) STORED, PRIMARY KEY (col1_1, col1_7, col1_0));
        test_1        |             CREATE INDEX ON table1 (col1_2, (col1_0 + col1_2) DESC, col1_1 ASC, col1_3 ASC, col1_6 DESC, lower(CAST(col1_1 AS TEXT)), col1_0 ASC) INCLUDE (col1_4);
        test_1        |             CREATE INDEX ON table1 (col1_5 DESC, col1_4 DESC, col1_6 DESC, col1_2 ASC, col1_0 ASC);
        test_1        |             CREATE INDEX ON table1 (col1_0 ASC, col1_1 ASC, col1_5 DESC, col1_4 DESC, col1_3 ASC) INCLUDE (col1_6);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_4);
        test_1        |             CREATE TABLE table2 (col2_0 OID NULL, col2_1 BIT(11) NOT NULL, col2_2 "char", col2_3 DATE NOT NULL, col2_4 REGROLE, col2_5 TIMESTAMPTZ NOT NULL, col2_6 NAME NULL, col2_7 TEXT NOT NULL, col2_8 REGTYPE NOT NULL, col2_9 REGPROCEDURE NOT NULL, col2_10 FLOAT8 NOT NULL, col2_11 INET NOT NULL, col2_12 INT4, col2_13 "char" NOT NULL, col2_14 BIT NULL, col2_15 BYTEA NULL);
        test_1        |             CREATE INDEX ON table2 (lower(CAST(col2_3 AS TEXT)), col2_4 DESC, col2_14 DESC, col2_5 ASC, col2_2, col2_13, col2_9 DESC, col2_3 ASC, col2_1);
        test_1        |             CREATE INDEX ON table2 (lower(CAST(col2_4 AS TEXT)), col2_9 ASC, col2_14 ASC, col2_7 DESC, col2_12 DESC, col2_1, col2_10 ASC, lower(col2_2) ASC, lower(CAST(col2_8 AS TEXT)), col2_13, col2_15 DESC, col2_5 DESC, col2_4, col2_11 DESC, col2_8, col2_2 ASC);
        test_1        |             CREATE INDEX ON table2 (col2_13 DESC, col2_6 ASC, lower(col2_6), col2_14 DESC, col2_2 ASC, col2_11 DESC, col2_1 ASC, col2_0 ASC, col2_9 ASC, col2_15 ASC, col2_8 DESC, col2_12, col2_3, col2_10 DESC, col2_4, col2_5);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_2 DESC, col2_1, col2_10 DESC, col2_4 DESC, col2_5, col2_3 DESC, col2_0, col2_6, col2_12, col2_15, col2_7 ASC, col2_11 ASC, col2_8 DESC, (CASE WHEN col2_5 IS NULL THEN e'\x134kB\x19p'::TEXT ELSE ''::TEXT END) ASC, col2_13) INCLUDE (col2_14);
        test_1        |             CREATE INDEX ON table2 (col2_10 ASC, col2_4 ASC, col2_7 DESC, col2_6 ASC, lower(CAST(col2_8 AS TEXT)), col2_0 DESC) INCLUDE (col2_5, col2_8, col2_12);
        test_1        |             CREATE TABLE table3 (col3_0 FLOAT8 NOT NULL, col3_1 REGTYPE, col3_2 DATE NULL, col3_3 GEOGRAPHY NOT NULL, col3_4 REGCLASS NULL, col3_5 TEXT NOT NULL, col3_6 GEOGRAPHY NOT NULL, col3_7 REGNAMESPACE, col3_8 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col3_4 IS NULL THEN 'h'::TEXT ELSE 'kGUbr'::TEXT END) STORED, col3_9 TEXT NOT NULL GENERATED ALWAYS AS (lower(col3_5)) STORED, col3_10 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col3_2 IS NULL THEN e'\x1d\x11z%=\x13'::TEXT ELSE e'i\x05SJ\tSGV'::TEXT END) STORED, col3_11 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col3_2 IS NULL THEN '?6op6'::TEXT ELSE '"'::TEXT END) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_9 DESC) INCLUDE (col3_1, col3_2, col3_3, col3_4);
        test_1        |             CREATE TABLE table4 (col4_0 CHAR NOT NULL, col4_1 FLOAT8 NOT NULL, col4_2 "char" NOT NULL, col4_3 TIME NOT NULL, col4_4 REGCLASS, col4_5 GEOGRAPHY NULL, col4_6 TEXT, col4_7 NAME NOT NULL, col4_8 "char" NULL, col4_9 "char", col4_10 INT8 NULL, col4_11 DECIMAL NULL, col4_12 UUID NOT NULL, col4_13 REGPROCEDURE, col4_14 FLOAT8 NOT NULL GENERATED ALWAYS AS (col4_1 + 1.3162511875153093::FLOAT8) STORED, col4_15 TEXT GENERATED ALWAYS AS (CASE WHEN col4_4 IS NULL THEN e'\x02'::TEXT ELSE e'w\x19C&z\tw/2'::TEXT END) STORED, col4_16 TEXT GENERATED ALWAYS AS (lower(col4_9)) STORED, col4_17 TEXT GENERATED ALWAYS AS (lower(col4_9)) STORED);
        test_1        |             CREATE INDEX ON table4 (col4_0 DESC, col4_9 ASC, lower(col4_9));
        test_1        |             CREATE INDEX ON table4 (col4_12 DESC, col4_4 ASC, col4_8, col4_11, col4_17 DESC, col4_7 DESC, col4_10 DESC, col4_15, col4_9 DESC, (col4_11 + 284485.3058136902601::DECIMAL), col4_13);
        test_1        |             CREATE TABLE table5 (col5_0 VARCHAR, col5_1 BOOL);
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_1 DESC, col5_0 ASC);
        test_1        |             CREATE INDEX ON table5 (lower(col5_0) ASC, col5_0 ASC);
        test_1        |             CREATE INDEX ON table5 (col5_1 DESC, col5_0 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table5 (lower(col5_0), col5_0 ASC);
        test_1        |             ALTER TABLE table2 ADD FOREIGN KEY (col2_6) REFERENCES table1 (col1_4) ON UPDATE RESTRICT;
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('qnwy', 'kcacy');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('jchuk', 'n', 'm', 'mmfs', 'ffp', 'iqi');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('zbup', 'celro', 'pwdf', 'h', 'prtt');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4033020&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4033020&tab=artifacts#/) on master @ [e3be114fc6fe8eda07aafbc7ee43cc452456f270](https://github.com/cockroachdb/cockroach/commits/e3be114fc6fe8eda07aafbc7ee43cc452456f270):


```
        test_1        | --- FAIL: TestCompare (3.24s)
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (0.00s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 54 [running]:
        test_1        | testing.tRunner.func1.2({0x1a5b8c0, 0xc000d854e0})
        test_1        | 	/usr/local/go/src/testing/testing.go:1209 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	/usr/local/go/src/testing/testing.go:1212 +0x218
        test_1        | panic({0x1a5b8c0, 0xc000d854e0})
        test_1        | 	/usr/local/go/src/runtime/panic.go:1038 +0x215
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x3756680, {0x2787dc0, 0xc000da92c0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:368 +0x467
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x2715be0, 0x3756680)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:301 +0x14c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc001fa7980)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:220 +0x92
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc000da8c30, {0xc000db6900, 0x4, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:251 +0xa4d
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0x90, 0x1bc344b, {0xc000db6900, 0x8e, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:97 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x203000, {0x1bc344b, 0x5}, 0x4, {0xc000bb5a30, 0x3, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:85 +0x157
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x272a960, 0xc0000f8000)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:87 +0x187
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0x5b33f3ef97c9e4e0)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:74 +0x33
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x1bc88af)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:44 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc001fb3520)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:128 +0xe5
        test_1        | testing.tRunner(0xc001fb3520, 0xc000da8ba0)
        test_1        | 	/usr/local/go/src/testing/testing.go:1259 +0x102
        test_1        | created by testing.(*T).Run
        test_1        | 	/usr/local/go/src/testing/testing.go:1306 +0x35a
        compare_test_1 exited with code 2
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 2
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4042022&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4042022&tab=artifacts#/) on master @ [260144b9bec2b2da2cf96c7198197a88b97c942c](https://github.com/cockroachdb/cockroach/commits/260144b9bec2b2da2cf96c7198197a88b97c942c):


```
        test_1        | --- FAIL: TestCompare (614.26s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (10.26s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	nil,
        test_1        |             + 	[]uint8{0x01, 0x08, 0x21, 0x0c, 0x12, 0x02, 0x48},
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(st_astwkb(st_makepoint(6.0:::FLOAT8::FLOAT8::FLOAT8, 9.0:::FLOAT8::FLOAT8::FLOAT8, 1.0:::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, 0:::INT8::INT4::INT4, NULL::INT4::INT4, 9:::INT8::INT4::INT4)::BYTES::BYTES || '\x48':::BYTES::BYTES::BYTES)::BYTES
        test_1        |             				AS col_9517
        test_1        |             	)
        test_1        |             		AS tab_5547
        test_1        |             ORDER BY
        test_1        |             	col_9517 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(st_astwkb(st_makepoint(6.0::FLOAT8::FLOAT8::FLOAT8, 9.0::FLOAT8::FLOAT8::FLOAT8, 1.0::FLOAT8::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY, 0::INT8::INT4::INT4, NULL::INT4::INT4, 9::INT8::INT4::INT4)::BYTEA::BYTEA || '\x48'::BYTEA::BYTEA::BYTEA)::BYTEA
        test_1        |             				AS col_9517
        test_1        |             	)
        test_1        |             		AS tab_5547
        test_1        |             ORDER BY
        test_1        |             	col_9517 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4051272&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4051272&tab=artifacts#/) on master @ [3401e4d902953f7684aebb819090f586c9acfec9](https://github.com/cockroachdb/cockroach/commits/3401e4d902953f7684aebb819090f586c9acfec9):


```
        postgres_1    | 				JOIN postgres.public.table1 AS tab_305101 ON (tab_305100.col1_6) = (tab_305101.col1_6)
        postgres_1    | 				FULL JOIN postgres.public.table1 AS tab_305102 ON false::BOOL
        postgres_1    | 			WHERE
        postgres_1    | 				NULL::BOOL
        postgres_1    | 			ORDER BY
        postgres_1    | 				tab_305102.col1_1 ASC, tab_305101.col1_2 ASC
        postgres_1    | 		)
        postgres_1    | 			AS tab_305103
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_536773 NULLS FIRST, col_536774 NULLS FIRST, col_536775 NULLS FIRST
        test_1        | --- FAIL: TestCompare (1002.91s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (398.81s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	nil,
        test_1        |             + 	string("BOX(-90 45,-45 90)"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_box2dfromgeohash('F':::STRING::STRING::STRING, NULL::INT4::INT4)::BOX2D AS col_537022) AS tab_305243
        test_1        |             ORDER BY
        test_1        |             	CAST(col_537022 AS STRING) NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_box2dfromgeohash('F'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::BOX2D AS col_537022) AS tab_305243
        test_1        |             ORDER BY
        test_1        |             	CAST(col_537022 AS TEXT) NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4061808&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4061808&tab=artifacts#/) on master @ [d191f64e5600b773043da18d28e288bf6ae86b72](https://github.com/cockroachdb/cockroach/commits/d191f64e5600b773043da18d28e288bf6ae86b72):


```
        postgres_1    | 2022-01-07 06:16:09.881 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-01-07 06:16:09.886 UTC [70] LOG:  database system was shut down at 2022-01-07 06:16:09 UTC
        postgres_1    | 2022-01-07 06:16:09.891 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-01-07 06:16:13.076 UTC [80] ERROR:  invalid byte sequence for encoding "UTF8": 0x00
        postgres_1    | 2022-01-07 06:16:13.076 UTC [80] STATEMENT:  CREATE TABLE table1 (col1_0 DECIMAL NULL, col1_1 BYTEA NOT NULL, col1_2 REGPROCEDURE NOT NULL, col1_3 REGTYPE, col1_4 TEXT NULL, col1_5 INET NOT NULL, col1_6 TIMESTAMPTZ NULL, col1_7 GEOMETRY NOT NULL, col1_8 "char", col1_9 REGCLASS NULL, col1_10 BYTEA NOT NULL, col1_11 REGPROCEDURE NOT NULL, col1_12 REGPROCEDURE NOT NULL, col1_13 DATE NULL, col1_14 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_10 IS NULL THEN e'w0]xL7^\x1e'::TEXT ELSE e'c\x17 '::TEXT END) STORED, col1_15 TEXT GENERATED ALWAYS AS (lower(col1_8)) STORED, col1_16 TEXT NULL GENERATED ALWAYS AS (lower(col1_4)) STORED);
        postgres_1    | 	CREATE TABLE table2 (col2_0 CHAR, col2_1 TIMESTAMP NOT NULL, col2_2 TIMESTAMPTZ, col2_3 TIMESTAMPTZ, col2_4 DATE NULL, col2_5 TIMETZ NULL, col2_6 REGPROCEDURE NOT NULL, col2_7 INT8 NOT NULL, PRIMARY KEY (col2_7));
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_1 DESC, col2_2);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_3 ASC, col2_6, col2_2 ASC, col2_7 ASC, col2_0 DESC, (CASE WHEN col2_1 IS NULL THEN e'8\x01\x02'::TEXT ELSE e'\x1betzE\b\x7f'::TEXT END) DESC, col2_4, col2_5 DESC);
        postgres_1    | 	CREATE INDEX ON table2 (col2_0 DESC, col2_2, col2_4, col2_5);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 ((CASE WHEN col2_4 IS NULL THEN e'u\x11'::TEXT ELSE e'\x15'::TEXT END), col2_3, col2_1 ASC, col2_6 DESC, col2_2 ASC, col2_4, col2_7 ASC);
        postgres_1    | 	CREATE INDEX ON table2 ((CASE WHEN col2_6 IS NULL THEN e'\x00'::TEXT ELSE '!T]!_RM'::TEXT END) ASC, col2_5 ASC, col2_2 ASC, col2_1 DESC) INCLUDE (col2_3, col2_4);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_0 DESC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_3, col2_2 DESC, col2_5 ASC, col2_4, col2_7, col2_0, col2_6 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_4, col2_6, col2_0);
        postgres_1    | 	ALTER TABLE table1 ADD FOREIGN KEY (col1_13, col1_9, col1_4) REFERENCES table2 (col2_4, col2_6, col2_0);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('lkpwg', 'pnf', 'ulhuzb', 'l');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.69s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (2.72s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 DECIMAL NULL, col1_1 BYTEA NOT NULL, col1_2 REGPROCEDURE NOT NULL, col1_3 REGTYPE, col1_4 TEXT NULL, col1_5 INET NOT NULL, col1_6 TIMESTAMPTZ NULL, col1_7 GEOMETRY NOT NULL, col1_8 "char", col1_9 REGCLASS NULL, col1_10 BYTEA NOT NULL, col1_11 REGPROCEDURE NOT NULL, col1_12 REGPROCEDURE NOT NULL, col1_13 DATE NULL, col1_14 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_10 IS NULL THEN e'w0]xL7^\x1e'::TEXT ELSE e'c\x17 '::TEXT END) STORED, col1_15 TEXT GENERATED ALWAYS AS (lower(col1_8)) STORED, col1_16 TEXT NULL GENERATED ALWAYS AS (lower(col1_4)) STORED);
        test_1        |             CREATE TABLE table2 (col2_0 CHAR, col2_1 TIMESTAMP NOT NULL, col2_2 TIMESTAMPTZ, col2_3 TIMESTAMPTZ, col2_4 DATE NULL, col2_5 TIMETZ NULL, col2_6 REGPROCEDURE NOT NULL, col2_7 INT8 NOT NULL, PRIMARY KEY (col2_7));
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_1 DESC, col2_2);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_3 ASC, col2_6, col2_2 ASC, col2_7 ASC, col2_0 DESC, (CASE WHEN col2_1 IS NULL THEN e'8\x01\x02'::TEXT ELSE e'\x1betzE\b\x7f'::TEXT END) DESC, col2_4, col2_5 DESC);
        test_1        |             CREATE INDEX ON table2 (col2_0 DESC, col2_2, col2_4, col2_5);
        test_1        |             CREATE UNIQUE INDEX ON table2 ((CASE WHEN col2_4 IS NULL THEN e'u\x11'::TEXT ELSE e'\x15'::TEXT END), col2_3, col2_1 ASC, col2_6 DESC, col2_2 ASC, col2_4, col2_7 ASC);
        test_1        |             CREATE INDEX ON table2 ((CASE WHEN col2_6 IS NULL THEN e'\x00'::TEXT ELSE '!T]!_RM'::TEXT END) ASC, col2_5 ASC, col2_2 ASC, col2_1 DESC) INCLUDE (col2_3, col2_4);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_0 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_3, col2_2 DESC, col2_5 ASC, col2_4, col2_7, col2_0, col2_6 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_4, col2_6, col2_0);
        test_1        |             ALTER TABLE table1 ADD FOREIGN KEY (col1_13, col1_9, col1_4) REFERENCES table2 (col2_4, col2_6, col2_0);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('lkpwg', 'pnf', 'ulhuzb', 'l');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4070351&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4070351&tab=artifacts#/) on master @ [d66c14663e027a5d9699b8a15e3d6f5821877ef5](https://github.com/cockroachdb/cockroach/commits/d66c14663e027a5d9699b8a15e3d6f5821877ef5):


```
        test_1        |     --- FAIL: TestCompare/postgres (320.93s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |               	s"2000-01-01 02:00:00 +0000 UTC",
        test_1        |             - 	string("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffffC"),
        test_1        |             + 	string("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff/128C"),
        test_1        |               	nil,
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			'2000-01-01 02:00:00':::TIMESTAMP::TIMESTAMP AS col_407870,
        test_1        |             			(hostmask('::4/0':::INET::INET::INET)::INET::INET || 'C':::STRING::STRING::STRING)::STRING AS col_407871,
        test_1        |             			NULL::OIDVECTOR AS col_407872
        test_1        |             	)
        test_1        |             		AS tab_236202
        test_1        |             ORDER BY
        test_1        |             	col_407870 NULLS FIRST, col_407871 NULLS FIRST, col_407872 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			'2000-01-01 02:00:00'::TIMESTAMP::TIMESTAMP AS col_407870,
        test_1        |             			(hostmask('::4/0'::INET::INET::INET)::INET::INET || 'C'::TEXT::TEXT::TEXT)::TEXT AS col_407871,
        test_1        |             			NULL::OIDVECTOR AS col_407872
        test_1        |             	)
        test_1        |             		AS tab_236202
        test_1        |             ORDER BY
        test_1        |             	col_407870 NULLS FIRST, col_407871 NULLS FIRST, col_407872 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4075133&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4075133&tab=artifacts#/) on master @ [6ff6c77e7aebec2edc947c1ed99a1f7cb67e1a3c](https://github.com/cockroachdb/cockroach/commits/6ff6c77e7aebec2edc947c1ed99a1f7cb67e1a3c):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | 2022-01-09 06:13:51.539 UTC [49] LOG:  received fast shutdown request
        postgres_1    | waiting for server to shut down....2022-01-09 06:13:51.540 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2022-01-09 06:13:51.542 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2022-01-09 06:13:51.544 UTC [51] LOG:  shutting down
        postgres_1    | 2022-01-09 06:13:51.632 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2022-01-09 06:13:51.659 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2022-01-09 06:13:51.659 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2022-01-09 06:13:51.660 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2022-01-09 06:13:51.663 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-01-09 06:13:51.668 UTC [68] LOG:  database system was shut down at 2022-01-09 06:13:51 UTC
        postgres_1    | 2022-01-09 06:13:51.672 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-01-09 06:13:53.062 UTC [78] ERROR:  syntax error at or near "(" at character 513
        postgres_1    | 2022-01-09 06:13:53.062 UTC [78] STATEMENT:  CREATE TABLE table1 (col1_0 GEOMETRY, col1_1 BIT(11), col1_2 OID NULL, col1_3 CHAR NULL, col1_4 INET, col1_5 INT8 NOT NULL, col1_6 TIME NOT NULL, col1_7 GEOMETRY, col1_8 INET NOT NULL, col1_9 DATE NULL, col1_10 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col1_9 AS TEXT))) STORED, col1_11 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col1_9 AS TEXT))) STORED, col1_12 TEXT GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, col1_13 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_6 AS TEXT))) STORED, INDEX ((col1_5 + (-1049093550623097344)::INT8) ASC, col1_13), INVERTED INDEX (col1_4 ASC, col1_8 ASC, col1_7), INDEX (lower(CAST(col1_6 AS TEXT)) DESC, col1_8, col1_9 DESC) INCLUDE (col1_1, col1_3, col1_5, col1_6, col1_7, col1_10, col1_11) WHERE (((((((table1.col1_10 <= e'\''::TEXT) AND (table1.col1_12 <= 'X'::TEXT)) AND (table1.col1_6 >= '00:00:00'::TIME)) AND (table1.col1_13 <= 'X'::TEXT)) OR (table1.col1_3 <= e'\''::TEXT)) AND (table1.col1_9 = '1970-01-01'::DATE)) AND (table1.col1_5 < (-1)::INT8)) OR (table1.col1_11 > e'\x00'::TEXT));
        postgres_1    | 	CREATE TABLE table2 (col2_0 TIMETZ NULL, col2_1 GEOGRAPHY NOT NULL, col2_2 TIME, col2_3 JSONB, col2_4 REGTYPE NULL, col2_5 BOX2D NULL, col2_6 TEXT NULL, col2_7 REGCLASS NULL, col2_8 TIMETZ NOT NULL, col2_9 GEOGRAPHY NOT NULL, col2_10 VARCHAR NULL, col2_11 REGROLE NOT NULL, col2_12 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col2_4 IS NULL THEN 'fo'::TEXT ELSE 'H'::TEXT END) STORED, col2_13 TEXT NULL GENERATED ALWAYS AS (lower(col2_10)) STORED, col2_14 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col2_4 IS NULL THEN NULL ELSE 'bS'::TEXT END) STORED, col2_15 TEXT NULL GENERATED ALWAYS AS (lower(col2_6)) STORED, col2_16 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_11 IS NULL THEN '"'::TEXT ELSE ''::TEXT END) STORED);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('sbd', 'iukh', 'vbaj', 'esz');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (604.43s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.79s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 GEOMETRY, col1_1 BIT(11), col1_2 OID NULL, col1_3 CHAR NULL, col1_4 INET, col1_5 INT8 NOT NULL, col1_6 TIME NOT NULL, col1_7 GEOMETRY, col1_8 INET NOT NULL, col1_9 DATE NULL, col1_10 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col1_9 AS TEXT))) STORED, col1_11 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col1_9 AS TEXT))) STORED, col1_12 TEXT GENERATED ALWAYS AS (lower(CAST(col1_1 AS TEXT))) STORED, col1_13 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_6 AS TEXT))) STORED, INDEX ((col1_5 + (-1049093550623097344)::INT8) ASC, col1_13), INVERTED INDEX (col1_4 ASC, col1_8 ASC, col1_7), INDEX (lower(CAST(col1_6 AS TEXT)) DESC, col1_8, col1_9 DESC) INCLUDE (col1_1, col1_3, col1_5, col1_6, col1_7, col1_10, col1_11) WHERE (((((((table1.col1_10 <= e'\''::TEXT) AND (table1.col1_12 <= 'X'::TEXT)) AND (table1.col1_6 >= '00:00:00'::TIME)) AND (table1.col1_13 <= 'X'::TEXT)) OR (table1.col1_3 <= e'\''::TEXT)) AND (table1.col1_9 = '1970-01-01'::DATE)) AND (table1.col1_5 < (-1)::INT8)) OR (table1.col1_11 > e'\x00'::TEXT));
        test_1        |             CREATE TABLE table2 (col2_0 TIMETZ NULL, col2_1 GEOGRAPHY NOT NULL, col2_2 TIME, col2_3 JSONB, col2_4 REGTYPE NULL, col2_5 BOX2D NULL, col2_6 TEXT NULL, col2_7 REGCLASS NULL, col2_8 TIMETZ NOT NULL, col2_9 GEOGRAPHY NOT NULL, col2_10 VARCHAR NULL, col2_11 REGROLE NOT NULL, col2_12 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col2_4 IS NULL THEN 'fo'::TEXT ELSE 'H'::TEXT END) STORED, col2_13 TEXT NULL GENERATED ALWAYS AS (lower(col2_10)) STORED, col2_14 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col2_4 IS NULL THEN NULL ELSE 'bS'::TEXT END) STORED, col2_15 TEXT NULL GENERATED ALWAYS AS (lower(col2_6)) STORED, col2_16 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_11 IS NULL THEN '"'::TEXT ELSE ''::TEXT END) STORED);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('sbd', 'iukh', 'vbaj', 'esz');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: syntax error at or near "(" (SQLSTATE 42601)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4080316&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4080316&tab=artifacts#/) on master @ [62392dcd0f03a2bef1dc6d0431e505f5125297e8](https://github.com/cockroachdb/cockroach/commits/62392dcd0f03a2bef1dc6d0431e505f5125297e8):


```
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_6 ASC, col2_9 ASC, col2_0 ASC, col2_16, col2_13 DESC) INCLUDE (col2_1, col2_2, col2_3, col2_11);
        postgres_1    | 	CREATE UNIQUE INDEX ON table2 (col2_16 ASC, col2_8 ASC, col2_13 DESC, col2_9 ASC);
        postgres_1    | 	CREATE INDEX ON table2 (col2_0 DESC, col2_5 ASC, col2_17) INCLUDE (col2_4, col2_6, col2_10, col2_14, col2_16);
        postgres_1    | 	CREATE TABLE table3 (col3_0 REGNAMESPACE, col3_1 OID NOT NULL, col3_2 INTERVAL NULL, col3_3 REGPROC NOT NULL, col3_4 GEOGRAPHY NOT NULL, col3_5 INT2 NULL, col3_6 CHAR, col3_7 TIMETZ NOT NULL, col3_8 DECIMAL NOT NULL, col3_9 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_3 IS NULL THEN e'\x0f\x01\x0f\x0bD\x12=V'::TEXT ELSE e'\x19\''::TEXT END) STORED, col3_10 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_7 AS TEXT))) STORED, col3_11 DECIMAL NOT NULL GENERATED ALWAYS AS (col3_8 + 'Infinity'::DECIMAL) STORED, col3_12 INT2 NULL GENERATED ALWAYS AS (col3_5 + NULL) STORED, col3_13 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col3_2 AS TEXT))) STORED, col3_14 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_7 AS TEXT))) STORED, col3_15 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_4 AS TEXT))) STORED);
        postgres_1    | 	CREATE UNIQUE INDEX ON table3 (col3_0 ASC, col3_7, col3_14, col3_6, col3_2, col3_15, col3_11) INCLUDE (col3_3, col3_4, col3_10);
        postgres_1    | 	CREATE INDEX ON table3 (col3_12, col3_10 ASC, col3_0 DESC, col3_9 ASC, col3_8, col3_1, col3_6 DESC, col3_14 DESC, col3_11, col3_5 DESC, (CASE WHEN col3_3 IS NULL THEN e'F\t\bMo'::TEXT ELSE e'R\''::TEXT END) ASC, col3_7, (CASE WHEN col3_3 IS NULL THEN ''::TEXT ELSE e'\x18 '::TEXT END) DESC, (CASE WHEN col3_3 IS NULL THEN '"'::TEXT ELSE e'0F\bW]>6<'::TEXT END) DESC);
        postgres_1    | 	CREATE INDEX ON table3 (col3_2 ASC, col3_9 DESC, col3_8 ASC, col3_12 ASC, col3_10 ASC, col3_3, col3_14 DESC, col3_15, col3_5, col3_13, (col3_8 + (-1)::DECIMAL)) INCLUDE (col3_0, col3_1, col3_4, col3_7);
        postgres_1    | 	CREATE UNIQUE INDEX ON table3 (col3_3, col3_15 DESC, col3_11 DESC, col3_7 ASC, col3_6);
        postgres_1    | 	CREATE UNIQUE INDEX ON table3 (col3_1, col3_11 ASC);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('v', 'hruuhp');
        postgres_1    | 	CREATE TYPE rand_typ_1 AS ENUM ('nlc', 'fwkg');
        postgres_1    | 	CREATE TYPE rand_typ_2 AS ENUM ('zqn', 'dtkpw', 'qecwop', 'iqburi', 'qqqw', 'xucu');
        postgres_1    | 	CREATE TYPE rand_typ_3 AS ENUM ('xvmcmp', 'ggqp');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (606.58s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (2.54s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 INTERVAL NOT NULL, col1_1 TIMETZ NOT NULL, col1_2 GEOGRAPHY NOT NULL, col1_3 INT2, col1_4 DATE NULL, col1_5 TIME[] NOT NULL, col1_6 GEOGRAPHY NOT NULL, col1_7 INET NOT NULL, col1_8 OID NOT NULL, col1_9 REGCLASS NULL, col1_10 CHAR NULL, col1_11 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_7 AS TEXT))) STORED, col1_12 INT2 GENERATED ALWAYS AS (col1_3 + (-1)::INT8) STORED, col1_13 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_7 AS TEXT))) STORED, col1_14 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col1_4 IS NULL THEN '"'::TEXT ELSE e'\n\x03\x01FZ.\x15\x1f-'::TEXT END) STORED);
        test_1        |             CREATE TABLE table2 (col2_0 BYTEA, col2_1 REGROLE, col2_2 GEOGRAPHY NOT NULL, col2_3 REGTYPE NOT NULL, col2_4 REGCLASS NULL, col2_5 TIMESTAMPTZ, col2_6 TIMESTAMP, col2_7 REGNAMESPACE, col2_8 REGPROC NOT NULL, col2_9 DATE NULL, col2_10 INT2 NOT NULL, col2_11 REGPROC NOT NULL, col2_12 GEOMETRY NOT NULL, col2_13 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_8 IS NULL THEN e'\x010\x1087'::TEXT ELSE ''::TEXT END) STORED, col2_14 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col2_9 IS NULL THEN e'#\x022rH^et='::TEXT ELSE 'Tq'::TEXT END) STORED, col2_15 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col2_11 IS NULL THEN '&q'::TEXT ELSE e'\'2lu'::TEXT END) STORED, col2_16 INT2 NOT NULL GENERATED ALWAYS AS (col2_10 + (-31737)::INT8) STORED, col2_17 TEXT GENERATED ALWAYS AS (CASE WHEN col2_7 IS NULL THEN 'uV9T'::TEXT ELSE e'x\b)6eG\x14[F'::TEXT END) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_6 ASC, col2_9 ASC, col2_0 ASC, col2_16, col2_13 DESC) INCLUDE (col2_1, col2_2, col2_3, col2_11);
        test_1        |             CREATE UNIQUE INDEX ON table2 (col2_16 ASC, col2_8 ASC, col2_13 DESC, col2_9 ASC);
        test_1        |             CREATE INDEX ON table2 (col2_0 DESC, col2_5 ASC, col2_17) INCLUDE (col2_4, col2_6, col2_10, col2_14, col2_16);
        test_1        |             CREATE TABLE table3 (col3_0 REGNAMESPACE, col3_1 OID NOT NULL, col3_2 INTERVAL NULL, col3_3 REGPROC NOT NULL, col3_4 GEOGRAPHY NOT NULL, col3_5 INT2 NULL, col3_6 CHAR, col3_7 TIMETZ NOT NULL, col3_8 DECIMAL NOT NULL, col3_9 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col3_3 IS NULL THEN e'\x0f\x01\x0f\x0bD\x12=V'::TEXT ELSE e'\x19\''::TEXT END) STORED, col3_10 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_7 AS TEXT))) STORED, col3_11 DECIMAL NOT NULL GENERATED ALWAYS AS (col3_8 + 'Infinity'::DECIMAL) STORED, col3_12 INT2 NULL GENERATED ALWAYS AS (col3_5 + NULL) STORED, col3_13 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col3_2 AS TEXT))) STORED, col3_14 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_7 AS TEXT))) STORED, col3_15 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col3_4 AS TEXT))) STORED);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_0 ASC, col3_7, col3_14, col3_6, col3_2, col3_15, col3_11) INCLUDE (col3_3, col3_4, col3_10);
        test_1        |             CREATE INDEX ON table3 (col3_12, col3_10 ASC, col3_0 DESC, col3_9 ASC, col3_8, col3_1, col3_6 DESC, col3_14 DESC, col3_11, col3_5 DESC, (CASE WHEN col3_3 IS NULL THEN e'F\t\bMo'::TEXT ELSE e'R\''::TEXT END) ASC, col3_7, (CASE WHEN col3_3 IS NULL THEN ''::TEXT ELSE e'\x18 '::TEXT END) DESC, (CASE WHEN col3_3 IS NULL THEN '"'::TEXT ELSE e'0F\bW]>6<'::TEXT END) DESC);
        test_1        |             CREATE INDEX ON table3 (col3_2 ASC, col3_9 DESC, col3_8 ASC, col3_12 ASC, col3_10 ASC, col3_3, col3_14 DESC, col3_15, col3_5, col3_13, (col3_8 + (-1)::DECIMAL)) INCLUDE (col3_0, col3_1, col3_4, col3_7);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_3, col3_15 DESC, col3_11 DESC, col3_7 ASC, col3_6);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_1, col3_11 ASC);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('v', 'hruuhp');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('nlc', 'fwkg');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('zqn', 'dtkpw', 'qecwop', 'iqburi', 'qqqw', 'xucu');
        test_1        |             CREATE TYPE rand_typ_3 AS ENUM ('xvmcmp', 'ggqp');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: invalid input syntax for type numeric: "Infinity" (SQLSTATE 22P02)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4108362&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4108362&tab=artifacts#/) on master @ [c3d71ac887844bef174abb6dab2a4e1ce9270ab7](https://github.com/cockroachdb/cockroach/commits/c3d71ac887844bef174abb6dab2a4e1ce9270ab7):


```
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (609.99s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (5.06s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 UUID NOT NULL, col1_1 TEXT NOT NULL, col1_2 FLOAT4 NOT NULL, col1_3 NAME NOT NULL, col1_4 INT2 NOT NULL, col1_5 DATE NOT NULL, col1_6 FLOAT4 NOT NULL GENERATED ALWAYS AS (col1_2 + (-0.5271008610725403)::FLOAT8) STORED, PRIMARY KEY (col1_4, col1_1, col1_6, col1_3, col1_5, col1_2, col1_0));
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_0 DESC, lower(col1_1) DESC);
        test_1        |             CREATE INDEX ON table1 (col1_5, col1_2, col1_3, col1_6 ASC, col1_4);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_0 ASC, col1_3);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_3 DESC, col1_5, col1_0 DESC);
        test_1        |             CREATE INDEX ON table1 (col1_1, col1_2, col1_3 DESC, col1_6, col1_4 DESC, col1_5 DESC, col1_0 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_3 ASC, col1_2);
        test_1        |             CREATE INDEX ON table1 (col1_0 ASC, lower(col1_1) ASC, col1_2, col1_3 ASC, col1_5 DESC);
        test_1        |             CREATE INDEX ON table1 (col1_3, col1_0, lower(CAST(col1_5 AS TEXT)) DESC);
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_6, col1_1, col1_2 ASC, col1_5 DESC, col1_0 DESC, col1_4, col1_3 DESC);
        test_1        |             CREATE TABLE table2 (col2_0 REGNAMESPACE NOT NULL, col2_1 CHAR NOT NULL, col2_2 INET NOT NULL, col2_3 BOX2D NOT NULL, col2_4 BOX2D, col2_5 GEOMETRY, col2_6 INET NOT NULL);
        test_1        |             CREATE UNIQUE INDEX ON table2 ((CASE WHEN col2_0 IS NULL THEN 'vrg&s'::TEXT ELSE ''::TEXT END), col2_2 ASC) INCLUDE (col2_0, col2_5, col2_6);
        test_1        |             CREATE INDEX ON table2 (col2_2 ASC) INCLUDE (col2_0, col2_1, col2_5, col2_6);
        test_1        |             CREATE INDEX ON table2 (col2_2 DESC);
        test_1        |             CREATE TABLE table3 (col3_0 REGROLE, col3_1 GEOMETRY NOT NULL, col3_2 INET NULL, col3_3 REGPROC, col3_4 BIT(49) NULL, col3_5 BIT(27), col3_6 TEXT NULL GENERATED ALWAYS AS (lower(CAST(col3_4 AS TEXT))) STORED, col3_7 TEXT GENERATED ALWAYS AS (CASE WHEN col3_0 IS NULL THEN 'Y'::TEXT ELSE e'\rg[q:BF'::TEXT END) STORED);
        test_1        |             CREATE INDEX ON table3 (col3_2 DESC, col3_0 DESC, col3_4 ASC);
        test_1        |             CREATE TABLE table4 (col4_0 CHAR NOT NULL, PRIMARY KEY (col4_0));
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_0 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_0 DESC);
        test_1        |             CREATE INDEX ON table4 (col4_0 DESC);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_0);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_0);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_0 DESC);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('wazneq', 'zboz', 't', 'ilfyd');
        test_1        |             CREATE TYPE rand_typ_1 AS ENUM ('tnj');
        test_1        |             CREATE TYPE rand_typ_2 AS ENUM ('c', 'ht', 'ywl');
        test_1        |             CREATE TYPE rand_typ_3 AS ENUM ('vran', 'omcwe', 'orjq');
        test_1        |             CREATE TYPE rand_typ_4 AS ENUM ('we');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: functions in index expression must be marked IMMUTABLE (SQLSTATE 42P17)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4115912&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4115912&tab=artifacts#/) on master @ [d6b99e92bf55b6f4a0d79800d67924e04d0b2a6d](https://github.com/cockroachdb/cockroach/commits/d6b99e92bf55b6f4a0d79800d67924e04d0b2a6d):


```
        postgres_1    | 		col_27868 NULLS FIRST,
        postgres_1    | 		col_27869 NULLS FIRST,
        postgres_1    | 		col_27870 NULLS FIRST,
        postgres_1    | 		col_27871 NULLS FIRST,
        postgres_1    | 		col_27872 NULLS FIRST,
        postgres_1    | 		col_27873 NULLS FIRST
        test_1        | --- FAIL: TestCompare (27.91s)
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (24.60s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:193: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			'A':::STRING::STRING AS col_27875,
        test_1        |             			tab_16504.col3_11 AS col_27876,
        test_1        |             			tab_16504.col3_11 AS col_27877,
        test_1        |             			'D':::STRING::STRING AS col_27878
        test_1        |             		FROM
        test_1        |             			postgres.public.table3 AS tab_16504
        test_1        |             			JOIN postgres.public.table3 AS tab_16505 ON
        test_1        |             					(tab_16504.col3_5) = (tab_16505.col3_5)
        test_1        |             					AND (tab_16504.col3_10) = (tab_16505.col3_10)
        test_1        |             					AND (tab_16504.col3_7) = (tab_16505.col3_7)
        test_1        |             					AND (tab_16504.col3_1) = (tab_16505.col3_1)
        test_1        |             					AND (tab_16504.col3_8) = (tab_16505.col3_8)
        test_1        |             					AND (tab_16504.col3_3) = (tab_16505.col3_3)
        test_1        |             	)
        test_1        |             		AS tab_16506
        test_1        |             ORDER BY
        test_1        |             	col_27875 NULLS FIRST, col_27876 NULLS FIRST, col_27877 NULLS FIRST, col_27878 NULLS FIRST
        test_1        |         compare_test.go:194: cockroach1: ping: conn closed
        test_1        |     --- FAIL: TestCompare/mutators (0.00s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:140: pgx conn: dial tcp 172.18.0.3:26257: connect: connection refused
        test_1        | FAIL
        compare_cockroach1_1 exited with code 2
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4126799&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4126799&tab=artifacts#/) on master @ [78419450178335b31f542bd1b14fefdf4ecee0e8](https://github.com/cockroachdb/cockroach/commits/78419450178335b31f542bd1b14fefdf4ecee0e8):


```
        postgres_1    | 		col_825 NULLS FIRST,
        postgres_1    | 		col_826 NULLS FIRST,
        postgres_1    | 		col_827 NULLS FIRST,
        postgres_1    | 		col_828 NULLS FIRST,
        postgres_1    | 		col_829 NULLS FIRST
        test_1        | --- FAIL: TestCompare (5.88s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (3.75s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:193: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			1:::OID::OID AS col_1053, 5:::OID::OID AS col_1054, tab_544.col1_8 AS col_1055
        test_1        |             		FROM
        test_1        |             			postgres.public.table1@[0] AS tab_544
        test_1        |             			JOIN postgres.public.table1 AS tab_545
        test_1        |             				JOIN postgres.public.table1 AS tab_546 ON
        test_1        |             						(tab_545.col1_1) = (tab_546.col1_1)
        test_1        |             						AND (tab_545.col1_7) = (tab_546.col1_7)
        test_1        |             						AND (tab_545.col1_9) = (tab_546.col1_9)
        test_1        |             						AND (tab_545.col1_5) = (tab_546.col1_5)
        test_1        |             						AND (tab_545.col1_8) = (tab_546.col1_8) ON (tab_544.col1_0) = (tab_546.col1_0)
        test_1        |             		WHERE
        test_1        |             			tab_544.col1_4
        test_1        |             		ORDER BY
        test_1        |             			tab_545.col1_0
        test_1        |             	)
        test_1        |             		AS tab_547
        test_1        |             ORDER BY
        test_1        |             	col_1053 NULLS FIRST, col_1054 NULLS FIRST, col_1055 NULLS FIRST
        test_1        |         compare_test.go:194: cockroach1: ping: conn closed
        test_1        |     --- FAIL: TestCompare/mutators (0.00s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:140: pgx conn: dial tcp 172.18.0.2:26257: connect: connection refused
        test_1        | FAIL
        compare_cockroach1_1 exited with code 2
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4131597&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4131597&tab=artifacts#/) on master @ [5ad21e3896ee809e9c3ebc28bb22166f1275acca](https://github.com/cockroachdb/cockroach/commits/5ad21e3896ee809e9c3ebc28bb22166f1275acca):


```
        postgres_1    | 	CREATE UNIQUE INDEX ON table3 (col3_0, lower(CAST(col3_0 AS TEXT)) ASC, col3_1 ASC);
        postgres_1    | 	CREATE INDEX ON table3 (col3_0, col3_1);
        postgres_1    | 	CREATE UNIQUE INDEX ON table3 (col3_0, col3_1);
        postgres_1    | 	CREATE INDEX ON table3 (col3_2 DESC, col3_1 DESC, col3_0 ASC);
        postgres_1    | 	CREATE UNIQUE INDEX ON table3 (col3_0 DESC, col3_2 DESC);
        postgres_1    | 	CREATE TABLE table4 (col4_0 REGPROC NOT NULL, col4_1 REGPROCEDURE, col4_2 DECIMAL NOT NULL, col4_3 UUID NOT NULL, col4_4 UUID NOT NULL, col4_5 BOX2D NOT NULL, col4_6 REGROLE NULL, col4_7 OID NOT NULL, col4_8 REGCLASS NOT NULL, col4_9 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col4_7 IS NULL THEN e'\x1ai'::TEXT ELSE e'\x03\x18'::TEXT END) STORED, PRIMARY KEY (col4_7, col4_0));
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (col4_1 ASC, col4_8 DESC, col4_9 ASC, lower(CAST(col4_4 AS TEXT)), col4_0 ASC, col4_2, col4_4);
        postgres_1    | 	CREATE UNIQUE INDEX ON table4 (col4_2 ASC, col4_9, col4_3, lower(CAST(col4_4 AS TEXT)) ASC, col4_7, col4_8 ASC, col4_6 ASC, col4_4, col4_0 DESC);
        postgres_1    | 	CREATE TABLE table5 (col5_0 TIMESTAMPTZ, col5_1 FLOAT4 NULL, col5_2 INTERVAL NOT NULL, col5_3 CHAR NOT NULL, col5_4 REGCLASS, col5_5 REGTYPE NOT NULL, col5_6 FLOAT8 NULL, col5_7 INTERVAL NOT NULL, col5_8 TIME NULL, col5_9 BOX2D NOT NULL, col5_10 REGNAMESPACE NOT NULL, col5_11 INT4 NOT NULL, col5_12 INTERVAL, col5_13 TIMETZ NOT NULL, col5_14 INT8 NOT NULL, col5_15 TIMESTAMP NOT NULL, col5_16 TEXT GENERATED ALWAYS AS (CASE WHEN col5_0 IS NULL THEN NULL ELSE '<-OyL'::TEXT END) STORED, col5_17 FLOAT4 NULL GENERATED ALWAYS AS (col5_1 + col5_6) STORED, col5_18 INT8 NOT NULL GENERATED ALWAYS AS (col5_14 + col5_11) STORED, PRIMARY KEY (col5_7, col5_5, col5_13, col5_3, col5_10, col5_2, col5_11));
        postgres_1    | 	CREATE UNIQUE INDEX ON table5 (col5_17 DESC, col5_11 ASC, col5_2 ASC, col5_8, col5_4, col5_7, col5_14 DESC, col5_18 ASC, col5_15 DESC, col5_13, (col5_6 + col5_1) DESC, col5_5, col5_0 ASC, col5_10 DESC, col5_6 DESC, col5_3 DESC);
        postgres_1    | 	CREATE TYPE rand_typ_0 AS ENUM ('phnxl', 'pj', 'zcoyrt', 'i');
        postgres_1    | 	
        test_1        | --- FAIL: TestCompare (608.08s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (3.34s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:154: CREATE TABLE table1 (col1_0 OID, col1_1 REGTYPE, col1_2 BIT NOT NULL, col1_3 REGPROCEDURE, col1_4 DATE, col1_5 REGPROCEDURE NOT NULL, col1_6 REGROLE, col1_7 GEOMETRY, col1_8 FLOAT8 NULL, col1_9 REGTYPE NOT NULL, col1_10 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_6 IS NULL THEN 'N%f'::TEXT ELSE e'Q\b.}/4'::TEXT END) STORED, col1_11 TEXT GENERATED ALWAYS AS (CASE WHEN col1_0 IS NULL THEN e'\n}[^U}B'::TEXT ELSE e'(\\7B2Z'::TEXT END) STORED, col1_12 TEXT GENERATED ALWAYS AS (CASE WHEN col1_6 IS NULL THEN '"'::TEXT ELSE 'ZH<cw'::TEXT END) STORED, col1_13 TEXT NOT NULL GENERATED ALWAYS AS (lower(CAST(col1_2 AS TEXT))) STORED, col1_14 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_0 IS NULL THEN e'\''::TEXT ELSE 'L'::TEXT END) STORED, col1_15 TEXT GENERATED ALWAYS AS (CASE WHEN col1_1 IS NULL THEN ''::TEXT ELSE NULL END) STORED, col1_16 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_5 IS NULL THEN e'\x19\x11'::TEXT ELSE e'\x04a?\x03\x01'::TEXT END) STORED, col1_17 TEXT GENERATED ALWAYS AS (CASE WHEN col1_1 IS NULL THEN e'\x07z(}\x15'::TEXT ELSE e's.+\x7f\x1c'::TEXT END) STORED, col1_18 TEXT NOT NULL GENERATED ALWAYS AS (CASE WHEN col1_9 IS NULL THEN e'\x00'::TEXT ELSE '"vtF9'::TEXT END) STORED, PRIMARY KEY (col1_9, col1_5, col1_10, col1_14));
        test_1        |             CREATE INDEX ON table1 ((CASE WHEN col1_5 IS NULL THEN e'|f^\x0en'::TEXT ELSE ',Q)K'::TEXT END) DESC, col1_4 ASC, (CASE WHEN col1_1 IS NULL THEN 'U['::TEXT ELSE e'\t^I'::TEXT END) DESC, col1_14, col1_9, col1_1 ASC, col1_0 ASC, col1_5, col1_11 DESC, col1_8 ASC, col1_16 ASC, col1_18 DESC) INCLUDE (col1_6, col1_7, col1_12);
        test_1        |             CREATE INDEX ON table1 (col1_14 DESC, col1_6 ASC, col1_17 ASC, col1_3 ASC, col1_10 ASC, (CASE WHEN col1_6 IS NULL THEN e'\x06|@#;.Z('::TEXT ELSE 'u;}'::TEXT END));
        test_1        |             CREATE UNIQUE INDEX ON table1 (col1_8 DESC);
        test_1        |             CREATE TABLE table2 (col2_0 TIME NULL, col2_1 VARCHAR, col2_2 GEOGRAPHY NULL, col2_3 DATE, col2_4 BYTEA NOT NULL, col2_5 NAME NULL, col2_6 BOOL NOT NULL, col2_7 INTERVAL NOT NULL, col2_8 TIME, col2_9 JSONB, col2_10 INT2, col2_11 UUID NOT NULL, col2_12 CHAR NOT NULL, col2_13 UUID NOT NULL, col2_14 REGROLE NULL, col2_15 VARCHAR NULL, PRIMARY KEY (col2_12, col2_13));
        test_1        |             CREATE TABLE table3 (col3_0 TIMETZ NOT NULL, col3_1 BIT(8) NOT NULL, col3_2 INT2 NOT NULL, PRIMARY KEY (col3_0, col3_1, col3_2));
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_0, lower(CAST(col3_0 AS TEXT)) ASC, col3_1 ASC);
        test_1        |             CREATE INDEX ON table3 (col3_0, col3_1);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_0, col3_1);
        test_1        |             CREATE INDEX ON table3 (col3_2 DESC, col3_1 DESC, col3_0 ASC);
        test_1        |             CREATE UNIQUE INDEX ON table3 (col3_0 DESC, col3_2 DESC);
        test_1        |             CREATE TABLE table4 (col4_0 REGPROC NOT NULL, col4_1 REGPROCEDURE, col4_2 DECIMAL NOT NULL, col4_3 UUID NOT NULL, col4_4 UUID NOT NULL, col4_5 BOX2D NOT NULL, col4_6 REGROLE NULL, col4_7 OID NOT NULL, col4_8 REGCLASS NOT NULL, col4_9 TEXT NULL GENERATED ALWAYS AS (CASE WHEN col4_7 IS NULL THEN e'\x1ai'::TEXT ELSE e'\x03\x18'::TEXT END) STORED, PRIMARY KEY (col4_7, col4_0));
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_1 ASC, col4_8 DESC, col4_9 ASC, lower(CAST(col4_4 AS TEXT)), col4_0 ASC, col4_2, col4_4);
        test_1        |             CREATE UNIQUE INDEX ON table4 (col4_2 ASC, col4_9, col4_3, lower(CAST(col4_4 AS TEXT)) ASC, col4_7, col4_8 ASC, col4_6 ASC, col4_4, col4_0 DESC);
        test_1        |             CREATE TABLE table5 (col5_0 TIMESTAMPTZ, col5_1 FLOAT4 NULL, col5_2 INTERVAL NOT NULL, col5_3 CHAR NOT NULL, col5_4 REGCLASS, col5_5 REGTYPE NOT NULL, col5_6 FLOAT8 NULL, col5_7 INTERVAL NOT NULL, col5_8 TIME NULL, col5_9 BOX2D NOT NULL, col5_10 REGNAMESPACE NOT NULL, col5_11 INT4 NOT NULL, col5_12 INTERVAL, col5_13 TIMETZ NOT NULL, col5_14 INT8 NOT NULL, col5_15 TIMESTAMP NOT NULL, col5_16 TEXT GENERATED ALWAYS AS (CASE WHEN col5_0 IS NULL THEN NULL ELSE '<-OyL'::TEXT END) STORED, col5_17 FLOAT4 NULL GENERATED ALWAYS AS (col5_1 + col5_6) STORED, col5_18 INT8 NOT NULL GENERATED ALWAYS AS (col5_14 + col5_11) STORED, PRIMARY KEY (col5_7, col5_5, col5_13, col5_3, col5_10, col5_2, col5_11));
        test_1        |             CREATE UNIQUE INDEX ON table5 (col5_17 DESC, col5_11 ASC, col5_2 ASC, col5_8, col5_4, col5_7, col5_14 DESC, col5_18 ASC, col5_15 DESC, col5_13, (col5_6 + col5_1) DESC, col5_5, col5_0 ASC, col5_10 DESC, col5_6 DESC, col5_3 DESC);
        test_1        |             CREATE TYPE rand_typ_0 AS ENUM ('phnxl', 'pj', 'zcoyrt', 'i');
        test_1        |             
        test_1        |         compare_test.go:155: postgres: exec: ERROR: invalid byte sequence for encoding "UTF8": 0x00 (SQLSTATE 22021)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4134687&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4134687&tab=artifacts#/) on master @ [4b41789120e019ab015e6dbb924df763897ebadb](https://github.com/cockroachdb/cockroach/commits/4b41789120e019ab015e6dbb924df763897ebadb):


```
        test_1        |     --- FAIL: TestCompare/postgres (134.09s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	nil,
        test_1        |             + 	string("BOX(-45 0,0 45)"),
        test_1        |               	bool(false),
        test_1        |               	uint32(4),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_box2dfromgeohash('E':::STRING::STRING::STRING, NULL::INT4::INT4)::BOX2D AS col_164629,
        test_1        |             			false::BOOL AS col_164630,
        test_1        |             			4:::OID::OID AS col_164631
        test_1        |             	)
        test_1        |             		AS tab_90045
        test_1        |             ORDER BY
        test_1        |             	CAST(col_164629 AS STRING) NULLS FIRST, col_164630 NULLS FIRST, col_164631 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			st_box2dfromgeohash('E'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::BOX2D AS col_164629,
        test_1        |             			false::BOOL AS col_164630,
        test_1        |             			4::OID::OID AS col_164631
        test_1        |             	)
        test_1        |             		AS tab_90045
        test_1        |             ORDER BY
        test_1        |             	CAST(col_164629 AS TEXT) NULLS FIRST, col_164630 NULLS FIRST, col_164631 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4140356&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4140356&tab=artifacts#/) on master @ [365b4da8bd02c06ee59d2130a56dec74ffc9ce21](https://github.com/cockroachdb/cockroach/commits/365b4da8bd02c06ee59d2130a56dec74ffc9ce21):


```
        postgres_1    | 		)
        postgres_1    | 			AS tab_29034
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_48095 NULLS FIRST,
        postgres_1    | 		col_48096 NULLS FIRST,
        postgres_1    | 		col_48097 NULLS FIRST,
        postgres_1    | 		col_48098 NULLS FIRST,
        postgres_1    | 		col_48099 NULLS FIRST,
        postgres_1    | 		col_48100 NULLS FIRST,
        postgres_1    | 		col_48101 NULLS FIRST
        test_1        | --- FAIL: TestCompare (648.70s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (44.87s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	nil,
        test_1        |             + 	string("BOX(-90 45,-45 90)"),
        test_1        |               }))
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_box2dfromgeohash('F'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::BOX2D AS col_48562) AS tab_29276
        test_1        |             ORDER BY
        test_1        |             	CAST(col_48562 AS TEXT) NULLS FIRST;
        test_1        |             
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_box2dfromgeohash('F':::STRING::STRING::STRING, NULL::INT4::INT4)::BOX2D AS col_48562) AS tab_29276
        test_1        |             ORDER BY
        test_1        |             	CAST(col_48562 AS STRING) NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4149624&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4149624&tab=artifacts#/) on master @ [912964e02ddd951c77d4f71981ae18b3894e9084](https://github.com/cockroachdb/cockroach/commits/912964e02ddd951c77d4f71981ae18b3894e9084):


```
        test_1        | --- FAIL: TestCompare (627.67s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (25.31s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	string("2000-01-01 06:00:04+00:00F"),
        test_1        |             + 	string("2000-01-01 06:00:04+00F"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(('00:00:04':::INTERVAL::INTERVAL::INTERVAL + '2000-01-01 06:00:00+00:00':::TIMESTAMPTZ::TIMESTAMPTZ::TIMESTAMPTZ)::TIMESTAMPTZ::TIMESTAMPTZ || 'F':::STRING::STRING::STRING)::STRING
        test_1        |             				AS col_30126
        test_1        |             	)
        test_1        |             		AS tab_15389
        test_1        |             ORDER BY
        test_1        |             	col_30126 NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			(('00:00:04'::INTERVAL::INTERVAL::INTERVAL + '2000-01-01 06:00:00+00:00'::TIMESTAMPTZ::TIMESTAMPTZ::TIMESTAMPTZ)::TIMESTAMPTZ::TIMESTAMPTZ || 'F'::TEXT::TEXT::TEXT)::TEXT
        test_1        |             				AS col_30126
        test_1        |             	)
        test_1        |             		AS tab_15389
        test_1        |             ORDER BY
        test_1        |             	col_30126 NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4159947&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4159947&tab=artifacts#/) on master @ [da01e4c0545f191a0573e1d097ff0366769e0d6b](https://github.com/cockroachdb/cockroach/commits/da01e4c0545f191a0573e1d097ff0366769e0d6b):


```
        postgres_1    | 				tab_132093.col1_2 ASC,
        postgres_1    | 				tab_132093.col1_2 ASC,
        postgres_1    | 				tab_132093.col1_2,
        postgres_1    | 				tab_132093.col1_2,
        postgres_1    | 				tab_132093.col1_2,
        postgres_1    | 				tab_132093.col1_2
        postgres_1    | 		)
        postgres_1    | 			AS tab_132094
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_233810 NULLS FIRST
        test_1        | --- FAIL: TestCompare (797.27s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (194.61s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	nil,
        test_1        |             + 	string("BOX(-90 0,-45 45)"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_box2dfromgeohash('D':::STRING::STRING::STRING, NULL::INT4::INT4)::BOX2D AS col_233913) AS tab_132183
        test_1        |             ORDER BY
        test_1        |             	CAST(col_233913 AS STRING) NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_box2dfromgeohash('D'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::BOX2D AS col_233913) AS tab_132183
        test_1        |             ORDER BY
        test_1        |             	CAST(col_233913 AS TEXT) NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4189688&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4189688&tab=artifacts#/) on master @ [8cd28089c6c7333615ba3201e841839001d2f0e1](https://github.com/cockroachdb/cockroach/commits/8cd28089c6c7333615ba3201e841839001d2f0e1):


```
        test_1        | --- FAIL: TestCompare (2.13s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/mutators (0.00s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 20 [running]:
        test_1        | testing.tRunner.func1.2({0x1b20320, 0xc001d12580})
        test_1        | 	/usr/local/go/src/testing/testing.go:1209 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	/usr/local/go/src/testing/testing.go:1212 +0x218
        test_1        | panic({0x1b20320, 0xc001d12580})
        test_1        | 	/usr/local/go/src/runtime/panic.go:1038 +0x215
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x39346e0, {0x291de80, 0xc001c7b5c0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:369 +0x487
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x28a9de0, 0x39346e0)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:302 +0x14c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc000627800)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:221 +0x92
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc001c7b230, {0xc001c7e5a0, 0x1, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:252 +0xa4d
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0x90, 0x1c8dff7, {0xc001c7e5a0, 0x8e, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:98 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.RandCreateTables(0x203000, {0x1c8dff7, 0x5}, 0x1, {0xc001bc5a30, 0x3, 0x0})
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/randgen/schema.go:85 +0x157
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTablesN(0x28beba0, 0xc0001c0000)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:87 +0x187
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0x64265e37c3364c9c)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:74 +0x33
        test_1        | github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.wrapCommonSetup.func1(0x1c93262)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:44 +0x22
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc001bff6c0)
        test_1        | 	/go/src/github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:128 +0xe5
        test_1        | testing.tRunner(0xc001bff6c0, 0xc001c7b1a0)
        test_1        | 	/usr/local/go/src/testing/testing.go:1259 +0x102
        test_1        | created by testing.(*T).Run
        test_1        | 	/usr/local/go/src/testing/testing.go:1306 +0x35a
        compare_test_1 exited with code 2
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:50: exit status 2
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4211166&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4211166&tab=artifacts#/) on master @ [e5d1c374c31dc0e80a596c570da8dc45d73f80b8](https://github.com/cockroachdb/cockroach/commits/e5d1c374c31dc0e80a596c570da8dc45d73f80b8):


```
        postgres_1    | 		(
        postgres_1    | 			SELECT
        postgres_1    | 				'D'::TEXT::CHAR AS col_43241,
        postgres_1    | 				tab_29022.col2_4 AS col_43242,
        postgres_1    | 				to_char_with_style('2000-01-05'::DATE::DATE::DATE, tab_29022.col2_3::TEXT)::TEXT AS col_43243
        postgres_1    | 			FROM
        postgres_1    | 				postgres.public.table2 AS tab_29022
        postgres_1    | 		)
        postgres_1    | 			AS tab_29023
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_43241 NULLS FIRST, col_43242 NULLS FIRST, col_43243 NULLS FIRST
        test_1        | --- FAIL: TestCompare (650.07s)
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (46.37s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	string("{NULL,NULL}"),
        test_1        |             + 	string("{NULL:NULL}"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			((NULL::GEOGRAPHY[]::GEOGRAPHY[] || NULL::GEOGRAPHY::GEOGRAPHY)::GEOGRAPHY[]::GEOGRAPHY[] || NULL::GEOGRAPHY::GEOGRAPHY)::GEOGRAPHY[]
        test_1        |             				AS col_43624
        test_1        |             	)
        test_1        |             		AS tab_29287
        test_1        |             ORDER BY
        test_1        |             	col_43624 NULLS FIRST;
        test_1        |             
        test_1        |             postgres: [same as previous]
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:62: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4224431&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4224431&tab=artifacts#/) on master @ [83e2df701688745fe7e7d8a0d0e5e7a4ba8633c8](https://github.com/cockroachdb/cockroach/commits/83e2df701688745fe7e7d8a0d0e5e7a4ba8633c8):


```
        postgres_1    | 			WHERE
        postgres_1    | 				true::BOOL
        postgres_1    | 			ORDER BY
        postgres_1    | 				tab_69586.col2_0 DESC
        postgres_1    | 		)
        postgres_1    | 			AS tab_69587
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_102159 NULLS FIRST, col_102160 NULLS FIRST, col_102161 NULLS FIRST, col_102162 NULLS FIRST
        postgres_1    | 2022-01-27 06:17:20.424 UTC [79] ERROR:  could not identify an ordering operator for type void at character 71
        postgres_1    | 2022-01-27 06:17:20.424 UTC [79] HINT:  Use an explicit ordering operator or modify the query.
        postgres_1    | 2022-01-27 06:17:20.424 UTC [79] STATEMENT:  SELECT * FROM (SELECT NULL::VOID AS col_102291) AS tab_69667 ORDER BY col_102291 NULLS FIRST
        test_1        | --- FAIL: TestCompare (699.90s)
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (96.33s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	string("{NULL,NULL}"),
        test_1        |             + 	string("{NULL:NULL}"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			((CASE WHEN false::BOOL THEN NULL::GEOMETRY ELSE NULL::GEOMETRY END::GEOMETRY || NULL::GEOMETRY[]::GEOMETRY[])::GEOMETRY[]::GEOMETRY[] || NULL::GEOMETRY::GEOMETRY)::GEOMETRY[]
        test_1        |             				AS col_102314
        test_1        |             	)
        test_1        |             		AS tab_69681
        test_1        |             ORDER BY
        test_1        |             	col_102314 NULLS FIRST;
        test_1        |             
        test_1        |             postgres: [same as previous]
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach1_1 ... done
        Stopping compare_cockroach2_1 ... done
        Aborting on container exit...
        
    compose_test.go:122: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4250886&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4250886&tab=artifacts#/) on master @ [a5158c4c93e9dca858b38ef7e94321f0a0a2b5c5](https://github.com/cockroachdb/cockroach/commits/a5158c4c93e9dca858b38ef7e94321f0a0a2b5c5):


```
        postgres_1    | 		col_104216 NULLS FIRST,
        postgres_1    | 		col_104217 NULLS FIRST,
        postgres_1    | 		col_104218 NULLS FIRST,
        postgres_1    | 		col_104219 NULLS FIRST,
        postgres_1    | 		col_104220 NULLS FIRST,
        postgres_1    | 		col_104221 NULLS FIRST,
        postgres_1    | 		col_104222 NULLS FIRST,
        postgres_1    | 		col_104223 NULLS FIRST,
        postgres_1    | 		col_104224 NULLS FIRST,
        postgres_1    | 		col_104225 NULLS FIRST
        test_1        | --- FAIL: TestCompare (699.17s)
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     --- FAIL: TestCompare/postgres (97.79s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	nil,
        test_1        |             + 	string("BOX(-45 0,0 45)"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_box2dfromgeohash('E':::STRING::STRING::STRING, NULL::INT4::INT4)::BOX2D AS col_105030) AS tab_64755
        test_1        |             ORDER BY
        test_1        |             	CAST(col_105030 AS STRING) NULLS FIRST;
        test_1        |             
        test_1        |             postgres:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(SELECT st_box2dfromgeohash('E'::TEXT::TEXT::TEXT, NULL::INT4::INT4)::BOX2D AS col_105030) AS tab_64755
        test_1        |             ORDER BY
        test_1        |             	CAST(col_105030 AS TEXT) NULLS FIRST;
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:122: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4275931&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4275931&tab=artifacts#/) on master @ [16859b99637d1adedb915a67998749dbfa4c0663](https://github.com/cockroachdb/cockroach/commits/16859b99637d1adedb915a67998749dbfa4c0663):


```
        test_1        |             	)
        test_1        |             		AS tab_42972
        test_1        |             ORDER BY
        test_1        |             	col_79536 NULLS FIRST, CAST(col_79537 AS TEXT) NULLS FIRST;
        test_1        |             
        test_1        |     --- FAIL: TestCompare/mutators (529.51s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: cockroach2
        test_1        |         compare_test.go:193: SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		WITH
        test_1        |             			with_88777 (col_511020) AS (SELECT * FROM (VALUES ((-31441):::INT8)) AS tab_328422 (col_511020))
        test_1        |             		SELECT
        test_1        |             			3927221520:::OID AS col_511022,
        test_1        |             			tab_328423.col5_6 AS col_511023,
        test_1        |             			126170335:::OID AS col_511024,
        test_1        |             			tab_328423.crdb_internal_mvcc_timestamp AS col_511025
        test_1        |             		FROM
        test_1        |             			postgres.public.table5@table5_pkey AS tab_328423,
        test_1        |             			postgres.public.table1@[0] AS tab_328424,
        test_1        |             			with_88777 AS cte_ref_25567
        test_1        |             		WHERE
        test_1        |             			cte_ref_25567.col_511020::INT8
        test_1        |             			NOT IN (
        test_1        |             					SELECT
        test_1        |             						tab_328425.col4_16::INT8 AS col_511021
        test_1        |             					FROM
        test_1        |             						postgres.public.table4@[0] AS tab_328425
        test_1        |             					WHERE
        test_1        |             						true
        test_1        |             				)
        test_1        |             	)
        test_1        |             		AS tab_328426
        test_1        |             ORDER BY
        test_1        |             	col_511022 NULLS FIRST, col_511023 NULLS FIRST, col_511024 NULLS FIRST, col_511025 NULLS FIRST
        test_1        |         compare_test.go:194: cockroach2: ping: conn closed
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:122: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4288307&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4288307&tab=artifacts#/) on master @ [f20c1a8d824f0d00f18c0d58ed8b4bc4955e0969](https://github.com/cockroachdb/cockroach/commits/f20c1a8d824f0d00f18c0d58ed8b4bc4955e0969):


```
        postgres_1    | 				false::BOOL
        postgres_1    | 		)
        postgres_1    | 			AS tab_252639
        postgres_1    | 	ORDER BY
        postgres_1    | 		col_481295 NULLS FIRST,
        postgres_1    | 		col_481296 NULLS FIRST,
        postgres_1    | 		col_481297 NULLS FIRST,
        postgres_1    | 		col_481298 NULLS FIRST,
        postgres_1    | 		col_481299 NULLS FIRST,
        postgres_1    | 		col_481300 NULLS FIRST,
        postgres_1    | 		col_481301 NULLS FIRST
        test_1        | --- FAIL: TestCompare (1041.66s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (435.66s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:133: initializing connection: cockroach1
        test_1        |         compare_test.go:133: initializing connection: postgres
        test_1        |         compare_test.go:184: compare cockroach1 to postgres: unexpected diff:
        test_1        |               []interface{}(Inverse(func1, []interface{}{
        test_1        |             - 	string("{NULL,NULL}"),
        test_1        |             + 	string("{NULL:NULL}"),
        test_1        |               }))
        test_1        |             cockroach1:
        test_1        |             SELECT
        test_1        |             	*
        test_1        |             FROM
        test_1        |             	(
        test_1        |             		SELECT
        test_1        |             			((NULL::GEOGRAPHY::GEOGRAPHY || NULL::GEOGRAPHY[]::GEOGRAPHY[])::GEOGRAPHY[]::GEOGRAPHY[] || NULL::GEOGRAPHY::GEOGRAPHY)::GEOGRAPHY[]
        test_1        |             				AS col_481334
        test_1        |             	)
        test_1        |             		AS tab_252660
        test_1        |             ORDER BY
        test_1        |             	col_481334 NULLS FIRST;
        test_1        |             
        test_1        |             postgres: [same as previous]
        test_1        |             
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:122: exit status 1
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5371428?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5371428?buildTab=artifacts#/) on master @ [90b5db50e1e1cdb0315d8b094081d261e6dcb336](https://github.com/cockroachdb/cockroach/commits/90b5db50e1e1cdb0315d8b094081d261e6dcb336):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2022-06-04 05:22:05.515 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2022-06-04 05:22:05.517 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2022-06-04 05:22:05.519 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2022-06-04 05:22:05.522 UTC [51] LOG:  shutting down
        postgres_1    | 2022-06-04 05:22:05.636 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2022-06-04 05:22:05.739 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2022-06-04 05:22:05.739 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2022-06-04 05:22:05.739 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2022-06-04 05:22:05.743 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-06-04 05:22:05.750 UTC [68] LOG:  database system was shut down at 2022-06-04 05:22:05 UTC
        postgres_1    | 2022-06-04 05:22:05.756 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-06-04 05:22:06.742 UTC [78] ERROR:  unrecognized configuration parameter "sql_safe_updates"
        postgres_1    | 2022-06-04 05:22:06.742 UTC [78] STATEMENT:  SET sql_safe_updates = false;
        test_1        | --- FAIL: TestCompare (4.16s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.87s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: postgres
        test_1        |         compare_test.go:157: SET sql_safe_updates = false;
        test_1        |         compare_test.go:158: postgres: exec: ERROR: unrecognized configuration parameter "sql_safe_updates" (SQLSTATE 42704)
        test_1        |     --- FAIL: TestCompare/mutators (0.15s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:157: CREATE TABLE table1 (col1_0 TIMESTAMPTZ NULL, col1_1 VARCHAR, col1_2 FLOAT8 NULL, col1_3 INT2 NOT NULL, col1_4 TIMESTAMP NOT NULL, col1_5 FLOAT8 NOT NULL, col1_6 INT8 NULL, col1_7 DATE NULL, col1_8 REGTYPE NULL, col1_9 DATE NOT NULL, col1_10 BOX2D NOT NULL, col1_11 INTERVAL, INDEX (col1_1, col1_7 DESC), INDEX (col1_10 ASC, (CASE WHEN col1_11 IS NULL THEN e'{\\Q\x7f+``\x13j':::STRING ELSE 'p':::STRING END) DESC, col1_1 DESC, col1_8 ASC, col1_3, col1_9 ASC, col1_2 ASC, col1_7, col1_0, col1_11 ASC, col1_6 ASC) STORING (col1_5) WHERE (((((table1.col1_7 > '4714-11-24 BC':::DATE) OR (table1.col1_6 <= 127:::INT8)) AND (table1.col1_9 != '-infinity':::DATE)) OR (table1.col1_3 > (-1):::INT8)) OR (table1.col1_5 > 3.4028234663852886e+38:::FLOAT8)) AND (table1.col1_1 = 'X':::STRING), INDEX (col1_4 ASC, col1_5, col1_1, col1_8 ASC) STORING (col1_0, col1_3, col1_6, col1_7, col1_9, col1_10, col1_11), UNIQUE (col1_4, col1_0, col1_10 DESC), INDEX (col1_9 DESC, col1_8 DESC, col1_3, col1_6 ASC, col1_2, (CASE WHEN col1_8 IS NULL THEN e'\x1e+9\x1c(\x7f\x03W':::STRING ELSE e'\x13f%Lc(':::STRING END) ASC, (col1_2 + col1_5) ASC, col1_1 DESC, col1_11 ASC, col1_0 ASC, col1_7) STORING (col1_5), UNIQUE (col1_3 DESC, (col1_6 + (-4271495114520665241):::INT8) DESC) STORING (col1_1, col1_2, col1_4, col1_5, col1_6, col1_8) PARTITION BY LIST (col1_3) (PARTITION table1_part_0 VALUES IN (((-13626):::INT8,), ((-16631):::INT8,), ((-22765):::INT8,), ((-8017):::INT8,), (19510:::INT8,), (23032:::INT8,), (31535:::INT8,), ((-1):::INT8,), (22747:::INT8,), ((-128):::INT8,)), PARTITION table1_part_1 VALUES IN ((5419:::INT8,), (31370:::INT8,), (19229:::INT8,), (1:::INT8,), (7257:::INT8,), (1676:::INT8,), ((-7625):::INT8,), ((-23671):::INT8,), ((-19951):::INT8,), (17896:::INT8,)), PARTITION table1_part_2 VALUES IN (((-10375):::INT8,), ((-28944):::INT8,), (8894:::INT8,), ((-18499):::INT8,), ((-24645):::INT8,), (18906:::INT8,), ((-8677):::INT8,), (24618:::INT8,), (21668:::INT8,), ((-30354):::INT8,)), PARTITION table1_part_3 VALUES IN ((22428:::INT8,), (27014:::INT8,), ((-10362):::INT8,), ((-30567):::INT8,), ((-12340):::INT8,), (5723:::INT8,), (14240:::INT8,), (32760:::INT8,), ((-29733):::INT8,)), PARTITION table1_part_4 VALUES IN ((18732:::INT8,), (15242:::INT8,), (27850:::INT8,), ((-1807):::INT8,), (18823:::INT8,), (20763:::INT8,), (32767:::INT8,), (13147:::INT8,), (3260:::INT8,)), PARTITION table1_part_5 VALUES IN (((-15031):::INT8,), (25901:::INT8,), ((-29326):::INT8,), ((-3272):::INT8,), ((-25993):::INT8,), ((-2425):::INT8,), (17063:::INT8,), ((-13396):::INT8,), (29676:::INT8,), ((-29302):::INT8,)), PARTITION table1_part_6 VALUES IN (((-4906):::INT8,), (31400:::INT8,), (16576:::INT8,), ((-24466):::INT8,), (11437:::INT8,), (4894:::INT8,), ((-4530):::INT8,), (19574:::INT8,)), PARTITION table1_part_7 VALUES IN ((5810:::INT8,), (21965:::INT8,), ((-27256):::INT8,), ((-25032):::INT8,), ((-806):::INT8,), (30233:::INT8,), (24095:::INT8,), (30107:::INT8,), (29909:::INT8,), (28862:::INT8,)), PARTITION table1_part_8 VALUES IN (((-6636):::INT8,), (26400:::INT8,), ((-32071):::INT8,), (24225:::INT8,), (5377:::INT8,), ((-3248):::INT8,), (23315:::INT8,), ((-19067):::INT8,), ((-21278):::INT8,), (16969:::INT8,)), PARTITION "DEFAULT" VALUES IN ((DEFAULT,))) WHERE table1.col1_0 >= '3000-01-01 00:00:00+00:00':::TIMESTAMPTZ, FAMILY (col1_4, col1_6), FAMILY (col1_11), FAMILY (col1_3, col1_2), FAMILY (col1_1, col1_5, col1_9, col1_10), FAMILY (col1_7, col1_8), FAMILY (col1_0))
        test_1        |         compare_test.go:158: cockroach1: exec: ERROR: use of partitions requires an enterprise license. see https://cockroachlabs.com/pricing?cluster=6d8d32d2-e541-4264-a011-cc92ed258c40 for details on how to enable enterprise features (SQLSTATE XXC02)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:120: exit status 1
--- FAIL: TestComposeCompare (21.77s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78068 compose: TestComposeCompare failed [C-test-failure O-robot T-sql-experience branch-release-22.1]
- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5377287?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5377287?buildTab=artifacts#/) on master @ [0fdac09ca1119b494661e1e1f64ea291d8649782](https://github.com/cockroachdb/cockroach/commits/0fdac09ca1119b494661e1e1f64ea291d8649782):


```
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: postgres
        test_1        |         compare_test.go:157: SET sql_safe_updates = false;
        test_1        |         compare_test.go:158: postgres: exec: ERROR: unrecognized configuration parameter "sql_safe_updates" (SQLSTATE 42704)
        test_1        |     --- FAIL: TestCompare/mutators (0.43s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: cockroach2
        test_1        | panic: geos: no locations to init GEOS [recovered]
        test_1        | 	panic: geos: no locations to init GEOS
        test_1        | 
        test_1        | goroutine 25 [running]:
        test_1        | testing.tRunner.func1.2({0x1ceb6e0, 0xc000169a40})
        test_1        | 	GOROOT/src/testing/testing.go:1209 +0x24e
        test_1        | testing.tRunner.func1()
        test_1        | 	GOROOT/src/testing/testing.go:1212 +0x218
        test_1        | panic({0x1ceb6e0, 0xc000169a40})
        test_1        | 	GOROOT/src/runtime/panic.go:1038 +0x215
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.encodeInvertedIndexHistogramUpperBounds(0x3ef7fe0, {0x2dfcb10, 0xc0020046c0})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:369 +0x445
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.randHistogram(0x2d8c2e0, 0x3ef7fe0)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:302 +0x14c
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator.func1(0xc001f871e0)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:221 +0x92
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.statisticsMutator(0xc00014c120, {0xc00014efa0, 0x1, 0x4760b4})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:252 +0xa4d
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.MultiStatementMutation.Mutate(0xc000680001, 0xc00014efa0, {0xc00014efa0, 0xc001fc3988, 0x4324bd})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:98 +0x2e
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.Apply(0x1c35060, {0xc00014efa0, 0x1fae040, 0xf6}, {0xc001f95300, 0x6, 0x43})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:109 +0xa6
        test_1        | github.com/cockroachdb/cockroach/pkg/sql/randgen.ApplyString(0xc001ffc210, {0xc00016a100, 0xf6}, {0xc0001134a0, 0x6, 0x1efcd6d})
        test_1        | 	github.com/cockroachdb/cockroach/pkg/sql/randgen/mutator.go:160 +0x159
        test_1        | github.com/cockroachdb/cockroach/pkg/compose/compare/compare.TestCompare.func2(0xc001f86000)
        test_1        | 	github.com/cockroachdb/cockroach/pkg/compose/compare/compare/compare_test.go:155 +0xbf1
        test_1        | testing.tRunner(0xc001f86000, 0xc00014c0c0)
        test_1        | 	GOROOT/src/testing/testing.go:1259 +0x102
        test_1        | created by testing.(*T).Run
        test_1        | 	GOROOT/src/testing/testing.go:1306 +0x35a
        compare_test_1 exited with code 2
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:120: exit status 2
--- FAIL: TestComposeCompare (21.97s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78068 compose: TestComposeCompare failed [C-test-failure O-robot T-sql-experience branch-release-22.1]
- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5382883?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5382883?buildTab=artifacts#/) on master @ [2c08debd6d8c89019f26a3e58cf3b5d3d97b6495](https://github.com/cockroachdb/cockroach/commits/2c08debd6d8c89019f26a3e58cf3b5d3d97b6495):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2022-06-06 05:22:41.352 UTC [48] LOG:  received fast shutdown request
        postgres_1    | .2022-06-06 05:22:41.354 UTC [48] LOG:  aborting any active transactions
        postgres_1    | 2022-06-06 05:22:41.356 UTC [48] LOG:  background worker "logical replication launcher" (PID 55) exited with exit code 1
        postgres_1    | 2022-06-06 05:22:41.358 UTC [50] LOG:  shutting down
        postgres_1    | 2022-06-06 05:22:41.475 UTC [48] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2022-06-06 05:22:41.575 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2022-06-06 05:22:41.575 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2022-06-06 05:22:41.575 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2022-06-06 05:22:41.580 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-06-06 05:22:41.586 UTC [67] LOG:  database system was shut down at 2022-06-06 05:22:41 UTC
        postgres_1    | 2022-06-06 05:22:41.592 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-06-06 05:22:43.323 UTC [77] ERROR:  unrecognized configuration parameter "sql_safe_updates"
        postgres_1    | 2022-06-06 05:22:43.323 UTC [77] STATEMENT:  SET sql_safe_updates = false;
        test_1        | --- FAIL: TestCompare (4.83s)
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (0.93s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: postgres
        test_1        |         compare_test.go:157: SET sql_safe_updates = false;
        test_1        |         compare_test.go:158: postgres: exec: ERROR: unrecognized configuration parameter "sql_safe_updates" (SQLSTATE 42704)
        test_1        |     --- FAIL: TestCompare/mutators (0.20s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:157: CREATE TABLE table2 (col2_0 CHAR, col2_1 REGNAMESPACE, col2_2 OID, col2_3 BIT(6) NOT NULL, col2_4 BOX2D NOT NULL, col2_5 REGCLASS NOT NULL, col2_6 STRING NOT NULL, col2_7 OID, col2_8 BOOL NOT NULL, col2_9 STRING NOT NULL AS (lower(CAST(col2_3 AS STRING))) VIRTUAL, col2_10 STRING AS (lower(col2_0)) VIRTUAL, col2_11 STRING NOT NULL AS (CASE WHEN col2_1 IS NULL THEN '"':::STRING ELSE NULL END) STORED, col2_12 STRING NOT NULL AS (lower(col2_6)) VIRTUAL, col2_13 STRING NOT NULL AS (lower(CAST(col2_2 AS STRING))) VIRTUAL, col2_14 STRING NULL AS (lower(CAST(col2_4 AS STRING))) VIRTUAL, col2_15 STRING AS (lower(CAST(col2_2 AS STRING))) STORED, col2_16 STRING NOT NULL AS (lower(CAST(col2_4 AS STRING))) VIRTUAL, PRIMARY KEY (col2_11 ASC, col2_13 DESC, col2_5 ASC, col2_4 DESC, col2_16), UNIQUE (col2_11, col2_3 DESC, col2_0 DESC, col2_13 ASC, col2_5 ASC, col2_15, col2_14, col2_1 ASC, col2_7, lower(CAST(col2_3 AS STRING)) ASC, col2_10 ASC, col2_16, col2_8 ASC) PARTITION BY LIST (col2_11, col2_3, col2_0, col2_13, col2_5, col2_15, col2_14, col2_1, col2_7) (PARTITION table2_part_0 VALUES IN ((e'\x0fHkN,':::STRING, B'000000', 'N':::STRING, e'\x06s \'l':::STRING, 1711891680:::OID, e'r4\x18_]h}W':::STRING, 'x':::STRING, 0:::OID, 3939543233:::OID)), PARTITION table2_part_1 VALUES IN ((e'?H3\x1cj/%':::STRING, B'011111', 'Q':::STRING, '':::STRING, 931871536:::OID, '':::STRING, '':::STRING, 2117187252:::OID, 3218568155:::OID)), PARTITION "DEFAULT" VALUES IN ((DEFAULT, DEFAULT, DEFAULT, DEFAULT, DEFAULT, DEFAULT, DEFAULT, DEFAULT, DEFAULT))), INDEX (col2_11 ASC, col2_8 DESC, col2_3 DESC, col2_10, col2_9, col2_16 DESC, col2_0 ASC, col2_4 ASC), UNIQUE (col2_0 ASC, col2_13 DESC, col2_12, col2_4 DESC, col2_6 DESC, col2_2 DESC, col2_7, col2_14 ASC, col2_16 ASC, col2_10 DESC, col2_1 ASC) WHERE (((((table2.col2_11 >= e'\x00':::STRING) AND (table2.col2_12 >= '':::STRING)) OR (table2.col2_6 <= e'\'':::STRING)) OR (NOT table2.col2_8)) AND (table2.col2_13 > e'\'':::STRING)) AND (table2.col2_14 > e'\x00':::STRING), UNIQUE (lower(CAST(col2_8 AS STRING)) ASC, col2_2, col2_3 DESC, col2_4 ASC, col2_9 ASC, col2_5 DESC, col2_12, col2_10 ASC, col2_7 ASC, col2_8, col2_14 ASC, col2_16 DESC, col2_1 DESC, col2_13 DESC) STORING (col2_0, col2_6, col2_15))
        test_1        |         compare_test.go:158: cockroach1: exec: ERROR: use of partitions requires an enterprise license. see https://cockroachlabs.com/pricing?cluster=f8bf1a23-3447-42e9-8c30-e39482f6bb68 for details on how to enable enterprise features (SQLSTATE XXC02)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:120: exit status 1
--- FAIL: TestComposeCompare (22.27s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78068 compose: TestComposeCompare failed [C-test-failure O-robot T-sql-experience branch-release-22.1]
- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5393789?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5393789?buildTab=artifacts#/) on master @ [5bd3fdd33a351564e7af5e64ce931e82db5e23a1](https://github.com/cockroachdb/cockroach/commits/5bd3fdd33a351564e7af5e64ce931e82db5e23a1):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2022-06-07 05:22:38.821 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2022-06-07 05:22:38.824 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2022-06-07 05:22:38.825 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2022-06-07 05:22:38.828 UTC [51] LOG:  shutting down
        postgres_1    | 2022-06-07 05:22:38.922 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2022-06-07 05:22:39.043 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2022-06-07 05:22:39.044 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2022-06-07 05:22:39.044 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2022-06-07 05:22:39.046 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-06-07 05:22:39.051 UTC [68] LOG:  database system was shut down at 2022-06-07 05:22:38 UTC
        postgres_1    | 2022-06-07 05:22:39.056 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-06-07 05:22:40.659 UTC [78] ERROR:  unrecognized configuration parameter "sql_safe_updates"
        postgres_1    | 2022-06-07 05:22:40.659 UTC [78] STATEMENT:  SET sql_safe_updates = false;
        test_1        | --- FAIL: TestCompare (4.31s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/mutators (0.14s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:157: CREATE TABLE table1 (col1_0 TIMESTAMPTZ NULL, col1_1 INT8 NULL, col1_2 FLOAT8 NOT NULL, col1_3 FLOAT8 NOT NULL, col1_4 NAME, col1_5 BIT(3), col1_6 FLOAT8 NOT NULL AS (col1_3 + col1_2) VIRTUAL, PRIMARY KEY (col1_2 ASC, col1_6 DESC, col1_3), INDEX (col1_5 ASC, col1_3 ASC, col1_2 ASC), UNIQUE (col1_2 DESC, (col1_2 + col1_3) ASC, col1_5 ASC, col1_3 DESC, col1_0 ASC, col1_1 DESC) STORING (col1_4) WHERE ((((table1.col1_6 = 'NaN':::FLOAT8) AND (table1.col1_2 >= 0.0:::FLOAT8)) OR (table1.col1_4 != e'\x00':::STRING)) AND (table1.col1_1 < (-1):::INT8)) AND (table1.col1_0 <= '294276-12-31 23:59:59.999999+00:00':::TIMESTAMPTZ), INDEX (col1_1 DESC, col1_0 ASC, col1_5 ASC, col1_3 DESC), INDEX (col1_4, col1_5 DESC, col1_0 DESC, col1_2) WHERE (((((table1.col1_3 != 'NaN':::FLOAT8) AND (table1.col1_1 = (-1):::INT8)) AND (table1.col1_6 > (-1.0):::FLOAT8)) AND (table1.col1_0 != '-2000-01-01 00:00:00+00:00':::TIMESTAMPTZ)) AND (table1.col1_2 >= 1.401298464324817e-45:::FLOAT8)) AND (table1.col1_4 != '"':::STRING), INDEX (col1_4, col1_6 ASC, col1_2, col1_0 DESC) PARTITION BY LIST (col1_4) (PARTITION table1_part_0 VALUES IN ((e'$+wc>p_Y\b':::STRING:::NAME,), (e'\x1d"oJ\x06Fj':::STRING:::NAME,), (e' \x0bW6^\\\x1f':::STRING:::NAME,)), PARTITION table1_part_1 VALUES IN (('7yO':::STRING:::NAME,), (e'>\x1f\x1e2R':::STRING:::NAME,), (e'1I\x11f%^\x17':::STRING:::NAME,)), PARTITION table1_part_2 VALUES IN (('"':::STRING,), ('b':::STRING:::NAME,), (e'O:?)\x1f':::STRING:::NAME,)), PARTITION table1_part_3 VALUES IN (('uD(&<m':::STRING:::NAME,), (e'<&W\x02\x11v':::STRING:::NAME,), (e'*\r\x1bxk\x1e\x05r\x11':::STRING:::NAME,)), PARTITION table1_part_4 VALUES IN ((e'^\x04nD\x05g\b=5':::STRING:::NAME,), (e'wdp(P#@\\':::STRING:::NAME,), (e'\x17JCv':::STRING:::NAME,)), PARTITION table1_part_5 VALUES IN (('':::STRING:::NAME,), (e'\x06\x1f\x13\x05L':::STRING:::NAME,)), PARTITION table1_part_6 VALUES IN ((e'\x1e\f9tpO':::STRING:::NAME,), (e'\b(3qnH':::STRING:::NAME,)), PARTITION table1_part_7 VALUES IN ((e's\x02':::STRING:::NAME,), (e'(\tahF\x0f\x13':::STRING:::NAME,), (e'\x19':::STRING:::NAME,)), PARTITION table1_part_8 VALUES IN ((NULL,), (e'\x14@\x06g.Yn':::STRING:::NAME,), (' e(P3iA':::STRING:::NAME,))))
        test_1        |         compare_test.go:158: cockroach1: exec: ERROR: use of partitions requires an enterprise license. see https://cockroachlabs.com/pricing?cluster=cf5a6d5c-a81d-4b1e-a050-cd3ca4e40444 for details on how to enable enterprise features (SQLSTATE XXC02)
        test_1        |     --- FAIL: TestCompare/postgres (1.04s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: postgres
        test_1        |         compare_test.go:157: SET sql_safe_updates = false;
        test_1        |         compare_test.go:158: postgres: exec: ERROR: unrecognized configuration parameter "sql_safe_updates" (SQLSTATE 42704)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:120: exit status 1
--- FAIL: TestComposeCompare (21.61s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78068 compose: TestComposeCompare failed [C-test-failure O-robot T-sql-experience branch-release-22.1]
- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
- #67844 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5407297?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5407297?buildTab=artifacts#/) on master @ [175eebd03e5f2674a2d1f6bfea8bd72f131a6532](https://github.com/cockroachdb/cockroach/commits/175eebd03e5f2674a2d1f6bfea8bd72f131a6532):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down....2022-06-08 05:22:00.596 UTC [49] LOG:  received fast shutdown request
        postgres_1    | 2022-06-08 05:22:00.598 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2022-06-08 05:22:00.600 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2022-06-08 05:22:00.602 UTC [51] LOG:  shutting down
        postgres_1    | 2022-06-08 05:22:00.710 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2022-06-08 05:22:00.818 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2022-06-08 05:22:00.818 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2022-06-08 05:22:00.818 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2022-06-08 05:22:00.822 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-06-08 05:22:00.827 UTC [68] LOG:  database system was shut down at 2022-06-08 05:22:00 UTC
        postgres_1    | 2022-06-08 05:22:00.832 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-06-08 05:22:02.066 UTC [78] ERROR:  unrecognized configuration parameter "sql_safe_updates"
        postgres_1    | 2022-06-08 05:22:02.066 UTC [78] STATEMENT:  SET sql_safe_updates = false;
        test_1        | --- FAIL: TestCompare (4.20s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/mutators (0.20s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:157: CREATE TABLE table2 (col2_0 CHAR, col2_1 CHAR NOT NULL, col2_2 INET NOT NULL, col2_3 REGCLASS NOT NULL, col2_4 REGPROCEDURE NOT NULL, col2_5 BOOL NULL, col2_6 UUID NULL, col2_7 REGROLE NULL, col2_8 INT4 NULL, col2_9 TIMETZ NOT NULL, col2_10 GEOMETRY NOT NULL, col2_11 TIME NULL, col2_12 NAME, col2_13 REGPROC NULL, col2_14 STRING AS (lower(col2_0)) VIRTUAL, col2_15 STRING NOT NULL AS (CASE WHEN col2_4 IS NULL THEN 'f':::STRING ELSE e'\x12nvr"VEe':::STRING END) VIRTUAL, col2_16 STRING AS (lower(col2_0)) VIRTUAL, INDEX (col2_15 DESC, col2_7 ASC, col2_4 ASC, col2_1 DESC, col2_16 DESC, col2_5 ASC, col2_0, col2_12 DESC, col2_11 ASC, col2_8 DESC) STORING (col2_3, col2_6, col2_9, col2_10) PARTITION BY LIST (col2_15, col2_7, col2_4) (PARTITION table2_part_0 VALUES IN ((e'sO)7\x13':::STRING, NULL, 3853080666:::OID)), PARTITION "DEFAULT" VALUES IN ((DEFAULT, DEFAULT, DEFAULT))) WHERE (((((((((table2.col2_11 != '24:00:00':::TIME) AND (table2.col2_15 < '"':::STRING)) OR (NOT table2.col2_5)) AND (table2.col2_8 != (-1):::INT8)) AND (table2.col2_0 < '':::STRING)) AND (table2.col2_12 = e'\x00':::STRING)) OR (table2.col2_1 != 'X':::STRING)) OR (table2.col2_14 <= 'X':::STRING)) AND (table2.col2_16 > '':::STRING)) OR (table2.col2_9 != '24:00:00-15:59:00':::TIMETZ), UNIQUE (col2_16 DESC) STORING (col2_3, col2_4, col2_7, col2_9, col2_10, col2_11, col2_12, col2_13), INDEX (col2_16 DESC, col2_7 DESC) STORING (col2_1, col2_4, col2_8, col2_9), INVERTED INDEX (col2_3 DESC, col2_7 ASC, col2_6 DESC, col2_4 DESC, col2_2 ASC, col2_11 DESC, col2_8, lower(CAST(col2_11 AS STRING)) ASC, lower(CAST(col2_5 AS STRING)) DESC, col2_12, (CASE WHEN col2_13 IS NULL THEN e'<2\x04':::STRING ELSE e'il\x14':::STRING END), col2_10 DESC) WHERE ((((((table2.col2_8 != 0:::INT8) OR (NOT table2.col2_5)) AND (table2.col2_14 <= '':::STRING)) AND (table2.col2_11 = '00:00:00':::TIME)) AND (table2.col2_12 < '':::STRING)) AND (table2.col2_9 >= '24:00:00-15:59:00':::TIMETZ)) AND (table2.col2_1 >= e'\x00':::STRING), INVERTED INDEX (col2_0 DESC, col2_5 DESC, col2_8 DESC, col2_2, col2_10) WHERE ((((table2.col2_9 > '00:00:00+15:59:00':::TIMETZ) AND (table2.col2_11 >= '24:00:00':::TIME)) AND (NOT table2.col2_5)) AND (table2.col2_12 != '"':::STRING)) OR (table2.col2_1 >= e'\x00':::STRING), INDEX (col2_15, col2_6, col2_1 DESC) STORING (col2_0, col2_4, col2_5, col2_8, col2_9, col2_10, col2_13), UNIQUE (col2_12 DESC), INDEX (col2_7 DESC, col2_12, col2_8 ASC, col2_1 DESC, col2_5, col2_16 DESC, col2_4 ASC, col2_13 ASC, col2_14 DESC, col2_11 ASC))
        test_1        |         compare_test.go:158: cockroach1: exec: ERROR: use of partitions requires an enterprise license. see https://cockroachlabs.com/pricing?cluster=d8c8eae8-1b40-4698-b835-e4573769d2e6 for details on how to enable enterprise features (SQLSTATE XXC02)
        test_1        |     --- FAIL: TestCompare/postgres (0.87s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: postgres
        test_1        |         compare_test.go:157: SET sql_safe_updates = false;
        test_1        |         compare_test.go:158: postgres: exec: ERROR: unrecognized configuration parameter "sql_safe_updates" (SQLSTATE 42704)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:120: exit status 1
--- FAIL: TestComposeCompare (22.14s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78068 compose: TestComposeCompare failed [C-test-failure O-robot T-sql-experience branch-release-22.1]
- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5419508?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5419508?buildTab=artifacts#/) on master @ [c070f2c70e0c92573a99dcb80b8a0cb0532bd06f](https://github.com/cockroachdb/cockroach/commits/c070f2c70e0c92573a99dcb80b8a0cb0532bd06f):


```
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2022-06-09 05:23:34.062 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2022-06-09 05:23:34.064 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2022-06-09 05:23:34.066 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2022-06-09 05:23:34.067 UTC [51] LOG:  shutting down
        postgres_1    | 2022-06-09 05:23:34.174 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2022-06-09 05:23:34.285 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2022-06-09 05:23:34.285 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2022-06-09 05:23:34.285 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2022-06-09 05:23:34.289 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-06-09 05:23:34.294 UTC [68] LOG:  database system was shut down at 2022-06-09 05:23:34 UTC
        postgres_1    | 2022-06-09 05:23:34.299 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-06-09 05:23:35.509 UTC [78] ERROR:  unrecognized configuration parameter "sql_safe_updates"
        postgres_1    | 2022-06-09 05:23:35.509 UTC [78] STATEMENT:  SET sql_safe_updates = false;
        test_1        | --- FAIL: TestCompare (4.78s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.89s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: postgres
        test_1        |         compare_test.go:157: SET sql_safe_updates = false;
        test_1        |         compare_test.go:158: postgres: exec: ERROR: unrecognized configuration parameter "sql_safe_updates" (SQLSTATE 42704)
        test_1        |     --- FAIL: TestCompare/mutators (0.76s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: cockroach2
        test_1        |         compare_test.go:157: CREATE TABLE table1 (col1_0 INT4, col1_1 OID NULL, col1_2 INTERVAL NULL, col1_3 BYTES NOT NULL, col1_4 CHAR NULL, col1_5 VARCHAR NULL, col1_6 JSONB NOT NULL, col1_7 STRING NOT NULL, col1_8 STRING NULL AS (CASE WHEN col1_2 IS NULL THEN '':::STRING ELSE e'p.O\\\x1d|61':::STRING END) VIRTUAL, UNIQUE (col1_5 ASC, lower(CAST(col1_6 AS STRING)) ASC) STORING (col1_0, col1_1, col1_2, col1_3), INDEX (lower(col1_4), col1_1) WHERE ((((table1.col1_3 < '\x27':::BYTES) AND (table1.col1_8 != e'\U00002603':::STRING)) AND (table1.col1_4 <= e'\U00002603':::STRING)) AND (table1.col1_0 != (-128):::INT8)) OR (table1.col1_5 <= '':::STRING), INDEX (col1_1, col1_0 ASC, col1_7 ASC) STORING (col1_2) WHERE (table1.col1_3 = e'\\x22':::BYTES) AND (table1.col1_4 > '"':::STRING), FAMILY (col1_0), FAMILY (col1_1), FAMILY (col1_7, col1_5, col1_4, col1_3, col1_2), FAMILY (col1_6));
        test_1        |             ALTER TABLE table1 INJECT STATISTICS '[{"avg_size": 0, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}, {"avg_size": 0, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 0, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 0}]':::JSONB;
        test_1        |             ALTER TABLE table1 INJECT STATISTICS e'[{"avg_size": 0, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1137927, "histo_col_type": "", "name": "__auto__", "null_count": 887445, "row_count": 3000000}, {"avg_size": 0, "columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 497, "histo_buckets": [{"distinct_range": 0, "num_eq": 0, "num_range": 0, "upper_bound": "310844409"}, {"distinct_range": 0, "num_eq": 1000000, "num_range": 50, "upper_bound": "528046157"}, {"distinct_range": 0, "num_eq": 6953844736610931876, "num_range": 1298403246110707711, "upper_bound": "1767388955"}, {"distinct_range": 492139420489967200, "num_eq": 1000, "num_range": 687674801459709399, "upper_bound": "2476324610"}, {"distinct_range": 0, "num_eq": 70000000000, "num_range": 2961467616960948528, "upper_bound": "2626664371"}, {"distinct_range": 0, "num_eq": 6114042065430181489, "num_range": 3806112312998441874, "upper_bound": "2940441661"}, {"distinct_range": 4891469884012798000, "num_eq": 5408535842687689547, "num_range": 8315292553591331015, "upper_bound": "2943693748"}, {"distinct_range": 0, "num_eq": 2324960371935090871, "num_range": 500000, "upper_bound": "3793986467"}], "histo_col_type": "OID", "name": "__auto__", "null_count": 1393413, "row_count": 3000000}, {"avg_size": 0, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2741619, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000000}, {"avg_size": 0, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2557136, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000000}, {"avg_size": 0, "columns": ["col1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1689118, "histo_col_type": "", "name": "__auto__", "null_count": 1012039, "row_count": 3000000}, {"avg_size": 0, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2982701, "histo_col_type": "", "name": "__auto__", "null_count": 383616, "row_count": 3000000}, {"avg_size": 0, "columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1476280, "histo_col_type": "", "name": "__auto__", "null_count": 2340046, "row_count": 3000000}, {"avg_size": 0, "columns": ["col1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1754970, "histo_buckets": [{"distinct_range": 0, "num_eq": 7770682064479963257, "num_range": 0, "upper_bound": ""}, {"distinct_range": 4000000000, "num_eq": 2384322591082841483, "num_range": 4000000000, "upper_bound": "6f3K"}, {"distinct_range": 6000, "num_eq": 400000, "num_range": 6000, "upper_bound": "7"}, {"distinct_range": 2804935536229623000, "num_eq": 0, "num_range": 2804935536229622640, "upper_bound": "H\\u001e"}, {"distinct_range": 200000, "num_eq": 800, "num_range": 200000, "upper_bound": "_O\\t\\u001aZ|\\u001e<\\u0014"}], "histo_col_type": "VARCHAR", "name": "__auto__", "null_count": 2716340, "row_count": 3000000}, {"avg_size": 0, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2652804, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 3000000}]':::JSONB;
        test_1        |             
        test_1        |         compare_test.go:158: cockroach2: exec: ERROR: cannot inject statistics in an explicit transaction (SQLSTATE XXUUU)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:120: exit status 1
--- FAIL: TestComposeCompare (22.40s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78068 compose: TestComposeCompare failed [C-test-failure O-robot T-sql-experience branch-release-22.1]
- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5431523?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5431523?buildTab=artifacts#/) on master @ [3b04ced1801d02b5dd50dab0c93f957c5c1c5720](https://github.com/cockroachdb/cockroach/commits/3b04ced1801d02b5dd50dab0c93f957c5c1c5720):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2022-06-10 05:23:39.206 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2022-06-10 05:23:39.209 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2022-06-10 05:23:39.210 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2022-06-10 05:23:39.211 UTC [51] LOG:  shutting down
        postgres_1    | 2022-06-10 05:23:39.309 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2022-06-10 05:23:39.429 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2022-06-10 05:23:39.429 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2022-06-10 05:23:39.429 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2022-06-10 05:23:39.432 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-06-10 05:23:39.437 UTC [68] LOG:  database system was shut down at 2022-06-10 05:23:39 UTC
        postgres_1    | 2022-06-10 05:23:39.442 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-06-10 05:23:40.750 UTC [78] ERROR:  unrecognized configuration parameter "sql_safe_updates"
        postgres_1    | 2022-06-10 05:23:40.750 UTC [78] STATEMENT:  SET sql_safe_updates = false;
        test_1        | --- FAIL: TestCompare (4.35s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (1.04s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: postgres
        test_1        |         compare_test.go:157: SET sql_safe_updates = false;
        test_1        |         compare_test.go:158: postgres: exec: ERROR: unrecognized configuration parameter "sql_safe_updates" (SQLSTATE 42704)
        test_1        |     --- FAIL: TestCompare/mutators (0.19s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:157: CREATE TABLE table2 (col2_0 REGNAMESPACE NULL, col2_1 DECIMAL, col2_2 JSONB NULL, col2_3 INT2 NULL, col2_4 INT8 NULL, col2_5 INET NOT NULL, col2_6 GEOGRAPHY, col2_7 FLOAT8, INVERTED INDEX (col2_4, col2_3, col2_7, lower(CAST(col2_5 AS STRING)), col2_6 ASC), INDEX (col2_1, col2_3 ASC) STORING (col2_0, col2_6) PARTITION BY LIST (col2_1) (PARTITION table2_part_0 VALUES IN ((129.990496383556609:::DECIMAL,), ((-163.9159649657767518):::DECIMAL,), ((-592407780685763.3446):::DECIMAL,), (NULL,), ((-2.870741973946181703E+33):::DECIMAL,), ((-39726676365.13127546):::DECIMAL,), ((-1.901219654661639901E+20):::DECIMAL,)), PARTITION table2_part_1 VALUES IN (((-5730248.51680695572):::DECIMAL,), ((-1.148191798553284606E+21):::DECIMAL,), (59227264188772737.27:::DECIMAL,), ((-8814859789101497650):::DECIMAL,), (3.192558705095571250E+19:::DECIMAL,), ((-0.06511405126419676185):::DECIMAL,)), PARTITION table2_part_2 VALUES IN ((393756684064223498.1:::DECIMAL,), ((-0.01408455629733399613):::DECIMAL,), (56985379740555.67546:::DECIMAL,), ((-86051690714727773.71):::DECIMAL,), ((-568804857740906.5053):::DECIMAL,), ((-8.759037440016425541):::DECIMAL,)), PARTITION table2_part_3 VALUES IN (((-9.202736268743453605E+34):::DECIMAL,), (0.06937179090340380440:::DECIMAL,), ((-8.049324757185317855E+35):::DECIMAL,), (2.458718759164266381E+22:::DECIMAL,), ((-2.931415104328380770E+34):::DECIMAL,), ((-8.203143575759762504E+27):::DECIMAL,), ((-2.081689434640563933E+24):::DECIMAL,)), PARTITION table2_part_4 VALUES IN (((-194415202219439394.9):::DECIMAL,), (904815331989794.8862:::DECIMAL,), ((-5.632014891002755067E+35):::DECIMAL,), ((-7.907585969153410982E+24):::DECIMAL,), (6.553876520924249645E+37:::DECIMAL,))) WHERE (((table2.col2_1 = '-Infinity':::DECIMAL) OR (table2.col2_7 = 1.0:::FLOAT8)) AND (table2.col2_3 < 1:::INT8)) AND (table2.col2_4 >= 1:::INT8))
        test_1        |         compare_test.go:158: cockroach1: exec: ERROR: use of partitions requires an enterprise license. see https://cockroachlabs.com/pricing?cluster=92b38ebf-b464-4b1f-930f-9f7edd77ea4e for details on how to enable enterprise features (SQLSTATE XXC02)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_postgres_1   ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:120: exit status 1
--- FAIL: TestComposeCompare (21.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78068 compose: TestComposeCompare failed [C-test-failure O-robot T-sql-experience branch-release-22.1 sync-me-8]
- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2 sync-me-8]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5442467?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5442467?buildTab=artifacts#/) on master @ [45ee8026c1997df738ee184fe6a606e927c0a8c5](https://github.com/cockroachdb/cockroach/commits/45ee8026c1997df738ee184fe6a606e927c0a8c5):


```
        postgres_1    | 
        postgres_1    | 2022-06-11 05:18:06.921 UTC [49] LOG:  received fast shutdown request
        postgres_1    | waiting for server to shut down....2022-06-11 05:18:06.922 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2022-06-11 05:18:06.924 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2022-06-11 05:18:06.927 UTC [51] LOG:  shutting down
        postgres_1    | 2022-06-11 05:18:07.023 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2022-06-11 05:18:07.142 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2022-06-11 05:18:07.142 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2022-06-11 05:18:07.142 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2022-06-11 05:18:07.146 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-06-11 05:18:07.150 UTC [68] LOG:  database system was shut down at 2022-06-11 05:18:07 UTC
        postgres_1    | 2022-06-11 05:18:07.155 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-06-11 05:18:08.299 UTC [78] ERROR:  unrecognized configuration parameter "sql_safe_updates"
        postgres_1    | 2022-06-11 05:18:08.299 UTC [78] STATEMENT:  SET sql_safe_updates = false;
        test_1        | --- FAIL: TestCompare (4.31s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.85s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: postgres
        test_1        |         compare_test.go:157: SET sql_safe_updates = false;
        test_1        |         compare_test.go:158: postgres: exec: ERROR: unrecognized configuration parameter "sql_safe_updates" (SQLSTATE 42704)
        test_1        |     --- FAIL: TestCompare/mutators (0.33s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: cockroach2
        test_1        |         compare_test.go:157: CREATE TABLE table1 (col1_0 GEOGRAPHY NOT NULL, col1_1 JSONB NOT NULL, col1_2 REGNAMESPACE NOT NULL, col1_3 FLOAT4 NOT NULL, col1_4 REGPROCEDURE NULL, col1_5 INTERVAL NULL, col1_6 BYTES NOT NULL, col1_7 TIME NULL, col1_8 BOOL NOT NULL, col1_9 INT8, col1_10 GEOMETRY NOT NULL, col1_11 STRING, PRIMARY KEY (col1_8 ASC), INVERTED INDEX (col1_2 ASC, col1_11 DESC, col1_8 DESC, lower(CAST(col1_7 AS STRING)) DESC, col1_3 ASC, col1_4 DESC, col1_1), FAMILY (col1_10, col1_11, col1_9), FAMILY (col1_0, col1_7), FAMILY (col1_8), FAMILY (col1_1), FAMILY (col1_2), FAMILY (col1_5), FAMILY (col1_3), FAMILY (col1_6), FAMILY (col1_4));
        test_1        |             ALTER TABLE table1 INJECT STATISTICS '[{"avg_size": 0, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 7138885, "histo_buckets": [{"distinct_range": 0, "num_eq": 216839502106281744, "num_range": 0, "upper_bound": "0"}, {"distinct_range": 553334167843516740, "num_eq": 3262208781208111666, "num_range": 7328911732145734186, "upper_bound": "789743258"}, {"distinct_range": 10518989022.48871, "num_eq": 4257593180562062446, "num_range": 80000000000, "upper_bound": "807134913"}, {"distinct_range": 2.1939862114194737, "num_eq": 40, "num_range": 30, "upper_bound": "934710387"}, {"distinct_range": 10000000000, "num_eq": 300000000, "num_range": 10000000000, "upper_bound": "2061649395"}, {"distinct_range": 225716239112081200, "num_eq": 44409057729778653, "num_range": 3637178065278467404, "upper_bound": "2318838620"}, {"distinct_range": 2313861684454421500, "num_eq": 400, "num_range": 6622525685318977003, "upper_bound": "3809522926"}], "histo_col_type": "REGNAMESPACE", "name": "__auto__", "null_count": 0, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 8530384, "histo_col_type": "", "name": "__auto__", "null_count": 8680309, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4767773, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 8760309, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6133748, "histo_col_type": "", "name": "__auto__", "null_count": 3288797, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5627182, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6720384, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6462051, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 7164371, "histo_col_type": "", "name": "__auto__", "null_count": 9365111, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1991582, "histo_col_type": "", "name": "__auto__", "null_count": 6430754, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2058913, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 10000000}, {"avg_size": 0, "columns": ["col1_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 22856, "histo_col_type": "", "name": "__auto__", "null_count": 6715370, "row_count": 10000000}]':::JSONB;
        test_1        |             ALTER TABLE table1 INJECT STATISTICS '[{"avg_size": 0, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 76476436693798161, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 225134853673172639, "histo_buckets": [{"distinct_range": 0, "num_eq": 40000000000, "num_range": 0, "upper_bound": "1033846845"}, {"distinct_range": 0, "num_eq": 1000000000, "num_range": 400000000, "upper_bound": "1491264647"}, {"distinct_range": 0, "num_eq": 100, "num_range": 5139714819243617631, "upper_bound": "1512447460"}, {"distinct_range": 4163190301.6819677, "num_eq": 298433708071323700, "num_range": 5000000000, "upper_bound": "2045658077"}, {"distinct_range": 3155972780397620700, "num_eq": 8016976194013708937, "num_range": 3155972780397620668, "upper_bound": "2385017309"}, {"distinct_range": 0, "num_eq": 2504457514911022744, "num_range": 5784291359011465700, "upper_bound": "3144972206"}, {"distinct_range": 3124868.9671007614, "num_eq": 60000000, "num_range": 6000000, "upper_bound": "3372099562"}, {"distinct_range": 0, "num_eq": 30000000000, "num_range": 7000000, "upper_bound": "4033642954"}, {"distinct_range": 70, "num_eq": 4272702456326667932, "num_range": 70, "upper_bound": "4099504049"}], "histo_col_type": "REGNAMESPACE", "name": "__auto__", "null_count": 0, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 422820713607723094, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 253734401008100404, "histo_col_type": "", "name": "__auto__", "null_count": 360739588237421296, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 55041285573131413, "histo_col_type": "", "name": "__auto__", "null_count": 402174950452284536, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 240202579872218349, "histo_col_type": "", "name": "__auto__", "null_count": 160595861987723142, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 106266342822829135, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 362252400877740175, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 307875808621582596, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 35694478586103177, "histo_col_type": "", "name": "__auto__", "null_count": 294509733571183916, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 198132911562923830, "histo_col_type": "", "name": "__auto__", "null_count": 341640580741255770, "row_count": 449720504768073439}, {"avg_size": 0, "columns": ["col1_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 347686549423465411, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 449720504768073439}]':::JSONB;
        test_1        |             
        test_1        |         compare_test.go:158: cockroach2: exec: ERROR: cannot inject statistics in an explicit transaction (SQLSTATE XXUUU)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:120: exit status 1
--- FAIL: TestComposeCompare (21.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78068 compose: TestComposeCompare failed [C-test-failure O-robot T-sql-experience branch-release-22.1 sync-me-8]
- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2 sync-me-8]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5448043?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5448043?buildTab=artifacts#/) on master @ [45ee8026c1997df738ee184fe6a606e927c0a8c5](https://github.com/cockroachdb/cockroach/commits/45ee8026c1997df738ee184fe6a606e927c0a8c5):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | 2022-06-12 05:19:14.601 UTC [49] LOG:  received fast shutdown request
        postgres_1    | waiting for server to shut down....2022-06-12 05:19:14.602 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2022-06-12 05:19:14.604 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2022-06-12 05:19:14.606 UTC [51] LOG:  shutting down
        postgres_1    | 2022-06-12 05:19:14.702 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2022-06-12 05:19:14.822 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2022-06-12 05:19:14.822 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2022-06-12 05:19:14.822 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2022-06-12 05:19:14.825 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-06-12 05:19:14.829 UTC [68] LOG:  database system was shut down at 2022-06-12 05:19:14 UTC
        postgres_1    | 2022-06-12 05:19:14.834 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-06-12 05:19:15.960 UTC [78] ERROR:  unrecognized configuration parameter "sql_safe_updates"
        postgres_1    | 2022-06-12 05:19:15.960 UTC [78] STATEMENT:  SET sql_safe_updates = false;
        test_1        | --- FAIL: TestCompare (4.05s)
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     --- FAIL: TestCompare/postgres (0.78s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: postgres
        test_1        |         compare_test.go:157: SET sql_safe_updates = false;
        test_1        |         compare_test.go:158: postgres: exec: ERROR: unrecognized configuration parameter "sql_safe_updates" (SQLSTATE 42704)
        test_1        |     --- FAIL: TestCompare/mutators (0.13s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:157: CREATE TABLE table2 (col2_0 INTERVAL NOT NULL, col2_1 BOX2D NOT NULL, col2_2 BIT(28) NOT NULL, col2_3 TIMESTAMP NOT NULL, col2_4 FLOAT8 NOT NULL, col2_5 FLOAT8 NOT NULL AS (col2_4 + NULL) STORED, PRIMARY KEY (col2_2 DESC, col2_1 DESC, col2_0 DESC, col2_5 ASC, col2_4 ASC, col2_3 ASC), INDEX (col2_3, col2_0 DESC, col2_1, col2_2, col2_5 DESC) PARTITION BY LIST (col2_3) (PARTITION table2_part_0 VALUES IN (('2008-09-12 10:12:04.000219':::TIMESTAMP,), ('1993-05-28 17:09:48.000352':::TIMESTAMP,), ('1984-03-05 05:21:30.000071':::TIMESTAMP,), ('2003-12-01 23:26:45.000554':::TIMESTAMP,), ('1992-01-03 15:27:53.00051':::TIMESTAMP,)), PARTITION "DEFAULT" VALUES IN ((DEFAULT,))) WHERE ((table2.col2_5 < '+Inf':::FLOAT8) OR (table2.col2_4 != '-Inf':::FLOAT8)) AND (table2.col2_3 = '3000-01-01 00:00:00':::TIMESTAMP), INDEX (col2_2, col2_1 ASC, col2_4 DESC, col2_0, col2_5 DESC, col2_3 ASC), INDEX (col2_2 ASC, (CASE WHEN col2_3 IS NULL THEN e'\x06@.Ls8\x11':::STRING ELSE 'PQ':::STRING END), col2_1 DESC) PARTITION BY LIST (col2_2) (PARTITION table2_part_0 VALUES IN ((B'1001000001010000011000011100',), (B'1100010000100101000101101001',), (B'1100100101010010000101011110',), (B'0101000010000101000100011011',), (B'0100110001110000011110100101',), (B'0010111110001001000010010100',), (B'0101010001110111100000101111',), (B'1001000010100101001111101000',), (B'0100000000010100010111001111',)), PARTITION table2_part_1 VALUES IN ((B'0010011011101101111101100110',), (B'0011011000000011001100000100',), (B'0000101111110110000111001010',), (B'0000000001010001010001111100',), (B'0110000111000010001110101101',), (B'0101100111111101000100000101',), (B'0100100100000011101010011010',), (B'0111111001111001111010100101',), (B'1111101101111111011011000101',)), PARTITION table2_part_2 VALUES IN ((B'0000011001100110100011001100',), (B'0011000101001100111100001000',), (B'1100111100101011000011000100',), (B'0010010100111010001011110110',), (B'0010011111010100010100010010',), (B'1001101001101011000110000100',), (B'0111010100110000010010111010',), (B'1000000011100111101001100011',), (B'1100011011011110011100011010',)), PARTITION table2_part_3 VALUES IN ((B'0011001111011101111001000001',), (B'1111111100001110000011111001',), (B'1101110111011101110101010111',), (B'0100000111101000100000001010',), (B'1011100110010111001100110100',), (B'0101101001011001110100111001',), (B'1000110100011000101110110111',), (B'1010011101010001001100100101',), (B'0010100001110010101110000001',)), PARTITION table2_part_4 VALUES IN ((B'1100011110001100111010000001',), (B'1000001111001110000101000101',), (B'0000000000000000000000000000',), (B'0110001101110100101011110110',), (B'0101111111110111000010011101',), (B'1110010101101011101001000010',), (B'0111111111111111111111111111',), (B'0100101111001100000110111000',), (B'1110001101110010000101011000',)), PARTITION "DEFAULT" VALUES IN ((DEFAULT,))), INDEX (col2_4 ASC, col2_3 ASC, col2_1 ASC, col2_2) WHERE ((table2.col2_5 < (-1.0):::FLOAT8) AND (table2.col2_3 > '-2000-01-01 00:00:00':::TIMESTAMP)) OR (table2.col2_4 < 1.401298464324817e-45:::FLOAT8), FAMILY (col2_4), FAMILY (col2_1, col2_2), FAMILY (col2_3), FAMILY (col2_0, col2_5))
        test_1        |         compare_test.go:158: cockroach1: exec: ERROR: use of partitions requires an enterprise license. see https://cockroachlabs.com/pricing?cluster=8e8015ce-0364-449c-b7ac-0092a4254d84 for details on how to enable enterprise features (SQLSTATE XXC02)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach2_1 ... 
        Stopping compare_cockroach1_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:120: exit status 1
--- FAIL: TestComposeCompare (20.91s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78068 compose: TestComposeCompare failed [C-test-failure O-robot T-sql-experience branch-release-22.1 sync-me-8]
- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2 sync-me-8]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

compose.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5453831?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/5453831?buildTab=artifacts#/) on master @ [87da966cbc15e058591d80c477c78d6059483a5c](https://github.com/cockroachdb/cockroach/commits/87da966cbc15e058591d80c477c78d6059483a5c):


```
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | CREATE EXTENSION
        postgres_1    | 
        postgres_1    | waiting for server to shut down...2022-06-13 05:18:50.632 UTC [49] LOG:  received fast shutdown request
        postgres_1    | .2022-06-13 05:18:50.634 UTC [49] LOG:  aborting any active transactions
        postgres_1    | 2022-06-13 05:18:50.636 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
        postgres_1    | 2022-06-13 05:18:50.638 UTC [51] LOG:  shutting down
        postgres_1    | 2022-06-13 05:18:50.732 UTC [49] LOG:  database system is shut down
        postgres_1    |  done
        postgres_1    | server stopped
        postgres_1    | 
        postgres_1    | PostgreSQL init process complete; ready for start up.
        postgres_1    | 
        postgres_1    | 2022-06-13 05:18:50.853 UTC [1] LOG:  starting PostgreSQL 13.5 (Debian 13.5-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
        postgres_1    | 2022-06-13 05:18:50.853 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
        postgres_1    | 2022-06-13 05:18:50.853 UTC [1] LOG:  listening on IPv6 address "::", port 5432
        postgres_1    | 2022-06-13 05:18:50.856 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
        postgres_1    | 2022-06-13 05:18:50.860 UTC [68] LOG:  database system was shut down at 2022-06-13 05:18:50 UTC
        postgres_1    | 2022-06-13 05:18:50.864 UTC [1] LOG:  database system is ready to accept connections
        postgres_1    | 2022-06-13 05:18:51.799 UTC [78] ERROR:  unrecognized configuration parameter "sql_safe_updates"
        postgres_1    | 2022-06-13 05:18:51.799 UTC [78] STATEMENT:  SET sql_safe_updates = false;
        test_1        | --- FAIL: TestCompare (3.47s)
        test_1        |     compare_test.go:117: Checking connection to: cockroach1
        test_1        |     compare_test.go:117: Checking connection to: cockroach2
        test_1        |     compare_test.go:117: Checking connection to: postgres
        test_1        |     --- FAIL: TestCompare/postgres (0.82s)
        test_1        |         compare_test.go:126: starting test: postgres
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:135: initializing connection: postgres
        test_1        |         compare_test.go:157: SET sql_safe_updates = false;
        test_1        |         compare_test.go:158: postgres: exec: ERROR: unrecognized configuration parameter "sql_safe_updates" (SQLSTATE 42704)
        test_1        |     --- FAIL: TestCompare/mutators (0.15s)
        test_1        |         compare_test.go:126: starting test: mutators
        test_1        |         compare_test.go:135: initializing connection: cockroach1
        test_1        |         compare_test.go:157: CREATE TABLE table1 (col1_0 TIMESTAMPTZ NOT NULL, col1_1 REGROLE NOT NULL, col1_2 TIME NOT NULL, col1_3 REGPROC NOT NULL, col1_4 INT4 NOT NULL, col1_5 TIMESTAMP NOT NULL, col1_6 BOX2D NOT NULL, col1_7 INT4 NOT NULL AS (col1_4 + (-1198032512):::INT8) STORED, col1_8 STRING NULL AS (CASE WHEN col1_5 IS NULL THEN e'5\nP':::STRING ELSE e'\n\x19Z\x01F\'O\x16``':::STRING END) STORED, PRIMARY KEY (col1_4 DESC, col1_6 ASC, col1_7, col1_1 DESC, col1_3 ASC, col1_2, col1_5 ASC, col1_0 ASC), INDEX (col1_7 ASC), UNIQUE (col1_3 DESC, col1_1, col1_7, (CASE WHEN col1_3 IS NULL THEN e'B\x16!G\x05\x0e\x13':::STRING ELSE '%':::STRING END), col1_8 ASC, col1_4 ASC, col1_6 ASC, col1_0, col1_2 DESC), UNIQUE ((CASE WHEN col1_3 IS NULL THEN '``n/&G':::STRING ELSE '':::STRING END) ASC, col1_1 ASC, (CASE WHEN col1_5 IS NULL THEN '(o':::STRING ELSE '&_':::STRING END), lower(CAST(col1_6 AS STRING)), col1_5 ASC, col1_0 ASC, col1_3) WHERE table1.col1_2 < '24:00:00':::TIME, UNIQUE (col1_7 ASC, col1_5, col1_0, col1_4 DESC, col1_8 ASC, col1_3, col1_2 DESC, col1_1, col1_6 ASC), UNIQUE (col1_0 ASC, col1_1) WHERE (table1.col1_2 = '24:00:00':::TIME) OR (table1.col1_7 >= (-32768):::INT8), UNIQUE (col1_3 ASC, lower(CAST(col1_6 AS STRING)) DESC, col1_6 ASC) PARTITION BY LIST (col1_3) (PARTITION table1_part_0 VALUES IN ((1650892768:::OID,), (4021388836:::OID,), (341432551:::OID,), (1749894998:::OID,)), PARTITION table1_part_1 VALUES IN ((3002741814:::OID,), (1482878132:::OID,), (187141501:::OID,), (1757231414:::OID,)), PARTITION table1_part_2 VALUES IN ((1216824933:::OID,), (1751175650:::OID,), (95078095:::OID,), (390410651:::OID,)), PARTITION table1_part_3 VALUES IN ((1123384399:::OID,), (166317912:::OID,), (3500916421:::OID,), (4158118632:::OID,)), PARTITION table1_part_4 VALUES IN ((316014510:::OID,), (2700160855:::OID,), (4131574108:::OID,), (2831850478:::OID,))), INDEX (col1_5 DESC) WHERE ((table1.col1_5 < '-4713-11-24 00:00:00':::TIMESTAMP) AND (table1.col1_2 <= '00:00:00':::TIME)) OR (table1.col1_0 <= '3000-01-01 00:00:00+00:00':::TIMESTAMPTZ), FAMILY (col1_8), FAMILY (col1_6), FAMILY (col1_5, col1_1, col1_7), FAMILY (col1_2, col1_0, col1_4), FAMILY (col1_3))
        test_1        |         compare_test.go:158: cockroach1: exec: ERROR: use of partitions requires an enterprise license. see https://cockroachlabs.com/pricing?cluster=048fc64b-2b8d-4392-a6c2-767d324ab9e7 for details on how to enable enterprise features (SQLSTATE XXC02)
        test_1        | FAIL
        compare_test_1 exited with code 1
        Stopping compare_cockroach1_1 ... 
        Stopping compare_cockroach2_1 ... 
        Stopping compare_postgres_1   ... 
        Stopping compare_postgres_1   ... done
        Stopping compare_cockroach2_1 ... done
        Stopping compare_cockroach1_1 ... done
        Aborting on container exit...
        
    compose_test.go:120: exit status 1
--- FAIL: TestComposeCompare (20.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78068 compose: TestComposeCompare failed [C-test-failure O-robot T-sql-experience branch-release-22.1 sync-me-8]
- #73421 compose: TestComposeCompare failed [C-test-failure O-robot branch-release-21.2 sync-me-8]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/compose/compare/compare/compare_test.go
+++ b/pkg/compose/compare/compare/compare_test.go
@@ -50,6 +50,8 @@ func TestCompare(t *testing.T) {
 				"CREATE EXTENSION IF NOT EXISTS postgis",
 				"CREATE EXTENSION IF NOT EXISTS postgis_topology",
 				"CREATE EXTENSION IF NOT EXISTS fuzzystrmatch;",
+				"CREATE EXTENSION IF NOT EXISTS \"uuid-ossp\";",
+				"CREATE EXTENSION IF NOT EXISTS pg_trgm;",
 			},
 		},
 		"cockroach1": {

--- a/pkg/sql/create_extension.go
+++ b/pkg/sql/create_extension.go
@@ -45,13 +45,14 @@ func (n *createExtensionNode) startExec(params runParams) error {
 	switch n.CreateExtension.Name {
 	case "postgis",
 		"pg_trgm",
+		"fuzzystrmatch",
+		"pgcrypto",
 		"uuid-ossp":
 		telemetry.Inc(sqltelemetry.CreateExtensionCounter(n.CreateExtension.Name))
 		return nil
 	case "postgis_raster",
 		"postgis_topology",
 		"postgis_sfcgal",
-		"fuzzystrmatch",
 		"address_standardizer",
 		"address_standardizer_data_us",
 		"postgis_tiger_geocoder":
@@ -85,7 +86,6 @@ func (n *createExtensionNode) startExec(params runParams) error {
 		"pageinspect",
 		"passwordcheck",
 		"pg_buffercache",
-		"pgcrypto",
 		"pg_freespacemap",
 		"pg_prewarm",
 		"pgrowlocks",

--- a/pkg/sql/randgen/mutator.go
+++ b/pkg/sql/randgen/mutator.go
@@ -663,6 +663,7 @@ var postgresStatementMutator MultiStatementMutation = func(rng *rand.Rand, stmts
 	for _, stmt := range stmts {
 		switch stmt := stmt.(type) {
 		case *tree.SetClusterSetting, *tree.SetVar, *tree.AlterTenantSetClusterSetting:
+			changed = true
 			continue
 		case *tree.CreateTable:
 			if stmt.PartitionByTable != nil {

